---
title: "sp2d2300_dcs_analysis"
author: "Haider Inam"
date: "2/27/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(stringr)
library(dplyr)
library(ggplot2)
library(plotly)

cleanup=theme_bw() +
  theme(plot.title = element_text(hjust=.5),
        panel.grid.major = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.background = element_blank(),
        # axis.line = element_line(color = "black"),
        axis.text = element_text(color="black",size="11"),
        text=element_text(size=11),
        axis.title=element_text(face="bold",size="11"))
```

```{r}
# tab=read.table("Sp2_D2_300.1.consensus.variant-calls.mut",header=T,stringsAsFactors = F)
alldata_abl1app=data.frame(read.table("Sp2_D2_300.1.consensus.variant-calls.mut",header = T,stringsAsFactors = F,check.names = F,fill=T))

alldata_abl1app=alldata_abl1app%>%filter(!variation_type%in%"indel",!alt%in%"<DEL>")

alldata_abl1app$alt_start_hg38=alldata_abl1app$start
# a=alldata_abl1app
alldata_abl1app$alt_start_hg38=alldata_abl1app$alt_start_hg38+1
# alldata_abl1app$end=alldata_abl1app$end+1
# a=alldata_abl1app
alldata_abl1app=vep_fromdf(alldata_abl1app)

alldata_abl1app=alldata_abl1app%>%rowwise()%>%mutate(substitution=gsub("/",protein_start,amino_acids))
# alldata_abl1app=alldata_abl1app%>%arrange(desc(ct))
truemutants=alldata_abl1app$substitution
alldata_abl1app$truemutant=T

ggplot(alldata_abl1app%>%filter(alt_depth<=5000,!amino_acids%in%c("L","G")),aes(reorder(substitution,-alt_depth),alt_depth/5000,fill=truemutant))+geom_col(color="black",position=position_dodge())+scale_x_discrete("Subtitution")+scale_y_continuous("Mutant Allele Frequency")+
  cleanup+
  theme(legend.position = "none",
        axis.text.x=element_text(angle=90,hjust=.5,vjust=.5),
        axis.title.x=element_blank())+
  scale_fill_manual(values=c("darkgreen"))
ggsave("sp2_d2_300_dcs_barplot.pdf",useDingbats=F)

```


Looking at mutants that don't show up in the twinstrand dcs app but show up in our app
Conclusion: so far, most mutants that don't show up in twinstrand's app but show up in our app have mutants that show up within the first 5 nucleotides
```{r}
alldata_dunovoapp=read.csv("analyzed/sp2d2300_snps_annotated_dscs.csv",header = T,stringsAsFactors = F)
alldata_dunovoapp=alldata_dunovoapp[,-1]
# a=alldata_dunovoapp%>%filter(protein_start%in%c(250))
a=alldata_dunovoapp%>%filter(protein_start%in%c(265,267,288,292,296,300,359,361,382,386,388,394,401,418,422,426,429,436,450,462),!amino_acids%in%"F/C")


alldata_dunovoapp=read.csv("analyzed/sp2d2300_sscs_calls.csv",header = T,stringsAsFactors = F)
alldata_dunovoapp=alldata_dunovoapp[,-1]
alldata_dunovoapp$substitution=c(alldata_dunovoapp$protein_start,alldata_dunovoapp$amino_acids)
alldata_dunovoapp=alldata_dunovoapp%>%mutate(substitution=paste(protein_start,amino_acids,sep=""))
alldata_dunovoapp=alldata_dunovoapp%>%mutate(substitution=gsub("/",protein_start,amino_acids))
alldata_dunovoapp=alldata_dunovoapp%>%arrange(desc(ct))
alldata_dunovoapp=alldata_dunovoapp%>%
  mutate(truemutant=case_when(substitution%in%truemutants~T,
                              T~F))

ggplot(alldata_dunovoapp%>%filter(ct<=5000,ct>=2,!amino_acids%in%c("L","G")),aes(reorder(substitution,-ct),ct/5000,fill=truemutant))+geom_col(color="black",position=position_dodge())+scale_x_discrete("Subtitution")+scale_y_continuous("Mutant Allele Frequency")+
  cleanup+
  theme(legend.position = "none",
        axis.text.x=element_text(angle=90,hjust=.5,vjust=.5),
        axis.title.x=element_blank())
ggsave("sp2_d2_300_sscs_barplot.pdf",width=12,height=3,units="in",useDingbats=F)


ggplot(alldata_dunovoapp%>%filter(ct>=15,!amino_acids%in%c("L","G")),aes(reorder(substitution,-ct),ct/25000,fill=truemutant))+geom_col(color="black",position=position_dodge())+scale_x_discrete("Subtitution")+scale_y_continuous("Mutant Allele Frequency")+
  cleanup+
  theme(legend.position = "none",
        axis.text.x=element_text(angle=90,hjust=.5,vjust=.5),
        axis.title.x=element_blank())+
  scale_fill_manual(values=c("red","darkgreen"))
# ggsave("sp2_d2_300_sscs_barplot_zoomed.pdf",width=4,height=3,units="in",useDingbats=F)
ggsave("sp2_d2_300_sscs_barplot_zoomed.pdf",useDingbats=F)

ggplot(alldata_dunovoapp%>%filter(ct<=5000,ct>=15,!amino_acids%in%c("L","G")),aes(reorder(substitution,-ct),ct/25000,fill=truemutant))+geom_col(color="black",position=position_dodge())+scale_x_discrete("Subtitution")+scale_y_continuous("Mutant Allele Frequency")+
  cleanup+
  theme(legend.position = "none",
        axis.text.x=element_text(angle=90,hjust=.5,vjust=.5),
        axis.title.x=element_blank())+
  scale_fill_manual(values=c("red","darkgreen"))
# ggsave("sp2_d2_300_sscs_barplot_zoomed.pdf",width=4,height=3,units="in",useDingbats=F)
ggsave("sp2_d2_300_sscs_barplot_doublezoomed.pdf",useDingbats=F)


coverage_comparison=data.frame(rbind(c("dcs","10000"),c("sscs","50000")))
colnames(coverage_comparison)=c("type","coverage")
coverage_comparison$coverage=as.numeric(coverage_comparison$coverage)
ggplot(coverage_comparison,aes(x=type,y=coverage))+geom_col(color="black")+scale_y_continuous(limits=c(0,55000))+cleanup
ggsave("coveragecomparison.pdf",useDingbats=F)
```

for BMES conference
can take out later
```{r}
coverage_comparison=data.frame(rbind(c("dcs","1e-7"),c("sscs","1e-4"),c("ngs","1e-2")))
colnames(coverage_comparison)=c("type","errorrate")
coverage_comparison$errorrate=as.numeric(as.character(coverage_comparison$errorrate))

coverage_comparison$type=factor(coverage_comparison$type,levels=c("ngs","sscs","dcs"))
ggplot(coverage_comparison,aes(x=type,y=errorrate))+geom_col(color="black")+scale_y_continuous(trans="log10")+cleanup
ggsave("dcs_errorrates.pdf",useDingbats=F)
```

