---
title: "mnv_caller_practice"
author: "Haider Inam"
date: "10/23/2021"
output: html_document
---

```{r setup, include=FALSE}
# rm(list=ls())
knitr::opts_chunk$set(echo = TRUE)
library("stringr")
library(dplyr)
library(ggplot2)
library(plotly)
```

####Function that uses start and stop positions and returns sequence
```{r}
#Reading ABL1 sequence. Starts
abl1_seq=read.table("abl1_sequence.txt")
abl1_seq=as.character(abl1_seq)
# substr(abl1_seq,1,2)
abl1_search=function(start,stop){
  seachresult=substr(abl1_seq,start-130835253,stop-130835253-1)
  seachresult
}

abl_genomic_coordinates=read.csv("../abl1_genomic_coordinates.csv",header = T,stringsAsFactors = F)
abl_genomic_coordinates=abl_genomic_coordinates%>%dplyr::select(-X)

####Adding sequence to each codon###
abl_genomic_coordinates=abl_genomic_coordinates%>%
  rowwise()%>%
  mutate(codon=abl1_search(start,end+1))

```

####Function that returns the codon number given the genomic position
```{r}
ref_codon_finder=function(position){
  abl_genomic_coordinates$match=NA
  match=abl_genomic_coordinates$match

  for(i in 1:length(abl_genomic_coordinates$start)){
    match[i]=position>=abl_genomic_coordinates$start[i]&&position<=abl_genomic_coordinates$end[i]
  }
  if(TRUE%in%match){
    return(abl_genomic_coordinates[grep(TRUE,match),1])
  }
  # else{
    return(NaN)
  # }
}
# position=130835456
# (ref_codon_finder(130835456))
# ref_codon_finder(130854218)
# ref_codon_finder(130873014)

```

####Functions to use Ensembl's VEP to find protein consequence of mutations.
```{r}
# The ensembl server where you can download multiple things from
# https://rest.ensembl.org/
# https://rest.ensembl.org/vep/human/region/9:130884090:130884091/A
library(httr)
library(jsonlite)
library(xml2)
# region="130862976:130862976/A"
# region="130854096:130854096/A" #splice variant
# region="130854240:130854240/G" #throws bad http error
# region="130854096:130854096/A" #first missense in dataset

vep_fromquery=function(region){
  #This function fetches ensembl vep for queries that look like this
  #Needs the packages httr, jsonlite, xml2
server <- "https://rest.ensembl.org"
ext=paste("/vep/human/region/9:",gsub("\"","",region),"?",sep = "")
# ext <- "/vep/human/region/9:130862976:130862976/A?"
 
r <- GET(paste(server, ext, sep = ""), content_type("application/json"))
 
stop_for_status(r)
 
# use this if you get a simple nested list back, otherwise inspect its structure
rest_results=(fromJSON(toJSON(content(r))))
transcript_consequences=(rest_results$transcript_consequences)[[1]]
transcript_consequences=transcript_consequences%>%filter(transcript_id%in%"ENST00000318560")
transcript_consequences
}

```


```{r}
# a=snps_sum[c(1:50),]
# a=mnv_sum
# a=vep_fromdf(a)
# input_df=a
vep_fromdf=function(input_df){
  #Inputs: Dataframe with these fields: alt_start_pos, alt
  #Optional field in dataframe: alt_end_pos (for mnvs because their end position is different than their starting position)
  #This function adds ensembl variant effect predictor annotations to dataframes.
  #This needs the vep_fromquery function
  #Requires the dplyr package
input_df=input_df%>%filter(!ref==alt)

input_df=input_df%>%mutate(protein_start=NA,protein_end=NA,amino_acids=NA,codons=NA,impact=NA,polyphen_prediction=NA,consequence_terms=NA)
# input_df=snps_sum
# input_df=mnv_sum
# i=1
if(sum(as.numeric(grep("alt_end_pos",colnames(input_df))))==0){
  input_df$alt_end_pos=input_df$alt_start_pos+1
}
# i=1
for(i in 1:nrow(input_df)){
  transcript_consequences=vep_fromquery(paste(input_df$alt_start_pos[i],":",input_df$alt_end_pos[i]-1,"/",input_df$alt[i],sep = ""))
  # transcript_consequences=vep_fromquery(paste(130884390,":",130884390,"/",input_df$alt[i],sep = ""))
  # transcript_consequences=vep_fromquery(paste(130862864,":",130862864,"/","C",sep = ""))
  # transcript_consequences$protein_start
  
  if(grepl("splice|intron|stop",as.character(transcript_consequences$consequence_terms))%in%TRUE){
    input_df$consequence_terms[i]=transcript_consequences$consequence_terms[[1]][1]
  }
  else{
    input_df$protein_start[i]=transcript_consequences$protein_start[[1]][1]
    input_df$protein_end[i]=transcript_consequences$protein_end[[1]][1]
    input_df$amino_acids[i]=transcript_consequences$amino_acids[[1]][1]
    input_df$codons[i]=transcript_consequences$codons[[1]][1]
    input_df$impact[i]=transcript_consequences$impact[[1]][1]
    if(grepl("synonymous|frameshift",as.character(transcript_consequences$consequence_terms))%in%TRUE){
      input_df$polyphen_prediction[i]=NA
    }
    else{
      input_df$polyphen_prediction[i]=transcript_consequences$polyphen_prediction[[1]][1]  
    }
    input_df$consequence_terms[i]=transcript_consequences$consequence_terms[[1]][1]
    # transcript_consequences$consequence_terms[[1]][1]
  }
}
input_df
}
```





```{r}
# library(dplyr)
setwd("~/OneDrive - The Pennsylvania State University/RProjects/duplex_sequencing_screen/data/Dunovo/Novogene_lane2/")
files=list.files(".")
#Grabbing only csvs that contain sorted and sscs in their name
files=files[grepl("*.tsv",files)]
# for(filecurr in 1:length(files)){
# skipped: #7 and 16, 19
for(filecurr in 9:length(files)){
  # filecurr=9
# for(filecurr in 9){
setwd("~/OneDrive - The Pennsylvania State University/RProjects/duplex_sequencing_screen/data/Dunovo/Novogene_lane2/")
# alldata=read.table("d0_dcs_filtered_bigger.tsv",header = F,stringsAsFactors = F)
alldata=read.table("d0_dcs_filtered.tsv",header = F,stringsAsFactors = F)
# rm(list=ls())

###This script looks at the a sam file, filters out non-reference, non-E255I, non-SNP. And makes two databases, one for mnvs and one for snps.
# alldata=read.table("filtered.tsv",header = F,stringsAsFactors = F)
# alldata=read.table("il3indep1_dcs_filtered.tsv",header = F,stringsAsFactors = F)
# alldata=read.table("il3indep1_sscs_filtered.tsv",header = F,stringsAsFactors = F)
# alldata=read.table("il3indep1_sscs_all_filtered.tsv",header = F,stringsAsFactors = F)
# alldata=read.table("il3indep1_dcs_all_filtered.tsv",header = F,stringsAsFactors = F)
# alldata=read.table("sor",header = F,stringsAsFactors = F)
names(alldata)=c("pos","cigar","tlen","seq","mdz")
# names(alldata)=c("flag","pos","mapq","cigar","tlen","seq","mdz")
alldata=alldata%>%mutate(mdz=gsub("MD:Z:","",mdz))

######Detecting soft-clipped reads and noting how many bases were soft clipped
#This is because the positions on the mdz field exclude the soft clipped portion of the read.
#The soft clipped reads that are relevant to us are reads in which the soft clip appears before the mismatch. b/c if a clip appears after a mismatch, that's the part of the mdz field that i don't care about. Therefore, I'm gonna care about those only. Btw only 1/3rd of soft clipped reads seem to have a clip followed by a mismatch.
# sum(as.numeric(grepl("S.*M",alldata$cigar)))
# sum(as.numeric(grepl("M.*S",alldata$cigar)))
# # strsplit("1234S123","S")[[1]][1]
alldata$soft_clipped=grepl("S.*M",alldata$cigar)

alldata=alldata%>%
  rowwise()%>%
  mutate(clipdistance=case_when(soft_clipped%in%TRUE~strsplit(cigar,"S")[[1]][1],
                                TRUE~"0"))
alldata$clipdistance=as.numeric(alldata$clipdistance)


#######Filtering out deletions########
alldata=alldata[!grepl("\\^",alldata$mdz),]

#####Filtering out any reads containing Ns####
# Gets rid of 10% of reads. This might be too stringent a filter because of course you would expect Ns in a 10bp window###
alldata=alldata[!grepl("N",alldata$seq),]

#######Filtering for Non-Reference########
# alt_positions=grepl("[0-9]{3}$",alldata$mdz)
alt_positions=grepl("[a-zA-Z]",alldata$mdz)
sum(as.numeric(alt_positions))
alldata=alldata[alt_positions,]
#######Filtering for Non-E255I########
# note255i_positions=!grepl("G0A0G",alldata$mdz)
# sum(as.numeric(note255i_positions))
# alldata=alldata[note255i_positions,]
#######Dividing SNPs and MNVs########
##########################SNPS########################################
snps=alldata%>%mutate(n_nuc=str_count(mdz,"A|T|C|G"))%>%filter(n_nuc==1)
mnvs=alldata%>%mutate(n_nuc=str_count(mdz,"A|T|C|G"))%>%filter(n_nuc>=2)

snps=snps%>%rowwise()%>%mutate(alt_start_pos=pos+as.numeric(strsplit(mdz,"A|C|G|T")[[1]][1]),alt_end_pos=alt_start_pos+1)
snps=snps%>%
  # rowwise()%>%
  mutate(ref=abl1_search(alt_start_pos,alt_end_pos))

###filtering out mismatches that are at the start or end of the read (and hence of the mdz field)###
#mut_extr aka mutations at the extremities of the read
snps=snps%>%
  rowwise()%>%
  mutate(mut_extr=
           case_when(as.numeric(strsplit(mdz,"A|C|G|T")[[1]][1])%in%c(0,1)~TRUE,
                     as.numeric(strsplit(mdz,"A|C|G|T")[[1]][2])%in%c(0,1)~TRUE,
                     TRUE~FALSE))
snps=snps%>%filter(mut_extr%in%FALSE)


snps=snps%>%rowwise()%>%mutate(alt=substr(seq,alt_start_pos-pos+1+clipdistance,alt_start_pos-pos+1+clipdistance))
sum(as.numeric(grepl("N",snps$alt)))
snps=snps%>%filter(!alt%in%"N")


snps_sum=snps%>%group_by(alt_start_pos,ref,alt)%>%dplyr::summarize(ct=n())

# good=snps%>%filter(alt_start_pos%in%130873027)
# bad=snps%>%filter(alt_start_pos%in%130872175)
# 
# good_sum=snps_sum%>%filter(alt_start_pos%in%130873027)
# bad_sum=snps_sum%>%filter(alt_start_pos%in%130872175)
# sum(snps_sum$ct)
##########################MNVs########################################
#######Defining neighbor distance########
###For 2 nucleotide substitutions, 96% start at the start of the codon. The possibilites are for the substitutions to be side by side or be separated by a nucleotide. e.g. AAA could mutate to TTA or TAT. I need to know whether a TTA type substitution is happening or a TAT type substitution. Therefore I am defining a neighbordistance term

mnvs=mnvs%>%
  mutate(neighbordistance=case_when(
    n_nuc%in%2&&head(strsplit(mdz,"A|G|C|T")[[1]][-1],-1)%in%0~1,
    n_nuc%in%2&&head(strsplit(mdz,"A|G|C|T")[[1]][-1],-1)%in%1~2,
    TRUE~NaN))
# mnvs$minuslast="NA"
# mnvs=mnvs%>%rowwise()%>%mutate(minuslast=head(strsplit(mdz,"A|G|C|T"),-1))
#######Filtering MNVs for side-by-side MNVs######
#Has to be within hamming distance of 2,
#Basically looking for a 0 or 1 with no neighboring numbers
# 30A0C0C94
# 38C0A3
# strsplit("30A0C0C94","A|G|C|T")
# head(strsplit("38C0A3","A|G|C|T")[[1]][-1],-1)
# a=lapply()
# a=mnvs%>%rowwise()%>%filter(head(strsplit(mdz,"A|G|C|T")[[1]][-1],-1)%in%"0")
list=mnvs$mdz
# i=1
mnv_status_compiled=rep("NA",length(list))
for(i in 1:length(list)){
  mnv_status_i=!grepl("2|3|4|5|6|7|8|9|11",head(strsplit(list[i],"A|G|C|T")[[1]][-1],-1))
  # grepl("0|1",c("0","0"))
  # grepl("0|1",c("1","0"))
  # grepl("0|1",c("1","1"))
  # grepl("0|1",c("1","1"))
  # grepl("0|1",c("1","12"))
  # !grepl("2|3|4|5|6|7|8|9|11",c("1","12"))
  if(sum(as.numeric(mnv_status_i))>=1){
    mnv_status_compiled[i]=TRUE
    # mnv_status_compiled[i]=TRUE
  } else {
    mnv_status_compiled[i]=FALSE
  }
}
mnvs$mnv_status=mnv_status_compiled

# class(mnvs$mnv_status)
#Regexp for 
# sum(as.numeric(mnvs$mnv_status))
# sum(mnvs$mnv_status%in%TRUE)

    

#### For my sscs replicates for novogene lane 4 il3 indep 1, I observed 3301 variant reads with >2 varaints out of which 1790 reads had variants that were side-by-side mnvs (hamming distance of 0 or 1)
#### For my dcs replicates, I observed 3301 variant reads with >2 varaints out of which 1790 reads had variants that were side-by-side mnvs (hamming distance of 0 or 1)
#We know that this library enriches for mnvs with hamming distance of <2 (true positive). The hypothesis is that if I were sequencing at a threshold below the error rate for sscs, then there would be a lot of reads in which the real mutant was masked by multiple nucleotide variants (hamming distance >2). However, if we were good on the sscs error rate, the type of sequencing would not predict whether a read with two variants had 
#Testing to see if the sscs or dcs predicts the hamming distance.
# fisher.test(rbind(c(1790,1511),c(596,470)))
a=mnvs[mnvs$n_nuc==3,]
a=mnvs[mnvs$n_nuc==2,]
# a=mnvs[mnvs$mnv_status%in%F,]
(strsplit(a$mdz,"A|G|C|T")[[1]][1])

(strsplit(a$mdz,"A|G|C|T")[[1]][a$n_nuc[1]+1])
###Calculating which reads had one mutant at the end of the read. 10% of mnvs had mutant at end###
mnvs=mnvs%>%
  rowwise()%>%
  mutate(mut_extr=case_when(as.numeric(strsplit(mdz,"A|G|C|T")[[1]][1])%in%c(0,1)~T,
                          as.numeric(strsplit(mdz,"A|C|G|T")[[1]][n_nuc[1]+1])%in%c(0,1)~T,
                                          T~F))
###Calculating which reads had L298L in the read###
#L298L occurs 20 residues into exon 5. What I realized is that a lot of mnvs for L298L had two wrong calls: one wrong call because the split read started in exon 4 and the variant caller thought there were mismatches in exon 5 when in0fact it was looking at exon 4. Need to definitely take these into account.

# a=mnvs%>%filter(mnv_status%in%F,mut_extr%in%F)

###For mnvs with L298L or second mutant at the end of read, count the SNP only.###
####The majority of MNVs that we see are "bad" because the mutants don't occur right next to each other. For mnvs that have mutants that are far apart, there could be different things that are going on: 1) there are actually 2 mutants that are far apart (10% reads in saturation mutagenesis libraries), 2) one of the mutants is erroneous, would expect this for mutants called near the end of the read so will filter those out. 3) both mutants are technically mutants but only one of those is a true mutant. For our case, L298L was a true synonymous mutant present in our library hence every human ABL1 cDNA read near 298 should read this mutant. So I will figure out a way to NOT filter those out.###

####Filtering for good mnvs
mnvs=mnvs%>%filter(mnv_status%in%TRUE)


#Determining the mnv start and end position on the genome
#For end position, if it's a 2 nuclteotide mnv that looks like a 3 nucleoitde mnv, add +1
mnvs=mnvs%>%
  rowwise()%>%
  mutate(alt_start_pos=pos+as.numeric(strsplit(mdz,"A|G|C|T")[[1]][1]),
         alt_end_pos=case_when(n_nuc%in%2&&neighbordistance%in%2~alt_start_pos+n_nuc+1,
                               T~alt_start_pos+n_nuc))



#Figuring out if mnv spans two codons
#Nearest real codon of an mnv is the one it starts right after. E.g. if we see mnv starting at position 100, it's nearest neighbor would be codon 99-102
#If it is a 3 nucleotide variant, it must start at the same start position as the start codon
#If it is a 2 nucleotide variant, its distance from the real codon can be 0 or 1 


mnvs=mnvs%>%
  rowwise()%>%
  mutate(in_frame_mnv=
           case_when(alt_start_pos%in%abl_genomic_coordinates$start~TRUE,
                     T~FALSE))

#Turns out most of these mnvs start at the start of a codon! Thats great, means they're real mnvs
###Turns out the majority of 2 nucleotide substitutions (96%) start at the first nucleotide in the codon. That's interesting. 
# a=mnvs%>%filter(n_nuc==2&&in_frame_mnv==FALSE)
mnvs=mnvs%>%filter(in_frame_mnv%in%TRUE)
# sort(unique(mnvs$alt_start_pos))



#Looks like T315V comprises most of the unique triple nucleotide mnvs. 15 unique Mnvs in this read so far #This was in the iL3 indep case
# a=mnvs%>%filter(alt_start_pos==130862940)%>%mutate(codon=substr(seq,alt_start_pos-pos+1,alt_start_pos-pos+3))
# sort(unique(a$codon))

mnvs=mnvs%>%mutate(ref=abl1_search(alt_start_pos,alt_end_pos))

mnvs=mnvs%>%
  rowwise()%>%
  mutate(alt=case_when(neighbordistance%in%1~substr(seq,alt_start_pos-pos+1+clipdistance,alt_start_pos-pos+2+clipdistance),
    T~substr(seq,alt_start_pos-pos+1+clipdistance,alt_start_pos-pos+3+clipdistance)))



#Filtering out ALT codons that contain Ns
#Do same for snps when you're there
# mnvs=mnvs[!grepl("N",mnvs$alt),]
#Not needed anymore becuase I filter out Ns in the filtering steps at the start of the code
###
mnvs$id=paste(mnvs$alt_start_pos,mnvs$alt,sep="")
mnvs$id=paste(mnvs$alt_start_pos,mnvs$ref,sep="")

mnv_sum=mnvs%>%group_by(alt_start_pos,alt_end_pos,ref,alt)%>%dplyr::summarize(ct=n())
# sum(mnv_sum$ct)

##########Annotating SNPS###############
# a=snps_sum[c(1:50),]
# a=vep_fromdf(a)


snps_sum_ann=vep_fromdf(snps_sum)
snps_sum_ann$type="snp"
# snps_sum_ann=snps_ann
# a=snps_ann%>%filter(consequence_terms%in%"missense_variant")


snps_ann=merge(snps,snps_sum_ann,by=c("alt_start_pos","alt_end_pos","ref","alt"))

##########Annotating MNVs###############
mnv_sum_ann=vep_fromdf(mnv_sum)
mnv_sum_ann$type="mnv"
mnvs_ann=merge(mnvs,mnv_sum_ann,by=c("alt_start_pos","alt_end_pos","ref","alt"))
# a=vep_fromdf(a)


#####Merging MNVs and SNPS#####

snps_sum_ann$alt_end_pos=snps_sum_ann$alt_start_pos+1
snps_sum_ann_reformatted=snps_sum_ann%>%dplyr::select(type,
                                          alt_start_pos,
                                          alt_end_pos,
                                          ref,
                                          alt,
                                          ct,
                                          protein_start,
                                          protein_end,
                                          amino_acids,
                                          codons,
                                          impact,
                                          polyphen_prediction,
                                          consequence_terms)
# snps_sum_ann$protein_start

mnv_sum_ann_reformatted=mnv_sum_ann%>%dplyr::select(type,
                                          alt_start_pos,
                                          alt_end_pos,
                                          ref,
                                          alt,
                                          ct,
                                          protein_start,
                                          protein_end,
                                          amino_acids,
                                          codons,
                                          impact,
                                          polyphen_prediction,
                                          consequence_terms)

calls_sum_merged=rbind(snps_sum_ann_reformatted,mnv_sum_ann_reformatted)
# colnames(snps_ann)
# colnames(mnvs_ann)

# write.csv(calls_sum_merged,gsub(files[filecurr],pattern = ".tsv",replacement = "_calls.csv"))
calls_missense=calls_sum_merged%>%filter(consequence_terms%in%"missense_variant")
}

# snps_ann_reformatted=snps_ann_reformatted%>%dplyr::select(type,
#                                           alt_start_pos,
#                                           alt_end_pos,
#                                           ref,
#                                           alt,
#                                           ct,
#                                           protein_start,
#                                           protein_end,
#                                           amino_acids,
#                                           codons,
#                                           impact,
#                                           polyphen_prediction,
#                                           consequence_terms)
```

####Reading all of novogene lane 4's filtered data
```{r}
setwd("/Users/haiderinam/OneDrive - The Pennsylvania State University/RProjects/duplex_sequencing_screen/data/Dunovo/Novogene_lane4/GalaxyData/")
####################IL3 1#####################
il3indep_1_sscs=read.csv("il3indep1_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
il3indep_1_sscs=il3indep_1_sscs%>%filter(consequence_terms%in%"missense_variant")
il3indep_1_sscs=il3indep_1_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

il3indep_1_dcs=read.csv("il3indep1_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
il3indep_1_dcs=il3indep_1_dcs%>%filter(consequence_terms%in%"missense_variant")
il3indep_1_dcs=il3indep_1_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

####################IL3 2#####################
il3indep_2_sscs=read.csv("il3indep2_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
il3indep_2_sscs=il3indep_2_sscs%>%filter(consequence_terms%in%"missense_variant")
il3indep_2_sscs=il3indep_2_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

il3indep_2_dcs=read.csv("il3indep3_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
il3indep_2_dcs=il3indep_2_dcs%>%filter(consequence_terms%in%"missense_variant")
il3indep_2_dcs=il3indep_2_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

####################IL3 3#####################
il3indep_3_sscs=read.csv("il3indep3_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
il3indep_3_sscs=il3indep_3_sscs%>%filter(consequence_terms%in%"missense_variant")
il3indep_3_sscs=il3indep_3_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

il3indep_3_dcs=read.csv("il3indep3_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
il3indep_3_dcs=il3indep_3_dcs%>%filter(consequence_terms%in%"missense_variant")
il3indep_3_dcs=il3indep_3_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

####################IL3 4#####################
il3indep_4_sscs=read.csv("il3indep4_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
il3indep_4_sscs=il3indep_4_sscs%>%filter(consequence_terms%in%"missense_variant")
il3indep_4_sscs=il3indep_4_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

il3indep_4_dcs=read.csv("il3indep4_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
il3indep_4_dcs=il3indep_4_dcs%>%filter(consequence_terms%in%"missense_variant")
il3indep_4_dcs=il3indep_4_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

####################IL3 5#####################
il3indep_5_sscs=read.csv("il3indep5_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
il3indep_5_sscs=il3indep_5_sscs%>%filter(consequence_terms%in%"missense_variant")
il3indep_5_sscs=il3indep_5_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

il3indep_5_dcs=read.csv("il3indep5_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
il3indep_5_dcs=il3indep_5_dcs%>%filter(consequence_terms%in%"missense_variant")
il3indep_5_dcs=il3indep_5_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

setwd("/Users/haiderinam/OneDrive - The Pennsylvania State University/RProjects/duplex_sequencing_screen/data/Dunovo/Novogene_lane4/GalaxyData/")
a=read.csv("sorted1_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
# a=a%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))
# sum(as.numeric((a$consequence_terms%in%c("synonymous_variant"))))
####################Sorted 1 #####################
sorted_1_sscs=read.csv("sorted1_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_1_sscs=sorted_1_sscs%>%filter(consequence_terms%in%"missense_variant")
sorted_1_sscs=sorted_1_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

sorted_1_dcs=read.csv("sorted1_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_1_dcs=sorted_1_dcs%>%filter(consequence_terms%in%"missense_variant")
sorted_1_dcs=sorted_1_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

####################Sorted 2 #####################
sorted_2_sscs=read.csv("sorted2_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_2_sscs=sorted_2_sscs%>%filter(consequence_terms%in%"missense_variant")
sorted_2_sscs=sorted_2_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

sorted_2_dcs=read.csv("sorted2_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_2_dcs=sorted_2_dcs%>%filter(consequence_terms%in%"missense_variant")
sorted_2_dcs=sorted_2_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

####################Sorted 3 #####################
sorted_3_sscs=read.csv("sorted3_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_3_sscs=sorted_3_sscs%>%filter(consequence_terms%in%"missense_variant")
sorted_3_sscs=sorted_3_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

sorted_3_dcs=read.csv("sorted3_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_3_dcs=sorted_3_dcs%>%filter(consequence_terms%in%"missense_variant")
sorted_3_dcs=sorted_3_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

####################Sorted 4 #####################
sorted_4_sscs=read.csv("sorted4_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_4_sscs=sorted_4_sscs%>%filter(consequence_terms%in%"missense_variant")
sorted_4_sscs=sorted_4_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

sorted_4_dcs=read.csv("sorted4_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_4_dcs=sorted_4_dcs%>%filter(consequence_terms%in%"missense_variant")
sorted_4_dcs=sorted_4_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

####################Sorted 5 #####################
sorted_5_sscs=read.csv("sorted5_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_5_sscs=sorted_5_sscs%>%filter(consequence_terms%in%"missense_variant")
sorted_5_sscs=sorted_5_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

sorted_5_dcs=read.csv("sorted5_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_5_dcs=sorted_5_dcs%>%filter(consequence_terms%in%"missense_variant")
sorted_5_dcs=sorted_5_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

####################Sorted 6 #####################
sorted_6_sscs=read.csv("sorted6_sscs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_6_sscs=sorted_6_sscs%>%filter(consequence_terms%in%"missense_variant")
sorted_6_sscs=sorted_6_sscs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))

sorted_6_dcs=read.csv("sorted6_dcs_filtered_calls.csv",header = T,stringsAsFactors = F)[,-1]
sorted_6_dcs=sorted_6_dcs%>%filter(consequence_terms%in%"missense_variant")
sorted_6_dcs=sorted_6_dcs%>%rowwise()%>%mutate(amino_acids=gsub(x = amino_acids,pattern = "/",replacement = protein_start))


```
####SSCS encompasses all mutants found in DCS, at a 3-5x higher coverage as expected. This trend is seen in all sorted and il3 indep samples except il3 indep #2 for some reason
```{r}
length(unique(sorted_6_sscs$amino_acids))
length(unique(sorted_6_dcs$amino_acids))
sum(as.numeric(sort(unique(sorted_6_sscs$amino_acids))%in%sort(unique(sorted_6_dcs$amino_acids))))
a=il3indep_1_sscs%>%filter(amino_acids%in%sort(unique(il3indep_1_dcs$amino_acids)))
####Taking closer look at the 78 mutants seen in il3indep2 dcs but not in sscs
a=il3indep_2_dcs%>%filter(!amino_acids%in%sort(unique(il3indep_2_sscs$amino_acids)))
```
```{r}
###############Combining Samples#################
# il3indep_1_sscs$name="il3indep_1_sscs"
# il3indep_2_sscs$name="il3indep_2_sscs"
# il3indep_3_sscs$name="il3indep_3_sscs"
# il3indep_4_sscs$name="il3indep_4_sscs"
# il3indep_5_sscs$name="il3indep_5_sscs"
# il3indep_6_sscs$name="il3indep_6_sscs"
il3indep_all=rbind(il3indep_1_sscs,il3indep_2_sscs,il3indep_3_sscs,il3indep_4_sscs,il3indep_5_sscs)
il3indep_all=il3indep_all%>%dplyr::select(alt_start_pos,ref,alt,protein_start,amino_acids,ct)
il3indep_grouped=il3indep_all%>%
  dplyr::group_by(amino_acids,alt_start_pos,ref,alt,protein_start)%>%
  dplyr::summarize(ct=sum(ct),nsamples=n())%>%
  dplyr::select(alt_start_pos,protein_start,ref,alt,amino_acids,ct,nsamples)
il3indep_grouped$altamino=substr(il3indep_grouped$amino_acids,nchar(il3indep_grouped$amino_acids),nchar(il3indep_grouped$amino_acids))




# sorted_1_sscs$name="sorted_1_sscs"
# sorted_2_sscs$name="sorted_2_sscs"
# sorted_3_sscs$name="sorted_3_sscs"
# sorted_4_sscs$name="sorted_4_sscs"
# sorted_5_sscs$name="sorted_5_sscs"
# sorted_6_sscs$name="sorted_6_sscs"
sorted_all=rbind(sorted_1_sscs,sorted_2_sscs,sorted_3_sscs,sorted_4_sscs,sorted_5_sscs,sorted_6_sscs)
sorted_all=sorted_all%>%dplyr::select(alt_start_pos,ref,alt,protein_start,amino_acids,ct)
sorted_grouped=sorted_all%>%
  dplyr::group_by(amino_acids,alt_start_pos,ref,alt,protein_start)%>%
  dplyr::summarize(ct=sum(ct),nsamples=n())%>%
  dplyr::select(alt_start_pos,protein_start,ref,alt,amino_acids,ct,nsamples)
sorted_grouped$altamino=substr(sorted_grouped$amino_acids,nchar(sorted_grouped$amino_acids),nchar(sorted_grouped$amino_acids))
# a=sorted_il3_all%>%filter(enr%in%"Inf",ct_il3>=5)


####################Plotting##################

ggplot(il3indep_grouped%>%filter(!amino_acids%in%NA),aes(x=protein_start,y=altamino))+geom_tile(color="black",aes(fill=ct))+scale_x_continuous(expand = c(0,0),limits = c(50,650))+scale_y_discrete(expand = c(0,0))+theme(plot.title = element_text(hjust = 0.5))+theme(plot.title = element_text(hjust = 0.5),axis.text.x=element_text(angle=90,hjust=.5,vjust=.5),axis.text=element_text(face="bold",size="9",color="black"))

ggplot(sorted_grouped%>%filter(!amino_acids%in%NA),aes(x=protein_start,y=altamino))+geom_tile(color="black",aes(fill=ct))+scale_x_continuous(expand = c(0,0),limits = c(50,650))+scale_y_discrete(expand = c(0,0))+theme(plot.title = element_text(hjust = 0.5))+theme(plot.title = element_text(hjust = 0.5),axis.text.x=element_text(angle=90,hjust=.5,vjust=.5),axis.text=element_text(face="bold",size="9",color="black"))

```
After Calculating Enrichment Scores and looking at distributions, I'm not super sure about one mutant particularly dominating the pool without Imatinib (growing out significantly better for iL3 independence) because the mutants with the highest allele frequency for iL3 independence had a high allele frequency for the sorted samples to start off with. The highest enrichments in the il3 case were from a mutant that was not observed, to observed 18 times.
```{r}
#################Calculating enrichments################
#Plots showing that mutants that are seen in 1 sample only seldom have counts above 5
ggplot(sorted_grouped,aes(x=nsamples,y=ct))+geom_point()+scale_y_continuous(limits = c(1,500),trans="log10")
ggplot(il3indep_grouped,aes(x=nsamples,y=ct))+geom_point()+scale_y_continuous(limits = c(1,500),trans="log10")

ggplot(sorted_grouped,aes(protein_start))+stat_ecdf(geom="step")+scale_x_continuous(limits=c(0,700))+geom_vline(xintercept = c(244,492),linetype="dashed")
ggplot(il3indep_grouped,aes(protein_start))+stat_ecdf(geom="step")+scale_x_continuous(limits=c(0,700))+geom_vline(xintercept = c(244,492),linetype="dashed")
# sum(sorted_all$ct[c(1:50)])


sorted_il3_all=merge(sorted_grouped,il3indep_grouped,by=c("alt_start_pos","ref","alt","protein_start","amino_acids","altamino"),all = T)

sorted_il3_all=sorted_il3_all%>%
  dplyr::select(-c(nsamples.x,nsamples.y),ct_sorted=ct.x,ct_il3=ct.y)
sorted_il3_all$ct_sorted[sorted_il3_all$ct_sorted%in%NA]=0
sorted_il3_all$ct_il3[sorted_il3_all$ct_il3%in%NA]=0
sorted_il3_all$enr=sorted_il3_all$ct_il3/sorted_il3_all$ct_sorted
sorted_il3_all$logenr=log10(sorted_il3_all$ct_il3/sorted_il3_all$ct_sorted)
sorted_il3_all=sorted_il3_all%>%
  mutate(status=case_when(enr%in%Inf~"il3only",
                          enr%in%0~"sortedonly",
                          T~"both"))
sum(as.numeric(sorted_il3_all$status%in%"sortedonly"))
sum(as.numeric(sorted_il3_all$status%in%"il3only"))
sum(as.numeric(sorted_il3_all$status%in%"both"))

ggplot(sorted_il3_all%>%filter(status%in%"both"),aes(x=enr))+geom_histogram(bins=50)+geom_vline(xintercept=1,linetype="dashed")

ggplot(sorted_il3_all%>%filter(!amino_acids%in%NA),aes(x=protein_start,y=altamino))+geom_tile(aes(fill=enr))+scale_x_continuous(expand = c(0,0),limits = c(50,650))+scale_y_discrete(expand = c(0,0))+theme(plot.title = element_text(hjust = 0.5))+theme(plot.title = element_text(hjust = 0.5),axis.text.x=element_text(angle=90,hjust=.5,vjust=.5),axis.text=element_text(face="bold",size="9",color="black"))+scale_fill_gradient2(low ="red" ,mid ="white",midpoint=1,high ="blue",name="Score")

ggplot(sorted_il3_all%>%filter(!amino_acids%in%NA),aes(x=protein_start,y=altamino))+geom_tile(aes(fill=logenr))+scale_x_continuous(expand = c(0,0),limits = c(50,650))+scale_y_discrete(expand = c(0,0))+theme(plot.title = element_text(hjust = 0.5))+theme(plot.title = element_text(hjust = 0.5),axis.text.x=element_text(angle=90,hjust=.5,vjust=.5),axis.text=element_text(face="bold",size="9",color="black"))+scale_fill_gradient2(low ="red" ,mid ="white",midpoint=0,high ="blue",name="Score")
```
```{r}
plotly=ggplot(sorted_all,aes(x=ct))+geom_histogram()+
  scale_x_continuous(limits=c(0,500))+
  scale_y_continuous(limits=c(0,200))
ggplotly(plotly)
sum(as.numeric(nchar(sorted_all$alt)==1))
write.csv(sorted_all,"ranomics_december_library.csv")

```


####Total Variants:
1685 for sorted
1635 for iL3
1145 of the iL3 indep variants were seen in sorted

Proportion mnvs: 
11% sorted sscs 1, 6.6% il3 sscs 1
25% sorted dcs 1, 17% il3 dcs 1

Proportion of reads that have a coverage of only 1:
45-50% for iL3 and 45-50% for sorted

Question: what about drug treated replicates? Do you see more of those?


```{r}
sorted_all_dcs=rbind(sorted_1_dcs,
                 sorted_2_dcs,
                 sorted_3_dcs,
                 sorted_4_dcs,
                 sorted_5_dcs,
                 sorted_6_dcs)

length(unique(sorted_all_dcs$amino_acids))

sorted_all_sscs=rbind(sorted_1_sscs,
                 sorted_2_sscs,
                 sorted_3_sscs,
                 sorted_4_sscs,
                 sorted_5_sscs,
                 sorted_6_sscs)

length(unique(sorted_all_sscs$amino_acids))

il3_all_dcs=rbind(il3indep_1_dcs,
                   il3indep_2_dcs,
                   il3indep_3_dcs,
                   il3indep_4_dcs,
                   il3indep_5_dcs)

length(unique(il3_all_dcs$amino_acids))

il3_all_sscs=rbind(il3indep_1_sscs,
                   il3indep_2_sscs,
                   il3indep_3_sscs,
                   il3indep_4_sscs,
                   il3indep_5_sscs)

length(unique(il3_all_sscs$amino_acids))

sum(as.numeric(unique(il3_all_sscs$amino_acids)%in%unique(sorted_all_sscs$amino_acids)))

sum(as.numeric(unique(sorted_all_sscs$amino_acids)%in%unique(il3_all_sscs$amino_acids)))

###Looking at the proportion of mnvs in dataset for sorted, il3, sscs, and dcs
a=sorted_3_sscs%>%filter(type%in%"mnv")
nrow(a[,1])/nrow(sorted_3_sscs[,1])*100
a=sorted_3_dcs%>%filter(type%in%"mnv")
nrow(a[,1])/nrow(sorted_3_dcs[,1])*100
a=il3indep_3_sscs%>%filter(type%in%"mnv")
nrow(a[,1])/nrow(il3indep_3_sscs[,1])*100
a=il3indep_3_dcs%>%filter(type%in%"mnv")
nrow(a[,1])/nrow(il3indep_3_dcs[,1])*100
```
Proportion of variants outside of the asked for window:
Sorted before resi 220: 12%
Sorted after resi 500: 23%
iL3 before resi 220: 14%
iL3 after resi 500: 26%

Important: Make CDF of mutations found across location of kinase

```{r}
length(unique(sorted_all_sscs$amino_acids))
a=sorted_all_sscs%>%rowwise()%>%filter(!protein_start>=220)
length(unique(a$amino_acids))
a=sorted_all_sscs%>%rowwise()%>%filter(!protein_start<=530)
length(unique(a$amino_acids))
```

How many more variants do we see as we add samples:
Original: 663
2 Samples: 905
3 Samples: 1008
4 Samples: 1164
5 Samples: 1413
6 Samples: 1683 sample 6 of course has a v high coverage
```{r}
a=rbind(sorted_1_sscs,sorted_2_sscs)
length(unique(a$amino_acids))
a=rbind(a,sorted_3_sscs)
length(unique(a$amino_acids))
a=rbind(a,sorted_4_sscs)
length(unique(a$amino_acids))
a=rbind(a,sorted_5_sscs)
length(unique(a$amino_acids))
a=rbind(a,sorted_6_sscs)
length(unique(a$amino_acids))
```

Important: Analyze how much overlap there is between samples if you only look at mutants that have a coverage of 1? Are mutants completely unique? How much overlap? How much overlap would you expect with improved coverage vs with error only? Maybe cross validate with subsampling

Important: look into why 1/3rd of all mutants are synonymous substitutions? Why are there so many stop codons? Why are there so many 2 nucleotide variants with hamming distance>2? Are these all occuring because of 1) SSCS error rate that you wouldn't see in duplex sequencing? 2) Mouse variants that would tend ot be synonymous, 3) Bad libarry with a lot of 2 nucleotide variants synonymous variants, and stop codons and variants that are outsdie the mutagenesis window
