---
title: "Lane_11_CDS_Translating_Script"
author: "Haider Inam"
date: '2022-06-30'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath(".."))
library(dplyr)
library(reshape2)
library(httr)
library(jsonlite)
library(xml2)
library(stringr)
library(dplyr)
library(ggplot2)
library(plotly)
```

```{r}
#Code to verify that our CDS to Hg38 mapping coordinates match Ensembl's definitions:
coordinates=read.csv("Novogene_lane11/abl1_full_coordinates.csv",header = T)
coordinates$match=F

for(i in seq(1:length(coordinates$chr_start))){
  if(transcript_verifier(coordinates$NM_005157.6_CDS_pos[i])==coordinates$chr_start[i])
  coordinates$match[i]=T
}

#Transcript verifier function:
library(httr)
library(jsonlite)
library(xml2)
transcript_verifier=function(pos){
  #Source: https://rest.ensembl.org/documentation/info/assembly_cds
  server <- "https://rest.ensembl.org"
  # ext <- "/map/cds/ENST00000318560/907..909?"
  ext=paste("/map/cds/ENST00000318560/",pos,"..",pos,"?",sep = "")
 
  r <- GET(paste(server, ext, sep = ""), content_type("application/json"))
 
  stop_for_status(r)
 
  rest_results=(fromJSON(toJSON(content(r))))
  tcrpt=rest_results$mappings  
  tcrpt$start[[1]][1]
}
```

####6.30.2022####
####Twinstrand is using CCDS35166.1 for BCRABL and NM_005228.5 for EGFR####
This is aimed to be a simple script, and eventually a function, that converts the transcript positions into genomic coordinates, and then genomic coordinates, ref, and alt positions into amino acid consequences using ENSEMBL
```{r}
mutfile=data.frame(read.table("sample2/Sample_2.1.consensus.variant-calls.mut",header = T,stringsAsFactors = F,check.names = F,fill=T))
mutfile=mutfile%>%mutate(start=start+1,end=end+1)
mutfile=mutfile%>%filter(!variation_type%in%"indel")
mutfile=mutfile%>%dplyr::select(start,end,variation_type,ref,alt,alt_depth,depth,no_calls,subtype,context)
genomic_coordinates=read.csv("abl1_full_coordinates.csv")
mutfile=mutfile%>%filter(variation_type%in%c("snv","mnv"))
mutfile=mutfile%>%filter(end-start<4)
# region="130872215:130872215/T"
# region="130862955:130862958/AAC"
# region="130862936:130862940/TGCT"
# vep_fromquery(region)
# b=vep_fromquery(region)

mutfile=merge(mutfile,genomic_coordinates,by.x  = "start",by.y="NM_005157.6_CDS_pos")
mutfile$alt_start_hg38=mutfile$chr_start
mutfile$alt_end_hg38=mutfile$chr_start+(mutfile$end-mutfile$start)





# mutfile_small=mutfile[c(1:100),]
# mutfile_translated=vep_fromdf(mutfile)
mutfile_splice=mutfile[c(1020:1120),]
sample2_splice=vep_fromdf(mutfile_splice)

# write.csv(mutfile_translated,"sample2_translated.csv")


# cds=read.csv("NM_005157.6_CDS.txt",header = F)
substr(cds,736,738)
substr(cds,739,741)
```

```{r}
library(httr)
library(jsonlite)
library(xml2)
 
server <- "https://rest.ensembl.org"
ext <- "/sequence/region/human/9:130872213..130872215:1?"
 
r <- GET(paste(server, ext, sep = ""), content_type("text/plain"))
 
stop_for_status(r)
 
 
print(content(r))

```


```{r}
sample2_data=mutfile_translated

sample2_data=sample2_data%>%dplyr::select(!chr_start,)
strsplit("E/L","/")[[1]]
sample2_data=sample2_data%>%
  filter(consequence_terms%in%"missense_variant")%>%
  rowwise()%>%
  mutate(ref_aa=strsplit(amino_acids,"/")[[1]][1],
         alt_aa=strsplit(amino_acids,"/")[[1]][2])

#How many amino acids per position?
#What is the distribution of the number of reads per amino acid?
#Make map of mutant covered by allele position

#How well do the Dunovo counts agree with the twinstrand counts
#How well do the DCS counts agree with the SSCS counts?
#How many more mutants do we see with SSCS
sample2_sum=sample2_data%>%group_by(protein_start)%>%summarize(num_aa=n())

ggplot(sample2_sum%>%filter(protein_start>241,protein_start<350),aes(x=protein_start,y=num_aa))+geom_col()
plotly=ggplot(sample2_data,aes(x=alt_depth))+geom_histogram(bins=100)+scale_x_continuous(limits = c(0,95))
ggplotly(plotly)

```

