<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Haider Inam" />

<meta name="date" content="2023-08-30" />

<title>ABL_kinasefunction_data_parser</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">duplex_sequencing_screen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/haiderinam/duplex_sequencing_screen">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ABL_kinasefunction_data_parser</h1>
<h4 class="author">Haider Inam</h4>
<h4 class="date">2023-08-30</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>duplex_sequencing_screen/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200402code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200402)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200402code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200402)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhaiderinamduplexsequencingscreentreef9eea377a2eb905c9e7abeb5515e61a733c72241targetblankf9eea37a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/tree/f9eea377a2eb905c9e7abeb5515e61a733c72241" target="_blank">f9eea37</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomhaiderinamduplexsequencingscreentreef9eea377a2eb905c9e7abeb5515e61a733c72241targetblankf9eea37a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/haiderinam/duplex_sequencing_screen/tree/f9eea377a2eb905c9e7abeb5515e61a733c72241" target="_blank">f9eea37</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/Consensus_Data/.Rhistory
    Ignored:    data/Consensus_Data/Novogene_lane11/sample1/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample1/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample1/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample1/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample1/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample1/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample2/archive/sscs_aligned_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample2/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample2/duplex/variant_caller_outputs/archive_ensembl/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample2/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample2/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample2/sscs/variant_caller_outputs/archive_ensembl/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample2/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample3/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample3/duplex/variant_caller_outputs/archive_ensembl/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample3/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample3/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample3/sscs/variant_caller_outputs/archive ensembl/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample3/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample4/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample4/duplex/variant_caller_outputs/archive ensembl/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample4/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample4/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample4/sscs/variant_caller_outputs/archive ensembl/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample4/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample5/variant_caller_outputs/sscs_L858R_aligned_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample5/variant_caller_outputs/sscs_L858R_aligned_filtered_sample5.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample6/archive/sscs_aligned_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample6/sscs_L858R_aligned_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample6/variant_caller_outputs/variants_ann_sample6.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample7/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample7/sscs/variant_caller_outputs/archive ensembl/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane11/sample7/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample1/low_sscscounts/sscs_aligned_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample1/low_sscscounts/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample1/sscs_aligned_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample1/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample3/sscs_combined_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample3/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample5/sscs_combined_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample5/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample7/sscs_combined_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample7/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample9/sscs_combined_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane12/sample9/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample1/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample1/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample1/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample1/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample10/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample10/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample10/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample10/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample10/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample10/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample11/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample11/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample11/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample11/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample11/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample11/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample12/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample12/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample12/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample12/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample12/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample12/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample2/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample2/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample3/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample3/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample4/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample4/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample5/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample5/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample6/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample6/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample7/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample7/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample7/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample7/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample7/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample7/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample8/
    Ignored:    data/Consensus_Data/Novogene_lane13/sample9/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample9/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample9/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample9/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample9/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane13/sample9/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample10_combined/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample10_combined/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample10_combined/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample10_combined/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample10_combined/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample10_combined/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample11/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample11/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample11/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample11/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample11/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample11/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample12/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample12/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample12/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample12/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample12/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample12/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample13/
    Ignored:    data/Consensus_Data/Novogene_lane14/sample14_combined/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample14_combined/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample14_combined/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample14_combined/sscs.filt_1.fa.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample14_combined/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample14_combined/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample14_combined/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample14b/
    Ignored:    data/Consensus_Data/Novogene_lane14/sample15/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample15/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample15/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample15/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample15/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample15/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample16/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample16/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample16/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample16/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample16/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample16/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample17/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample17/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample17/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample17/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample17/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample17/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample18/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample18/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample18/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample18/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample18/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample18/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample1_combined/
    Ignored:    data/Consensus_Data/Novogene_lane14/sample2_combined/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample2_combined/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample3/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample3/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample4/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample4/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample5/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample5/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample6/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample6/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample7/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample7/variant_caller_outputs/duplex/
    Ignored:    data/Consensus_Data/Novogene_lane14/sample7/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample8/
    Ignored:    data/Consensus_Data/Novogene_lane14/sample9/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample9/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample9/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample9/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample9/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane14/sample9/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/Novogene_lane2/
    Ignored:    data/Consensus_Data/Novogene_lane3/
    Ignored:    data/Consensus_Data/Novogene_lane4/
    Ignored:    data/Consensus_Data/Novogene_lane5/
    Ignored:    data/Consensus_Data/Novogene_lane6/
    Ignored:    data/Consensus_Data/Novogene_lane7/
    Ignored:    data/Consensus_Data/Ranomics_Pooled/
    Ignored:    data/Consensus_Data/archive/
    Ignored:    data/Consensus_Data/novogene_lane15/sample_1/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_1/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_1/firstrun(lowsequencing)/
    Ignored:    data/Consensus_Data/novogene_lane15/sample_1/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_2/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_2/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_2/firstrun(lowsequencing)/sscs/
    Ignored:    data/Consensus_Data/novogene_lane15/sample_2/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_2/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/firstrun(lowsequencing)/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/firstrun(lowsequencing)/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/firstrun(lowsequencing)/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/firstrun(lowsequencing)/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/ngs/Sample3_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/ngs/sample3a(firsthalf)/Sample3_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/ngs/sample3a(firsthalf)/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/ngs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_3/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_4/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_4/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_4/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_4/firstrun(lowsequencing)/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_4/firstrun(lowsequencing)/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_4/firstrun(lowsequencing)/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_4/firstrun(lowsequencing)/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_4/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_4/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_4/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/firstrun(lowsequencing)/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/firstrun(lowsequencing)/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/firstrun(lowsequencing)/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/firstrun(lowsequencing)/sscs/variant_caller_outputs/.empty/
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/firstrun(lowsequencing)/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_5/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_6/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_6/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_6/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_6/firstrun(lowsequencing)/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_6/firstrun(lowsequencing)/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_6/firstrun(lowsequencing)/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_6/firstrun(lowsequencing)/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_6/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_6/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_6/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_7/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_7/duplex/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_7/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_7/firstrun(lowsequencing)/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_7/firstrun(lowsequencing)/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_7/firstrun(lowsequencing)/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_7/firstrun(lowsequencing)/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_7/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_7/sscs/variant_caller_outputs/archive/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane15/sample_7/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample10/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample10/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample10/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample10/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample11/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample11/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample11/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample11/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample12/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample12/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample12/sscs/
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample13/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample13/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample13/sscs/
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample14/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample14/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample14/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample14/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample1_combined/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample1_combined/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample1_combined/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample1_combined/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample2/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample2/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample2/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample3/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample3/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample3/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample3/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample4/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample4/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample4/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample4/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample5/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample5/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample5/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample5/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample6/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample6/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample6/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample6/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample7/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample7/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample7/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample7/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample8/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample8/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample8/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample8/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample9/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample9/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample9/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/Sample9/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16a/duplex/
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample10/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample10/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample10/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample10/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample11/
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample15/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample15/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample15/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample15/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample1_combined/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample1_combined/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample1_combined/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample1_combined/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample2/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample2/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample2/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample2/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample3/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample3/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample3/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample3/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample4/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample4/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample4/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample4/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample5/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample5/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample5/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample5/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample6/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample6/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample6/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample6/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample7_combined/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample7_combined/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample7_combined/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample7_combined/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample8_combined/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample8_combined/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample8_combined/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample8_combined/sscs/variant_caller_outputs/archive/
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample8_combined/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample9/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample9/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample9/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane16b/Sample9/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample10/duplex/
    Ignored:    data/Consensus_Data/novogene_lane17/sample10/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample10/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample11/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample11/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample11/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample11/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample1_combined/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample1_combined/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample1_combined/low_depth/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample1_combined/low_depth/duplex/low_depth/
    Ignored:    data/Consensus_Data/novogene_lane17/sample1_combined/low_depth/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample1_combined/low_depth/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample1_combined/low_depth/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample1_combined/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample1_combined/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample2/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample2/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample2/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample2/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample3/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample3/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample3/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample3/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample4/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample4/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample4/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample4/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample5/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample5/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample5/low_seq_depth/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample5/low_seq_depth/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample5/low_seq_depth/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample5/low_seq_depth/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample5/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample5/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample6/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample6/low_seq_depths/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample6/low_seq_depths/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample6/low_seq_depths/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample6/low_seq_depths/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample6/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample6/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample7/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample7/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample7/low_seq_depths/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample7/low_seq_depths/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample7/low_seq_depths/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample7/low_seq_depths/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample7/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample7/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample8/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample8/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample8/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample9/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample9/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample9/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17/sample9/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17b/Sample1 copy 2/
    Ignored:    data/Consensus_Data/novogene_lane17b/Sample1 copy 3/
    Ignored:    data/Consensus_Data/novogene_lane17b/Sample1/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17b/Sample1/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17b/Sample1/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17b/Sample1/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17b/Sample2/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane17b/Sample2/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane17b/Sample2/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/l298l/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/nol298l/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample1/sscs/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/l298l/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/ngs/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/ngs/Sample_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/ngs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/nol298l/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample10/sscs/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/l298l/sscs/variant_caller_outputs/slow_variantcaller_outputs_forcomparison/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/ngs/Sample_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/ngs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample11/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample12/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample12/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample12/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample12/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample12/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample12/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample12/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample12/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample13/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample13/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample13/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample13/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample13/ngs/Sample_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample13/ngs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample13/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample13/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample13/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample13/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample14/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/sample14/l298l/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/sample14/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample14/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample14/ngs/Sample_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample14/ngs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample14/nol298l/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/sample14/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample14/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample15/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample15/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample15/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample15/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample15/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample15/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample15/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample15/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample16/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample16/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample16/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample16/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample16/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample16/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample16/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample16/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample17/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample17/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample17/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample17/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample17/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample17/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample17/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample17/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/l298l/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/nol298l/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample18/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/l298l/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/l298l/sscs/variant_caller_outputs/fastercaller/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/nol298l/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample2/sscs/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/l298l/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/nol298l/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample3/sscs/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane18/sample4/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample4/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample4/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample4/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample4/ngs/Sample_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample4/ngs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample4/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample4/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample4/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample4/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample5/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample5/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample5/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample5/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample5/ngs/Sample_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample5/ngs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample5/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample5/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample5/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample5/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample6/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample6/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample6/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample6/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample6/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample6/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample6/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample6/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample7/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample7/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample7/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample7/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample7/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample7/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample7/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample7/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample8/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample8/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample8/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample8/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample8/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample8/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample8/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample8/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample9/l298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample9/l298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample9/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample9/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample9/nol298l/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample9/nol298l/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample9/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/sample9/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample3/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample3/l298l/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample3/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample3/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample3/nol298l/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample3/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample3/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample3/sscs/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample5/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample5/l298l/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample5/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample5/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample5/nol298l/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample5/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample5/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample5/sscs/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample6/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample6/l298l/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample6/l298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample6/l298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample6/nol298l/duplex/
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample6/nol298l/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample6/nol298l/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane18/tlane18a_sample6/sscs/
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample1/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample1/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample1/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample1/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample10/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample10/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample10/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample10/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample11/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample11/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample11/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample11/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample12/duplex/
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample12/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample12/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample2/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample2/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample2/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample2/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample3/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample3/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample3/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample3/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample4/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample4/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample4/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample4/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample5/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample5/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample5/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample5/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample6/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample6/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample6/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample6/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample7/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample7/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample7/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample7/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample8/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample8/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample8/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample8/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample9/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample9/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample9/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19a_Sample9/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample1/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample1/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample1/sscs/
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample10/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample10/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample10/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample10/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample11/duplex/
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample11/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample11/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample12/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample12/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample12/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample12/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample13/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample13/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample13/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample13/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample14/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample14/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample14/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample14/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample15/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample15/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample15/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample15/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample16/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample16/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample16/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample16/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample2/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample2/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample2/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample2/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample3/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample3/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample3/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample3/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample4/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample4/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample4/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample4/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample5/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample5/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample5/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample5/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample6/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample6/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample6/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample6/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample7/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample7/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample7/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample7/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample8/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample8/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample8/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample8/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample9/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample9/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample9/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19/Ln19b_Sample9/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample10/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample10/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample10/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample10/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample11/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample11/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample11/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample11/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample2/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample2/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample2/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample2/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample3/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample3/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample3/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample3/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample4/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample4/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample4/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample4/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample5/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample5/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample5/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample5/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample6/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample6/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample6/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample6/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample7/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample7/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample7/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample7/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample8/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample8/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample8/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample8/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample9/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample9/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample9/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19a_Sample9/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19b_Sample9/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19b_Sample9/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19b_Sample9/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane19_20/Ln19b_Sample9/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/.DS_Store
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample1/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample1/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample1/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample1/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample2/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample2/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample2/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample2/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample3/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample3/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample3/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample3/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample4/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample4/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample4/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample4/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample5/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample5/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample5/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample5/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample6/duplex/
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample6/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample6/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample7/
    Ignored:    data/Consensus_Data/novogene_lane22/Lane22_Sample8/
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample13/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample13/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample13/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample13/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample14/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample14/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample14/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample14/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample15/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample15/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample15/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample15/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample16/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample16/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample16/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample16/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample17/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample17/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample17/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample17/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample18/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample18/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample18/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample18/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample19/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample19/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample19/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample19/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample20/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample20/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample20/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample20/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample21/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample21/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample21/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample21/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample22/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample22/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample22/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample22/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample23/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample23/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample23/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample23/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample24/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample24/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample24/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample24/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample25/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample25/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample25/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample25/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample26/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample26/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample26/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample26/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample27/duplex/duplex_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample27/duplex/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample27/sscs/sscs_sorted_filtered.tsv.gz
    Ignored:    data/Consensus_Data/novogene_lane23/Lane23_Sample27/sscs/variant_caller_outputs/variants_ann.csv.gz
    Ignored:    data/Consensus_Data/sscs_dcs_comparisons/
    Ignored:    output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/NA/
    Ignored:    output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/baf3_IL3_low_rep1vsrep2/
    Ignored:    output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/baf3_Imat_Lowvsk562_Imat_Medium/
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/il3_indep_1.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/il3_indep_1.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/il3_indep_2.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/il3_indep_2.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/il3_indep_3.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/il3_indep_3.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/sorted_1.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/sorted_1.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/sorted_2.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/sorted_2.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/sorted_3.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane3/sorted_3.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/il3_indep_1.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/il3_indep_1.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/il3_indep_2.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/il3_indep_2.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/il3_indep_3.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/il3_indep_3.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/il3_indep_4.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/il3_indep_4.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/il3_indep_5.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/il3_indep_5.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_1.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_1.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_2.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_2.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_3.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_3.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_4.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_4.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_5.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_5.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_6.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane4/sorted_6.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_High_D2.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_High_D2.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_High_D4.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_High_D4.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_Low_D2.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_Low_D2.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_Low_D4.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_Low_D4.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_Medium_D2.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_Medium_D2.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_Medium_D4.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP4_Im_Medium_D4.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_High_D2.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_High_D2.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_High_D4.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_High_D4.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_Low_D2.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_Low_D2.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_Low_D4.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_Low_D4.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_Medium_D2.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_Medium_D2.1.consensus.variant-calls.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_Medium_D4.1.consensus.variant-calls.genome.vcf.gz
    Ignored:    output/Twinstrand/ABL1AppOutput/Novogene_Lane5/RP5_Im_Medium_D4.1.consensus.variant-calls.vcf.gz

Untracked files:
    Untracked:  output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/baf3_IL3_rep1vsrep2_ft/screen_comparison_baf3_IL3_low_rep1vsrep2ft.csv

Unstaged changes:
    Modified:   .DS_Store
    Modified:   analysis/ABL_kinasefunction_data_parser.Rmd
    Modified:   code/compare_screens.R
    Modified:   data/Consensus_Data/novogene_lane18/tlane18a_sample6/.DS_Store
    Modified:   output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/.DS_Store
    Modified:   output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/ABL_kinasefunction_data_parser.Rmd</code>) and HTML
(<code>docs/ABL_kinasefunction_data_parser.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/f9eea377a2eb905c9e7abeb5515e61a733c72241/analysis/ABL_kinasefunction_data_parser.Rmd" target="_blank">f9eea37</a>
</td>
<td>
haiderinam
</td>
<td>
2023-08-30
</td>
<td>
2023 Updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>#Cleanup code for plotting
source(&quot;code/plotting/cleanup.R&quot;)</code></pre>
<p>#8.30.23 #The following code was taken from
ABL_SM_CRISPR_Cut_Analyses.Rmd #Essentially all this code does is that
it takes two BaF3 ABL Region 1 D0 samples and two D6 samples, and
creates a dataframe with enrichment scores. #Ivan wanted counts from
separate D0 backgrounds, and he wanted every mutant, regardless of
whether it was seen in both baseline samples (when I do analyses, I take
out any mutant that is not seen in both baseline samples). Let’s call
these “unseen mutants” #This code does not include unseen mutants by
default. To make sure that unseen mutants are seen, uncomment out the
Ivan section in compare_screens.R. Once you uncomment it, then run the
code below #One more note: this is Region 1 (Amino acids 242-322) data
from Lane 18 only. In the future you can include samples from other
lanes.</p>
<pre class="r"><code>source(&quot;code/compare_screens.R&quot;)

source(&quot;code/plotting/cleanup.R&quot;)
source(&quot;code/plotting/heatmap_plotting_function.R&quot;)
# rm(list=ls())

comparisons=read.csv(&quot;data/Consensus_Data/novogene_lane18/TwistRegion1Screen_Comparisons_Todo.csv&quot;)
# comparisons=comparisons%&gt;%filter(Completed%in%&quot;FALSE&quot;)

# for(i in 1:nrow(comparisons)){
  i=10
  dirname=comparisons$dirname[i]
  pathname=paste(&quot;output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/&quot;,dirname,sep = &quot;&quot;)
  # Create directory if it doesn&#39;t already exist
  if (!file.exists(pathname)){
    dir.create(pathname)
} 
before_screen1_identifier=unlist(strsplit(comparisons$before_screen1_identifier[i],&quot;,&quot;))
after_screen1_identifier=unlist(strsplit(comparisons$after_screen1_identifier[i],&quot;,&quot;))
before_screen2_identifier=unlist(strsplit(comparisons$before_screen2_identifier[i],&quot;,&quot;))
before_screen2_identifier=&quot;BTRID0_FT&quot;
after_screen2_identifier=unlist(strsplit(comparisons$after_screen2_identifier[i],&quot;,&quot;))
  # length(after_screen1_identifier)
# screen_compare_means=compare_screens(comparisons$before_screen1_identifier[i],
#                                      comparisons$after_screen1_identifier[i],
#                                      comparisons$before_screen2_identifier[i],
#                                      comparisons$after_screen2_identifier[i])
screen_compare_means=compare_screens(before_screen1_identifier,
                                     after_screen1_identifier,
                                     before_screen2_identifier,
                                     after_screen2_identifier)

screen_compare_means_forexport=apply(screen_compare_means,2,as.character)
# dirname
# write.csv(screen_compare_means_forexport,file=&quot;output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/baf3_IL3_rep1vsrep2_ft/screen_comparison_baf3_IL3_low_rep1vsrep2ft.csv&quot;)

# Plot 1. What does the heatmap look like from the average of the net growth rate?
heatmap_plotting_function(screen_compare_means,242,321,fill_variable = &quot;netgr_obs_mean&quot;,fill_name = &quot;Net growth rate&quot;)</code></pre>
<p><img src="figure/ABL_kinasefunction_data_parser.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste(&quot;output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/&quot;,dirname,&quot;/plot1_heatmap.pdf&quot;,sep=&quot;&quot;),width=10,height=6,units=&quot;in&quot;,useDingbats=F)
# screen_compare_means2=screen_compare_means%&gt;%filter(alt_codon%in%twist$Codon)

# Plot 2a: What do the correlations look like for net growth rate (show mutants in text)?

ggplot(screen_compare_means,aes(x=netgr_obs_screen1,y=netgr_obs_screen2,color=resmuts,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+stat_cor(method = &quot;pearson&quot;)+labs(color=&quot;Known\nResistant\nMutant&quot;)+scale_x_continuous(&quot;Net growth rate screen 1&quot;)+scale_y_continuous(&quot;Net growth rate screen 2&quot;)</code></pre>
<p><img src="figure/ABL_kinasefunction_data_parser.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste(&quot;output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/&quot;,dirname,&quot;/plot2a_Netgrowthrate_correlations_text.pdf&quot;,sep=&quot;&quot;),width=6,height=4,units=&quot;in&quot;,useDingbats=F)

# Plot 2b: What do the correlations look like for enrichment scores (show mutants in points)?
ggplot(screen_compare_means,aes(x=netgr_obs_screen1,y=netgr_obs_screen2,label=species))+geom_point(color=&quot;black&quot;,shape=21,size=2,aes(fill=resmuts))+geom_abline()+cleanup+stat_cor(method = &quot;pearson&quot;)+labs(fill=&quot;Known\nResistant\nMutant&quot;)+scale_x_continuous(&quot;Net growth rate screen 1&quot;)+scale_y_continuous(&quot;Net growth rate screen 2&quot;)</code></pre>
<p><img src="figure/ABL_kinasefunction_data_parser.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste(&quot;output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/&quot;,dirname,&quot;/plot2b_Netgrowthrate_correlations_points.pdf&quot;,sep=&quot;&quot;),width=6,height=4,units=&quot;in&quot;,useDingbats=F)
# Plot 2c: What do the correlations look like for enrichment scores (show mutants in text)?
ggplot(screen_compare_means,aes(x=score_screen1,y=score_screen2,color=resmuts,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+ stat_cor(method = &quot;pearson&quot;)+labs(color=&quot;Known\nResistant\nMutant&quot;)+scale_x_continuous(&quot;Enrichment score screen 1&quot;)+scale_y_continuous(&quot;Enrichment score screen 2&quot;)</code></pre>
<p><img src="figure/ABL_kinasefunction_data_parser.Rmd/unnamed-chunk-2-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste(&quot;output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/&quot;,dirname,&quot;/plot2c_Enrichmentscores_correlations_text.pdf&quot;,sep=&quot;&quot;),width=6,height=4,units=&quot;in&quot;,useDingbats=F)

# Plot 3a: Plots: what are the overall net growth rate distributions?
ggplot(screen_compare_means,aes(x=netgr_obs_mean,fill=resmuts))+geom_density(alpha=0.7)+cleanup+labs(fill=&quot;Known\nResistant\nMutant&quot;)+scale_x_continuous(&quot;Mean net growth rate of screens&quot;)</code></pre>
<p><img src="figure/ABL_kinasefunction_data_parser.Rmd/unnamed-chunk-2-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste(&quot;output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/&quot;,dirname,&quot;/plot3a_Netgrowthrate_distributions_resmuts.pdf&quot;,sep=&quot;&quot;),width=6,height=4,units=&quot;in&quot;,useDingbats=F)
# Plot 3b: Plots: what are the net growth rate distributions?

library(reshape2)
screen_compare_melt=melt(screen_compare_means%&gt;%dplyr::select(species,netgr_obs_screen1,netgr_obs_screen2),id.vars = &quot;species&quot;,measure.vars =c(&quot;netgr_obs_screen1&quot;,&quot;netgr_obs_screen2&quot;),variable.name = &quot;Condition&quot;,value.name = &quot;netgr_obs&quot;)


ggplot(screen_compare_melt,aes(x=netgr_obs,fill=Condition))+
  geom_density(alpha=0.7)+
  cleanup+
  scale_x_continuous(&quot;Net growth rate observed&quot;)+
  scale_fill_discrete(labels=c(&quot;Screen 1&quot;,&quot;Screen 2&quot;))</code></pre>
<p><img src="figure/ABL_kinasefunction_data_parser.Rmd/unnamed-chunk-2-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(paste(&quot;output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/&quot;,dirname,&quot;/plot3b_Netgrowthrate_distributions.pdf&quot;,sep=&quot;&quot;),width=6,height=4,units=&quot;in&quot;,useDingbats=F)

# }

# dirname=&quot;K562_Medium_rep1vs2&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] reshape2_1.4.4     ggpubr_0.4.0       RColorBrewer_1.1-2 doParallel_1.0.15 
 [5] iterators_1.0.12   foreach_1.5.0      tictoc_1.0         plotly_4.9.2.1    
 [9] ggplot2_3.3.3      dplyr_1.0.6        stringr_1.4.0     

loaded via a namespace (and not attached):
 [1] httr_1.4.2        sass_0.4.1        tidyr_1.1.3       jsonlite_1.7.2   
 [5] viridisLite_0.3.0 carData_3.0-3     bslib_0.3.1       assertthat_0.2.1 
 [9] cellranger_1.1.0  yaml_2.2.1        pillar_1.6.1      backports_1.1.7  
[13] glue_1.4.1        digest_0.6.25     promises_1.1.0    ggsignif_0.6.0   
[17] colorspace_1.4-1  htmltools_0.5.2   httpuv_1.5.2      plyr_1.8.6       
[21] pkgconfig_2.0.3   broom_0.7.6       haven_2.4.1       purrr_0.3.4      
[25] scales_1.1.1      whisker_0.4       openxlsx_4.1.5    later_1.0.0      
[29] rio_0.5.16        git2r_0.27.1      tibble_3.1.2      generics_0.0.2   
[33] farver_2.0.3      car_3.0-7         ellipsis_0.3.2    withr_2.4.2      
[37] lazyeval_0.2.2    magrittr_2.0.1    crayon_1.4.1      readxl_1.3.1     
[41] evaluate_0.14     fs_1.4.1          fansi_0.4.1       rstatix_0.6.0    
[45] forcats_0.5.1     foreign_0.8-78    tools_4.0.0       data.table_1.12.8
[49] hms_1.1.0         lifecycle_1.0.0   munsell_0.5.0     zip_2.0.4        
[53] compiler_4.0.0    jquerylib_0.1.4   rlang_0.4.11      grid_4.0.0       
[57] htmlwidgets_1.5.1 labeling_0.3      rmarkdown_2.14    gtable_0.3.0     
[61] codetools_0.2-16  abind_1.4-5       DBI_1.1.0         curl_4.3         
[65] R6_2.4.1          knitr_1.28        fastmap_1.1.0     utf8_1.1.4       
[69] workflowr_1.6.2   rprojroot_1.3-2   stringi_1.7.5     Rcpp_1.0.4.6     
[73] vctrs_0.3.8       tidyselect_1.1.0  xfun_0.31        </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
