<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">duplex_sequencing_screen</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/haiderinam/duplex_sequencing_screen">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>duplex_sequencing_screen/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version
1.6.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhaiderinamduplexsequencingscreentreeba25cb708062de1d10ed0119e287c8c6159659a9targetblankba25cb7a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/tree/ba25cb708062de1d10ed0119e287c8c6159659a9" target="_blank">ba25cb7</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomhaiderinamduplexsequencingscreentreeba25cb708062de1d10ed0119e287c8c6159659a9targetblankba25cb7a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/haiderinam/duplex_sequencing_screen/tree/ba25cb708062de1d10ed0119e287c8c6159659a9" target="_blank">ba25cb7</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/Consensus_Data/.Rhistory
    Ignored:    data/Consensus_Data/Novogene_lane13/sample7/variant_caller_outputs/
    Ignored:    data/Consensus_Data/Novogene_lane13/sample8/variant_caller_outputs/
    Ignored:    data/Consensus_Data/Novogene_lane14/sample10_combined/variant_caller_outputs/duplex/
    Ignored:    data/Consensus_Data/Novogene_lane14/sample13/
    Ignored:    data/Consensus_Data/Novogene_lane14/sample14b/
    Ignored:    data/Consensus_Data/Novogene_lane14/sample1_combined/
    Ignored:    data/Consensus_Data/Novogene_lane14/sample8/variant_caller_outputs/
    Ignored:    data/Consensus_Data/Novogene_lane2/abl_ref/kraken2-master/newdir/
    Ignored:    data/Consensus_Data/Novogene_lane4/.ipynb_checkpoints/
    Ignored:    data/Consensus_Data/Novogene_lane4/GalaxyData/initialanalysis/
    Ignored:    data/Consensus_Data/Novogene_lane4/output/
    Ignored:    data/Consensus_Data/Novogene_lane6/
    Ignored:    data/Consensus_Data/Novogene_lane7/
    Ignored:    data/Consensus_Data/Ranomics_Pooled/RP4/
    Ignored:    data/Consensus_Data/Ranomics_Pooled/RP5/

Untracked files:
    Untracked:  data/Consensus_Data/Novogene_lane14/sample10_combined/duplex_sorted_filtered.tsv
    Untracked:  data/Consensus_Data/Novogene_lane14/sample10_combined/variant_caller_outputs/.DS_Store

Unstaged changes:
    Modified:   .DS_Store
    Modified:   data/.DS_Store
    Modified:   data/Consensus_Data/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane11/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane12/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane12/sample1/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane12/sample3/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane14/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane14/sample10_combined/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane14/sample14_combined/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane14/sample15/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane14/sample16/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane14/sample17/.DS_Store
    Modified:   data/Consensus_Data/Novogene_lane14/sample18/.DS_Store
    Modified:   data/Refs/.DS_Store
    Modified:   data/Refs/EGFR/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/ba25cb708062de1d10ed0119e287c8c6159659a9/analysis/index.Rmd" target="_blank">ba25cb7</a>
</td>
<td>
haiderinam
</td>
<td>
2022-10-29
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/7ed0af87e715e9ea54a21e7b8236e01ebc9f5d9e/analysis/index.Rmd" target="_blank">7ed0af8</a>
</td>
<td>
haiderinam
</td>
<td>
2022-08-22
</td>
<td>
August 2022 Updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/86ce8ca79ec1cd6cc05fdf19c9caa74887f3b1e6/docs/index.html" target="_blank">86ce8ca</a>
</td>
<td>
haiderinam
</td>
<td>
2022-08-22
</td>
<td>
August 2022 Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/f36837112ba05a00bc046fbf25d1612fe6b171ef/docs/index.html" target="_blank">f368371</a>
</td>
<td>
haiderinam
</td>
<td>
2020-08-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/7fe0f9b85304bb797dde69f2183e1330580cf8a3/docs/index.html" target="_blank">7fe0f9b</a>
</td>
<td>
haiderinam
</td>
<td>
2020-07-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/e4a901dfd86a50c98bd6c33f8fe3b7680717a441/analysis/index.Rmd" target="_blank">e4a901d</a>
</td>
<td>
haiderinam
</td>
<td>
2020-07-06
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/231d21d5339b03b3396c16d9698e9d0d4cbeba92/docs/index.html" target="_blank">231d21d</a>
</td>
<td>
haiderinam
</td>
<td>
2020-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/8f50f9978b9aa2f947e4442ec1d64b9c1fe8f0f8/docs/index.html" target="_blank">8f50f99</a>
</td>
<td>
haiderinam
</td>
<td>
2020-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/5da3af039c2fb18c551e6e7435b5a7815f3f0f6c/analysis/index.Rmd" target="_blank">5da3af0</a>
</td>
<td>
haiderinam
</td>
<td>
2020-06-07
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/75ddcb1c935ec2719a2f628578198f5429435597/docs/index.html" target="_blank">75ddcb1</a>
</td>
<td>
haiderinam
</td>
<td>
2020-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/a17368310911a5e13feb7fce1c7311d94c1f87a1/analysis/index.Rmd" target="_blank">a173683</a>
</td>
<td>
haiderinam
</td>
<td>
2020-06-02
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/bd2dec5c9e0d688ae61ad17ec44ca8b2362c7feb/docs/index.html" target="_blank">bd2dec5</a>
</td>
<td>
haiderinam
</td>
<td>
2020-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/1bb59405cc7893438c8e7fed072da289bd6f8890/analysis/index.Rmd" target="_blank">1bb5940</a>
</td>
<td>
haiderinam
</td>
<td>
2020-06-02
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/c43acef02837e22dceb1d5f26525eb506585305e/docs/index.html" target="_blank">c43acef</a>
</td>
<td>
haiderinam
</td>
<td>
2020-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/d5c9f644b55f8903cf05001ad743d1ec4ce2347b/analysis/index.Rmd" target="_blank">d5c9f64</a>
</td>
<td>
haiderinam
</td>
<td>
2020-06-02
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/eaca616beefd04da11c6fa0e68ce3fbba3f95dc7/docs/index.html" target="_blank">eaca616</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/2bba93e7b78d69b4f88ba251c721013357bc157d/analysis/index.Rmd" target="_blank">2bba93e</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-20
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/05e3df6c8977af98a78efd2c9b43d4949f8f8b7e/docs/index.html" target="_blank">05e3df6</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/1c86f6901c62b2224d9ce7e53814c51c24fb4784/docs/index.html" target="_blank">1c86f69</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/e7319092e76176398f1e4d7613a417d5f7235a4d/analysis/index.Rmd" target="_blank">e731909</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-07
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/4516a7f7f33f88b93d79db2b3e814ea0a359910a/docs/index.html" target="_blank">4516a7f</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/3a57291d3b6b690f5538961c43757cf8fad9135b/analysis/index.Rmd" target="_blank">3a57291</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-06
</td>
<td>
wflow_publish(files = "analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/c2930d542662d758f1f66562bdd95d21502868f5/docs/index.html" target="_blank">c2930d5</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/52f688491dab228a6e8343b7838132b266d795b8/analysis/index.Rmd" target="_blank">52f6884</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-03
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/6af2cdcb35c47c946508eba79f1e07fd6c06137a/docs/index.html" target="_blank">6af2cdc</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/5cda9d34f94f99af13ca8ec50946b9d97bdf6a95/analysis/index.Rmd" target="_blank">5cda9d3</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-03
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/0b9b87b45c1b016aa2eddf751c8b2808fa35c141/docs/index.html" target="_blank">0b9b87b</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/fc5b9c0eeb6ac2e20223ac680578f94866f00f3d/analysis/index.Rmd" target="_blank">fc5b9c0</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/2bce927fcb15c952d01f4f64ae00f91ae3846ead/docs/index.html" target="_blank">2bce927</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/4ed9b3560b530899bbab5f15f974623de4ef55b2/docs/index.html" target="_blank">4ed9b35</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/e31836a3fd65b6233f7110de7ed1714b4ff107b4/analysis/index.Rmd" target="_blank">e31836a</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/99181d5229792022da28836551c5a2847ffc4334/docs/index.html" target="_blank">99181d5</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/17c01d498bf4262aeadd6dcd022ce33233aea4dd/analysis/index.Rmd" target="_blank">17c01d4</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
wflow_publish("analysis/index.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/1e2f4690868e3b2f4a611457556480f9670faa8f/docs/index.html" target="_blank">1e2f469</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/e11eec555cc2ccfb95e4483f59848ddc05b2edf5/docs/index.html" target="_blank">e11eec5</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/0053f96d03db97bcb06dde06df6bb30707105d31/analysis/index.Rmd" target="_blank">0053f96</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/haiderinam/duplex_sequencing_screen/b1cbbfa4ca253971653f3aa9197a30d4b6371df5/docs/index.html" target="_blank">b1cbbfa</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/haiderinam/duplex_sequencing_screen/blob/703346a1506471886f08db335f28926a077563df/analysis/index.Rmd" target="_blank">703346a</a>
</td>
<td>
haiderinam
</td>
<td>
2020-04-02
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="duplex-sequencing-spike-ins" class="section level1">
<h1><strong>Duplex Sequencing Spike-Ins</strong></h1>
<p><img src="assets/ds_coverages.png" width="400" height="300"  alt=""></p>
<ul>
<li><p>###Click <a href="dose_response_curve_fitting.html">here</a> for
the script that reads BCRABL dose response data, and fits it to a
4-parameter dose response curve.<br />
<img src="assets/4param_fit.png" width="400" height="300"  alt=""></p></li>
<li><p>###Click <a
href="twinstrand_spikeins_data_generation.html">here</a> for the script
that reads mutation annotation data and generates dataframes with
annotated variants and the correct cell counts</p></li>
<li><p>###Click <a href="growthrate_plots.html">here</a> for initial
growth rate plots<br />
<img src="assets/growthrates.png" width="400" height="300"  alt=""></p></li>
<li><p>###Click <a href="growthrate_plots_ENU.html">here</a> for growth
rate plots for ENU mutants<br />
<img src="assets/growthrates_ENU.png" width="400" height="300"  alt=""></p></li>
<li><p>###Click <a href="spikeins_depthofcoverages.html">here</a> for
analysis looking at whether we achieved our desired depth of coverages
<img src="assets/coverages.png" width="400" height="300"  alt=""></p></li>
<li><p>###Click <a href="clinical_abundance_predictions.html">here</a>
for analysis looking at how well our data predicts BCRABL clinical
abundance compared to conventional IC50s
<img src="assets/clinical_abundance_predictions.png" width="600" height="400"  alt=""></p></li>
<li><p>###Click <a href="misc.html">here</a> for method to obtain
confidence intervals on growth rates from IC50 measurements. Plotted
alongside observed growth rate data, these confidence intervals show how
well our pooled data matches predictions based off of IC50s. (Needs
updates in methodology)<br />
<img src="assets/pooled_trajectories_confidence.png" width="400" height="300"  alt=""></p></li>
<li><p>###Click <a href="4_7_20_update.html">here</a> for analyses of
replicate to replicate agreement between the spike in replicates. Also
includes this analysis for the mutagenesis replicates. Also includes the
agreement in observed vs predicted depth of coverages for all sequencing
pools. Note: this analysis does not include dosage corrections that
improves replicate to replicate heterogeneity.<br />
<img src="assets/mutagenesis_replicates_correlation.png" width="400" height="300"  alt=""></p></li>
<li><p>###Click <a href="nonlinear_growth_analysis.html">here</a> to see
analysis of predictions based on Enrich2 vs Shendure vs our
method</p></li>
<li><p>###Click <a href="dosing_normalization.html">here</a> for our
dosing normalization strategy<br />
<img src="assets/dosing_normalization.png" width="400" height="300"  alt=""></p></li>
<li><p>###Click <a href="enrichment_simulations.html">here</a> for
simulations showing enrichment and depletion events in a 3 mutant pool.
<img src="assets/enrichment_simulations_3mutants.png" width="400" height="300"  alt=""></p></li>
<li><p>###Click <a href="E255K_alphas_figure.html">here</a> showing the
conformity between the growth rates in the presence of drug measured at
a low MAF via FACs, sequencing, and via IC50 studies for a single
mutant.<br />
<img src="assets/e255k_alphas.png" width="400" height="300"  alt=""></p></li>
<li><p>Click <a href="lane14_comparisons.html">here</a> to view
preliminary BCRABL Imatinib and Asciminib heatmaps<br />
<img src="assets/bcrabl_heatmap.png" width="400" height="300"  alt=""></p></li>
</ul>
<div
id="click-here-to-be-redirected-to-the-github-page-that-contains-all-the-data-and-analysis-rmd-files."
class="section level4">
<h4>Click <a
href="https://github.com/haiderinam/duplex_sequencing_screen">here</a>
to be redirected to the github page that contains all the data and
analysis rmd files.</h4>
<p>What is in each directory:</p>
<ul>
<li><strong>Data</strong>: contains data downloaded prior to
analyses.<br />
</li>
<li><strong>Output</strong>: contains data that the code makes<br />
</li>
<li><strong>Analysis</strong>: contains your Rmarkdown files with the
code.<br />
</li>
<li><strong>Docs</strong>: contain the html output from the Rmd files in
the analysis directory.<br />
</li>
<li><strong>Code</strong>: contains .R files that are functions that the
Rmd files in the analysis folder use.</li>
</ul>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
