---
title: "ABL_kinasefunction_data_parser"
author: "Haider Inam"
date: '2023-08-30'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath(".."))

library(stringr)
library(dplyr)
library(ggplot2)
library(plotly)
library(tictoc)
library(doParallel)
library(foreach)
library(RColorBrewer)
library(ggpubr)

# 

```

```{r}
#Cleanup code for plotting
source("code/plotting/cleanup.R")
```

#8.30.23
#The following code was taken from ABL_SM_CRISPR_Cut_Analyses.Rmd
#Essentially all this code does is that it takes two BaF3 ABL Region 1 D0 samples and two D6 samples, and creates a dataframe with enrichment scores.
#Ivan wanted counts from separate D0 backgrounds, and he wanted every mutant, regardless of whether it was seen in both baseline samples (when I do analyses, I take out any mutant that is not seen in both baseline samples). Let's call these "unseen mutants"
#This code does not include unseen mutants by default. To make sure that unseen mutants are seen, uncomment out the Ivan section in compare_screens.R. Once you uncomment it, then run the code below
#One more note: this is Region 1 (Amino acids 242-322) data from Lane 18 only. In the future you can include samples from other lanes.

```{r}
source("code/compare_screens.R")

source("code/plotting/cleanup.R")
source("code/plotting/heatmap_plotting_function.R")
# rm(list=ls())

comparisons=read.csv("data/Consensus_Data/novogene_lane18/TwistRegion1Screen_Comparisons_Todo.csv")
# comparisons=comparisons%>%filter(Completed%in%"FALSE")

# for(i in 1:nrow(comparisons)){
  i=10
  dirname=comparisons$dirname[i]
  pathname=paste("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/",dirname,sep = "")
  # Create directory if it doesn't already exist
  if (!file.exists(pathname)){
    dir.create(pathname)
} 
before_screen1_identifier=unlist(strsplit(comparisons$before_screen1_identifier[i],","))
after_screen1_identifier=unlist(strsplit(comparisons$after_screen1_identifier[i],","))
before_screen2_identifier=unlist(strsplit(comparisons$before_screen2_identifier[i],","))
before_screen2_identifier="BTRID0_FT"
after_screen2_identifier=unlist(strsplit(comparisons$after_screen2_identifier[i],","))
  # length(after_screen1_identifier)
# screen_compare_means=compare_screens(comparisons$before_screen1_identifier[i],
#                                      comparisons$after_screen1_identifier[i],
#                                      comparisons$before_screen2_identifier[i],
#                                      comparisons$after_screen2_identifier[i])
screen_compare_means=compare_screens(before_screen1_identifier,
                                     after_screen1_identifier,
                                     before_screen2_identifier,
                                     after_screen2_identifier)

# Making sure "mutants" are within mutagenized region
screen_compare_means=screen_compare_means%>%filter(protein_start%in%c(position_start:position_end))
screen_compare_means_forexport=apply(screen_compare_means,2,as.character)
# dirname
# write.csv(screen_compare_means_forexport,file="output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/baf3_IL3_rep1vsrep2_ft/screen_comparison_baf3_IL3_low_rep1vsrep2ft.csv")

# Plot 1. What does the heatmap look like from the average of the net growth rate?
heatmap_plotting_function(screen_compare_means,242,321,fill_variable = "netgr_obs_mean",fill_name = "Net growth rate")
# ggsave(paste("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/",dirname,"/plot1_heatmap.pdf",sep=""),width=10,height=6,units="in",useDingbats=F)
# screen_compare_means2=screen_compare_means%>%filter(alt_codon%in%twist$Codon)

# Plot 2a: What do the correlations look like for net growth rate (show mutants in text)?

ggplot(screen_compare_means,aes(x=netgr_obs_screen1,y=netgr_obs_screen2,color=resmuts,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(color="Known\nResistant\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")
# ggsave(paste("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/",dirname,"/plot2a_Netgrowthrate_correlations_text.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)

# Plot 2b: What do the correlations look like for enrichment scores (show mutants in points)?
ggplot(screen_compare_means,aes(x=netgr_obs_screen1,y=netgr_obs_screen2,label=species))+geom_point(color="black",shape=21,size=2,aes(fill=resmuts))+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(fill="Known\nResistant\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")
# ggsave(paste("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/",dirname,"/plot2b_Netgrowthrate_correlations_points.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)
# Plot 2c: What do the correlations look like for enrichment scores (show mutants in text)?
ggplot(screen_compare_means,aes(x=score_screen1,y=score_screen2,color=resmuts,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+ stat_cor(method = "pearson")+labs(color="Known\nResistant\nMutant")+scale_x_continuous("Enrichment score screen 1")+scale_y_continuous("Enrichment score screen 2")
# ggsave(paste("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/",dirname,"/plot2c_Enrichmentscores_correlations_text.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)

# Plot 3a: Plots: what are the overall net growth rate distributions?
ggplot(screen_compare_means,aes(x=netgr_obs_mean,fill=resmuts))+geom_density(alpha=0.7)+cleanup+labs(fill="Known\nResistant\nMutant")+scale_x_continuous("Mean net growth rate of screens")
# ggsave(paste("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/",dirname,"/plot3a_Netgrowthrate_distributions_resmuts.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)
# Plot 3b: Plots: what are the net growth rate distributions?

library(reshape2)
screen_compare_melt=melt(screen_compare_means%>%dplyr::select(species,netgr_obs_screen1,netgr_obs_screen2),id.vars = "species",measure.vars =c("netgr_obs_screen1","netgr_obs_screen2"),variable.name = "Condition",value.name = "netgr_obs")


ggplot(screen_compare_melt,aes(x=netgr_obs,fill=Condition))+
  geom_density(alpha=0.7)+
  cleanup+
  scale_x_continuous("Net growth rate observed")+
  scale_fill_discrete(labels=c("Screen 1","Screen 2"))
# ggsave(paste("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/",dirname,"/plot3b_Netgrowthrate_distributions.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)

# }

# dirname="K562_Medium_rep1vs2"

```



#Ponatilink Analyses
#11.02.23
#The following code was taken from ABL_SM_CRISPR_Cut_Analyses.Rmd
#Essentially all this code does is that it takes two BaF3 ABL Region 1 D0 samples and two D6 samples, and creates a dataframe with enrichment scores.



```{r}
source("code/compare_screens.R")

source("code/plotting/cleanup.R")
source("code/plotting/heatmap_plotting_function.R")
# rm(list=ls())
cellcounts_matrix_dir="data/Consensus_Data/novogene_lane29a/TwistAllRegions_CellCounts_Matrix_ponatilink.csv"
comparisons=read.csv("data/Consensus_Data/novogene_lane29a/TwistAllRegions_Comparisons_Todo_ponatilink.csv")
comparisons=comparisons%>%filter(Completed%in%"FALSE")
comparisons=comparisons%>%filter(Skip_flag%in%"FALSE")

for(i in 1:nrow(comparisons)){
# for(i in 12:nrow(comparisons)){
# for(i in 10:10){
  # i=1
  dirname=comparisons$dirname[i]
  pathname=paste("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/",dirname,sep = "")
  # Create directory if it doesn't already exist
  if (!file.exists(pathname)){
    dir.create(pathname)
} 
before_screen1_identifier=unlist(strsplit(comparisons$before_screen1_identifier[i],","))
after_screen1_identifier=unlist(strsplit(comparisons$after_screen1_identifier[i],","))
before_screen2_identifier=unlist(strsplit(comparisons$before_screen2_identifier[i],","))

after_screen2_identifier=unlist(strsplit(comparisons$after_screen2_identifier[i],","))
position_start=comparisons$ABL_Start[i]
position_end=comparisons$ABL_End[i]
  # length(after_screen1_identifier)
# screen_compare_means=compare_screens(comparisons$before_screen1_identifier[i],
#                                      comparisons$after_screen1_identifier[i],
#                                      comparisons$before_screen2_identifier[i],
#                                      comparisons$after_screen2_identifier[i])
screen_compare_means=compare_screens(before_screen1_identifier,
                                     after_screen1_identifier,
                                     before_screen2_identifier,
                                     after_screen2_identifier,
                                     cellcounts_matrix_dir)
# Making sure "mutants" are within mutagenized region
screen_compare_means=screen_compare_means%>%filter(protein_start%in%c(position_start:position_end))
screen_compare_means_forexport=apply(screen_compare_means,2,as.character)
# dirname

write.csv(screen_compare_means_forexport,file = paste("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/",dirname,"/screen_comparison_",dirname,".csv",sep=""))

# Plot 1. What does the heatmap look like from the average of the net growth rate?
heatmap_plotting_function(screen_compare_means,position_start,position_end,fill_variable = "netgr_obs_mean",fill_name = "Net growth rate")
ggsave(paste("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/",dirname,"/plot1_heatmap.pdf",sep=""),width=10,height=6,units="in",useDingbats=F)
# screen_compare_means2=screen_compare_means%>%filter(alt_codon%in%twist$Codon)

# Plot 2a: What do the correlations look like for net growth rate (show mutants in text)?

ggplot(screen_compare_means,aes(x=netgr_obs_screen1,y=netgr_obs_screen2,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(color="Known\nResistant\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")
ggsave(paste("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/",dirname,"/plot2a_Netgrowthrate_correlations_text.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)

# Plot 2b: What do the correlations look like for enrichment scores (show mutants in points)?
ggplot(screen_compare_means,aes(x=netgr_obs_screen1,y=netgr_obs_screen2,label=species))+geom_point()+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(fill="Known\nResistant\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")
ggsave(paste("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/",dirname,"/plot2b_Netgrowthrate_correlations_points.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)
# Plot 2c: What do the correlations look like for enrichment scores (show mutants in text)?
ggplot(screen_compare_means,aes(x=score_screen1,y=score_screen2,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+ stat_cor(method = "pearson")+labs(color="Known\nResistant\nMutant")+scale_x_continuous("Enrichment score screen 1")+scale_y_continuous("Enrichment score screen 2")
ggsave(paste("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/",dirname,"/plot2c_Enrichmentscores_correlations_text.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)

# Plot 3a: Plots: what are the overall net growth rate distributions?
ggplot(screen_compare_means,aes(x=netgr_obs_mean))+geom_density(alpha=0.7)+cleanup+labs(fill="Known\nResistant\nMutant")+scale_x_continuous("Mean net growth rate of screens")
ggsave(paste("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/",dirname,"/plot3a_Netgrowthrate_distributions_resmuts.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)
# Plot 3b: Plots: what are the net growth rate distributions?

library(reshape2)
screen_compare_melt=melt(screen_compare_means%>%dplyr::select(species,netgr_obs_screen1,netgr_obs_screen2),id.vars = "species",measure.vars =c("netgr_obs_screen1","netgr_obs_screen2"),variable.name = "Condition",value.name = "netgr_obs")


ggplot(screen_compare_melt,aes(x=netgr_obs,fill=Condition))+
  geom_density(alpha=0.7)+
  cleanup+
  scale_x_continuous("Net growth rate observed")+
  scale_fill_discrete(labels=c("Screen 1","Screen 2"))
ggsave(paste("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/",dirname,"/plot3b_Netgrowthrate_distributions.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)

}

# dirname="K562_Medium_rep1vs2"

```

```{r}
screen_compare_means=screen_compare_means%>%
  mutate(ct=case_when(ct_screen1_before%in%1~"1",
                      ct_screen1_before%in%c(2:10)~"2-10",
                      ct_screen1_before%in%c(11:50)~"11-50",
                      ct_screen1_before>=51~"50+"))



twist=read.csv("data/codon_table.csv",header = T)
twist=twist%>%filter(Twist%in%T)
screen_compare_means=screen_compare_means%>%
  mutate(twist_codon=case_when(alt_codon%in%twist$Codon~T,
         T~F))


ggplot(screen_compare_means,aes(x=netgr_obs_screen1,y=netgr_obs_screen2,color=ct,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(color="Known\nResistant\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")

ggplot(screen_compare_means,aes(x=netgr_obs_screen1,y=netgr_obs_screen2,color=twist_codon,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(color="Known\nResistant\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")

# Do duplicates lead to lower correlations?
# Answer: it is not the fact that there are duplicates but the error rate
screen_compare_means=screen_compare_means%>%
  group_by(species)%>%
  mutate(n=n())
x=screen_compare_means[screen_compare_means$n%in%2,]
ggplot(screen_compare_means,aes(x=netgr_obs_screen1,y=netgr_obs_screen2,color=n,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(color="Known\nResistant\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")

# Does filtering for REF length of <=3 remove these duplicates?
# Ans: it removes half of them 
screen_compare_means_filtered=screen_compare_means%>%
  filter(nchar(ref)<=3)%>%
  group_by(species)%>%
  mutate(n=n())
x=screen_compare_means_filtered[screen_compare_means_filtered$n%in%2,]

ggplot(screen_compare_means_filtered,aes(x=netgr_obs_screen1,y=netgr_obs_screen2,color=factor(n),label=species))+geom_text(size=2.5)+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(color="Known\nResistant\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")
```

For ponatilink: Making a combined dataframe of all regions
```{r}
############# PONATILINK SCREENS ############# 

r1=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r1_PL1vsPL5/screen_comparison_k562_r1_PL1vsPL5.csv")
r1=r1%>%filter(protein_start%in%c(240:321))

r2=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r2_PL1vsPL5/screen_comparison_k562_r2_PL1vsPL5.csv")
r2=r2%>%filter(protein_start%in%c(322:393))

r3=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r3_PL1vsPL5/screen_comparison_k562_r3_PL1vsPL5.csv")
r3=r3%>%filter(protein_start%in%c(394:465))

r4=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r4_PL1vsPL5/screen_comparison_k562_r4_PL1vsPL5.csv")
r4=r4%>%filter(protein_start%in%c(466:518))

pl1vs5_all=rbind(r1,r2,r3,r4)

pl1vs5_all=pl1vs5_all%>%dplyr::select(-c("X",
                                         "alt_start_pos",
                                         "score_mean",
                                         "score_screen1",
                                         "ct_screen1_before",
                                         "depth_screen1_before",
                                         "ct_screen1_after",
                                         "depth_screen1_after",
                                         "maf_screen1_before",
                                         "maf_screen1_after",
                                         "score_screen2",
                                         "ct_screen2_before",
                                         "depth_screen2_before",
                                         "ct_screen2_after",
                                         "depth_screen2_after",
                                         "maf_screen2_before",
                                         "maf_screen2_after",
                                         "resmuts",
                                         "resresids",
                                         "netgr_obs_mean"))

pl1vs5_all=pl1vs5_all%>%rename(netgr_PL1=netgr_obs_screen1,netgr_PL5=netgr_obs_screen2)
pl1vs5_all=pl1vs5_all%>%dplyr::select(c("species","consequence_terms"),"ref_aa",everything())



############# Asciminib SCREENS ############# 

r1=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r1_Asc10vsAsc30/screen_comparison_k562_r1_Asc10vsAsc30.csv")
r1=r1%>%filter(protein_start%in%c(240:321))

r2=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r2_Asc10vsAsc30/screen_comparison_k562_r2_Asc10vsAsc30.csv")
r2=r2%>%filter(protein_start%in%c(322:393))

r3=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r3_Asc10vsAsc30/screen_comparison_k562_r3_Asc10vsAsc30.csv")
r3=r3%>%filter(protein_start%in%c(394:465))

r4=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r4_Asc10vsAsc30/screen_comparison_k562_r4_Asc10vsAsc30.csv")
r4=r4%>%filter(protein_start%in%c(466:518))

asc10vsasc30_all=rbind(r1,r2,r3,r4)

asc10vsasc30_all=asc10vsasc30_all%>%dplyr::select(-c("X",
                                         "alt_start_pos",
                                         "score_mean",
                                         "score_screen1",
                                         "ct_screen1_before",
                                         "depth_screen1_before",
                                         "ct_screen1_after",
                                         "depth_screen1_after",
                                         "maf_screen1_before",
                                         "maf_screen1_after",
                                         "score_screen2",
                                         "ct_screen2_before",
                                         "depth_screen2_before",
                                         "ct_screen2_after",
                                         "depth_screen2_after",
                                         "maf_screen2_before",
                                         "maf_screen2_after",
                                         "resmuts",
                                         "resresids",
                                         "netgr_obs_mean"))

asc10vsasc30_all=asc10vsasc30_all%>%rename(netgr_Asc10=netgr_obs_screen1,netgr_Asc30=netgr_obs_screen2)
asc10vsasc30_all=asc10vsasc30_all%>%dplyr::select(c("species","consequence_terms"),"ref_aa",everything())



############# Ponatinib SCREENS ############# 

r1=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r1_P_0.1vsP_0.4/screen_comparison_k562_r1_P_0.1vsP_0.4.csv")
r1=r1%>%filter(protein_start%in%c(240:321))

r2=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r2_P_0.1vsP_0.4/screen_comparison_k562_r2_P_0.1vsP_0.4.csv")
r2=r2%>%filter(protein_start%in%c(322:393))

r3=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r3_P_0.1vsP_0.4/screen_comparison_k562_r3_P_0.1vsP_0.4.csv")
r3=r3%>%filter(protein_start%in%c(394:465))

r4=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r4_Ponatinib0.1vsAsc30/screen_comparison_k562_r4_Ponatinib0.1vsAsc30.csv")
r4=r4%>%filter(protein_start%in%c(466:518))
r4=r4%>%mutate(netgr_obs_screen2=NA)

P0.1vsP0.4_all=rbind(r1,r2,r3,r4)

P0.1vsP0.4_all=P0.1vsP0.4_all%>%dplyr::select(-c("X",
                                         "alt_start_pos",
                                         "score_mean",
                                         "score_screen1",
                                         "ct_screen1_before",
                                         "depth_screen1_before",
                                         "ct_screen1_after",
                                         "depth_screen1_after",
                                         "maf_screen1_before",
                                         "maf_screen1_after",
                                         "score_screen2",
                                         "ct_screen2_before",
                                         "depth_screen2_before",
                                         "ct_screen2_after",
                                         "depth_screen2_after",
                                         "maf_screen2_before",
                                         "maf_screen2_after",
                                         "resmuts",
                                         "resresids",
                                         "netgr_obs_mean"))

P0.1vsP0.4_all=P0.1vsP0.4_all%>%rename(netgr_Pon0.1=netgr_obs_screen1,netgr_Pon0.4=netgr_obs_screen2)
P0.1vsP0.4_all=P0.1vsP0.4_all%>%dplyr::select(c("species","consequence_terms"),"ref_aa",everything())

# colnames(pl1vs5_all)
PL_Asc=merge(pl1vs5_all,
             asc10vsasc30_all,
             by=c("species",
                  "consequence_terms",
                  "ref_aa",
                  "protein_start",
                  "alt_aa",
                  "ref",
                  "alt",
                  "ref_codon",
                  "alt_codon",
                  "alt_codon_shortest",
                  "n_nuc_min"))

Ponatilink_all=merge(PL_Asc,
             P0.1vsP0.4_all,
             by=c("species",
                  "consequence_terms",
                  "ref_aa",
                  "protein_start",
                  "alt_aa",
                  "ref",
                  "alt",
                  "ref_codon",
                  "alt_codon",
                  "alt_codon_shortest",
                  "n_nuc_min"))

write.csv(Ponatilink_all,"output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/ponatilink_combined_growthrates_11.14.23.csv")
```


Plotting all ponatilink data
```{r}
# source("code/plotting/heatmap_plotting_function.R")
kr1d0=read.csv("data/Consensus_Data/novogene_lane29a/sample1/sscs/variant_caller_outputs/variants_unique_ann.csv")
kr1d0=kr1d0%>%filter(protein_start%in%c(242:321),consequence_terms%in%"missense_variant",ct>=1)
heatmap_plotting_function(kr1d0,242,321,fill_name = "Net growth rate")

kr2d0=read.csv("data/Consensus_Data/novogene_lane29a/sample6/sscs/variant_caller_outputs/variants_unique_ann.csv")
kr2d0=kr2d0%>%filter(protein_start%in%c(322:394),consequence_terms%in%"missense_variant",ct>=1)
heatmap_plotting_function(kr2d0,322,394,fill_name = "Net growth rate")

# kr3d6=read.csv("data/Consensus_Data/novogene_lane29a/sample12/sscs/variant_caller_outputs/variants_unique_ann.csv")
kr3d6=read.csv("data/Consensus_Data/novogene_lane29a/sample11/sscs/variant_caller_outputs/variants_unique_ann.csv")
# kr3d6=read.csv("data/Consensus_Data/novogene_lane29a/for_export/sample11/sscs/variant_caller_outputs/variants_unique_ann.csv")
kr3d6=kr3d6%>%filter(protein_start%in%c(394:465),consequence_terms%in%"missense_variant",ct>=1)
heatmap_plotting_function(kr3d6,394,465,fill_name = "Count")


kr4d0=read.csv("data/Consensus_Data/novogene_lane29a/sample16/sscs/variant_caller_outputs/variants_unique_ann.csv")
class(kr4d0$depth)
# class(1)
kr4d0=kr4d0%>%filter(protein_start%in%c(465:512),consequence_terms%in%"missense_variant",ct>=1)
heatmap_plotting_function(kr4d0,465,512,fill_variable = "depth",fill_name = "Net growth rate")


krall=rbind(kr1d0,kr2d0,kr3d6,kr4d0)
krall=krall%>%mutate(maf=ct/depth,species=paste(ref_aa,protein_start,alt_aa,sep=""))
heatmap_plotting_function(krall,242,502,fill_variable = "maf",fill_name = "Allele \nFrequency")
ggsave("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/Ponatilink_heatmap2.pdf",width=18,height=6,units = "in",useDingbats=F)
length(unique(krall$species))
```

```{r}
pl_pon_r4=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r4_Ponatilink5vsPonatinib0.1/screen_comparison_k562_r4_Ponatilink5vsPonatinib0.1.csv")
pl_asc_r4=read.csv("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/k562_r4_Ponatilink5vsAsc30/screen_comparison_k562_r4_Ponatilink5vsAsc30.csv")

# pl_pon_r4_sum=pl_pon_r4%>%group_by(protein_start)%>%summarize(netgr_obs_screen1=mean(netgr_obs_screen1),netgr_obs_screen2=mean(netgr_obs_screen2))
# Ponatilink vs Asciminib
# Ponatilink
pl_asc_r4=pl_asc_r4%>%mutate(resmuts=case_when(protein_start%in%c(468,502)~T,
                                               T~F))
pl_pon_r4=pl_pon_r4%>%mutate(resmuts=case_when(protein_start%in%c(468,502)~T,
                                               T~F))

ggplot(pl_asc_r4%>%filter(protein_start%in%c(465:518)),
                              aes(x=factor(protein_start),y=netgr_obs_screen1,fill=resmuts))+geom_boxplot()+
  geom_hline(yintercept = .025,color="red")+
  cleanup+
  scale_x_discrete(name="Residue on the ABL Kinase")+
  scale_y_continuous(name="Net Growth Rate (Hrs^-1)",limits = c(-.025,.05))+
  theme(axis.text.x=element_text(angle=90, hjust=1))+
  labs(fill="Asciminib \nResresid")
# ggplotly(plotly)
ggsave("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/R4_Resmuts_PL.pdf",width=8,height=3,units="in",useDingbats=F)


# Asciminib
ggplot(pl_asc_r4%>%filter(protein_start%in%c(465:518)),
                              aes(x=factor(protein_start),y=netgr_obs_screen2,fill=resmuts))+geom_boxplot()+
  geom_hline(yintercept = .025,color="red")+
  cleanup+
  scale_x_discrete(name="Residue on the ABL Kinase")+
  scale_y_continuous(name="Net Growth Rate (Hrs^-1)",limits = c(-.025,.05))+
  theme(axis.text.x=element_text(angle=90, hjust=1))+
  labs(fill="Asciminib \nResresid")
# ggplotly(plotly)
ggsave("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/R4_Resmuts_Asc.pdf",width=8,height=3,units="in",useDingbats=F)

# Ponatinib
ggplot(pl_pon_r4%>%filter(protein_start%in%c(465:518)),
                              aes(x=factor(protein_start),y=netgr_obs_screen2,fill=resmuts))+geom_boxplot()+
  geom_hline(yintercept = .025,color="red")+
  cleanup+
  scale_x_discrete(name="Residue on the ABL Kinase")+
  scale_y_continuous(name="Net Growth Rate (Hrs^-1)",limits = c(-.025,.05))+
  theme(axis.text.x=element_text(angle=90, hjust=1))+
  labs(fill="Asciminib \nResresid")
# ggplotly(plotly)
ggsave("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/R4_Resmuts_Pon.pdf",width=8,height=3,units="in",useDingbats=F)





pl_asc_r4_resmuts=pl_asc_r4%>%filter(protein_start%in%c(468,502))
pl_asc_r4_resmuts=pl_asc_r4_resmuts%>%
  dplyr::select(ref_aa,protein_start,alt_aa,n_nuc_min,netgr_obs_screen1,netgr_obs_screen2)
pl_asc_r4_resmuts=melt(pl_asc_r4_resmuts,id.vars = c("ref_aa","protein_start","alt_aa","n_nuc_min"),measure.vars = c("netgr_obs_screen1","netgr_obs_screen1"),variable.name = "screen",value.name = "netgr_obs")

pl_asc_r4_resmuts=ggplot()


# Making density plots:
head(pl_asc_r4$species)
# pl_asc_r4=pl_asc_r4%>%mutate(resmuts=case_when(species%in%c("V468F","I502L","P465S")~T,
                                               # T~F))
# pl_pon_r4=pl_pon_r4%>%mutate(resmuts=case_when(protein_start%in%c(468,502)~T,
                                               # T~F))
a=pl_asc_r4%>%dplyr::select(species,resmuts,ct_screen1_before,netgr_obs_screen1,netgr_obs_screen2)
ggplot(pl_asc_r4%>%filter(protein_start%in%c(465:518)),
                              aes(x=netgr_obs_screen2,fill=resmuts))+
  geom_density(alpha=0.7)+
  # geom_hline(yintercept = .025,color="red")+
  cleanup+
  # scale_x_discrete(name="Residue on the ABL Kinase")+
  # scale_y_continuous(name="Net Growth Rate (Hrs^-1)",limits = c(-.025,.05))+
  theme(axis.text.x=element_text(angle=90, hjust=1))+
  labs(fill="Asciminib \nResresid")

```


Plotting out Imatinib Lane 18(region1) and lane 28 (region234) heatmap
```{r}
# source("code/plotting/heatmap_plotting_function.R")
br1d0=read.csv("data/Consensus_Data/novogene_lane18/sample10/sscs/variant_caller_outputs/variants_unique_ann.csv")
br1d0=br1d0%>%dplyr::select(-X)
br1d0=br1d0%>%filter(protein_start%in%c(242:321),consequence_terms%in%"missense_variant",ct>=1)
heatmap_plotting_function(br1d0,242,321,fill_name = "Net growth rate")

# br2d0=read.csv("data/Consensus_Data/novogene_lane27b/sample1/sscs/variant_caller_outputs/variants_unique_ann.csv")
br2d0=merge_samples("novogene_lane27b/sample1/sscs","novogene_lane27b/sample4/sscs")
# br2d0=merge_samples("novogene_lane27b/sample1/sscs","novogene_lane28/for_export/sample1/sscs")
br2d0=merge_samples(br2d0,"novogene_lane28/for_export/sample1/sscs")
br2d0=br2d0%>%filter(protein_start%in%c(322:394),consequence_terms%in%"missense_variant",ct>=1)
heatmap_plotting_function(br2d0,322,394,fill_name = "Net growth rate")

br3d6=read.csv("data/Consensus_Data/novogene_lane27b/sample6/sscs/variant_caller_outputs/variants_unique_ann.csv")
br3d6=merge_samples("novogene_lane27b/sample5/sscs","novogene_lane27b/sample6/sscs")
br3d6=br3d6%>%filter(protein_start%in%c(394:465),consequence_terms%in%"missense_variant",ct>=1)
heatmap_plotting_function(br3d6,394,465,fill_name = "Net growth rate")
# heatmap_plotting_function(br3d6,242,494,fill_name = "Net growth rate")

# source("code/merge_samples.R")
# br4d0=read.csv("data/Consensus_Data/novogene_lane28/sample17/sscs/variant_caller_outputs/variants_unique_ann.csv")
br4d0=merge_samples("novogene_lane27b/BR4D0_Sample9_10/sscs","novogene_lane28/sample17/sscs")

br4d0=br4d0%>%filter(protein_start%in%c(465:512),consequence_terms%in%"missense_variant",ct>=1)
heatmap_plotting_function(br4d0,465,512,fill_variable = "depth",fill_name = "Net growth rate")


brall=rbind(br1d0,br2d0,br3d6,br4d0)
brall=brall%>%mutate(maf=ct/depth,species=paste(ref_aa,protein_start,alt_aa,sep=""))
heatmap_plotting_function(brall,242,502,fill_variable = "maf",fill_name = "Allele \nFrequency")
ggsave("output/ABLEnrichmentScreens/ABL_Ponatilink_Screens/Imatinib_heatmap.pdf",width=18,height=6,units = "in",useDingbats=F)

length(unique(brall$species))
```
