---
title: "ABL_BaseEditor_Analyses_v2"
author: "Haider Inam"
date: '2023-04-01'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath(".."))
# rm(list=ls())
# 
library(stringr)
library(dplyr)
library(ggplot2)
library(plotly)
library(tictoc)
library(doParallel)
library(foreach)
library(RColorBrewer)
library(pROC)
library(ggpubr)
```

```{r}
# rm(list=ls())
# Contingency table maker function.
# Input: dataframe of sgRNAs and mutants and whether the enrichment scores are significant
# Output: a 2x2 contingency table
# input_df=bedata_inner_simple_filtered%>%filter(Type%in%"ABE")
# input_df=testdata
contab_maker=function(input_df){
  PP=input_df%>%filter(significance_status%in%"Both")
  PP_unique=length(unique(PP$species))
  
  NN=input_df%>%filter(significance_status%in%"Neither")
  NN_unique=length(unique(NN$species))
  
  NP=input_df%>%filter(significance_status%in%"BEOnly")
  NP_unique=length(unique(NP$species))
  
  PN=input_df%>%filter(significance_status%in%"SMOnly")
  PN_unique=length(unique(PN$species))
  
  table=rbind(cbind(PP_unique,PN_unique),cbind(NP_unique,NN_unique))
  table
}
source("code/plotting/heatmap_plotting_function.R")
```
BE vs SM data
When comparing BE vs SM data, you can use a FET of independence to look at which dataset is more independent
You could also use an ROC curve that compares different prediction algorithms for BE screens and scores based on whether a mutant is also significant in the SM screens.
Can the BE data predict solvent accessibility? As well as SM screens?
solvent accessibility is a rigorous test for BE screens.
An easier test might be to see if BE screens can see Motif effects because 
those are not just focused on a single residue presumably

```{r}
source("code/plotting/cleanup.R")
# The following function parses bedata so that its compatible with the SM data and is ready to be merged with it.
# bedata_outer=read.csv("data/BE_ABL_Merged/BE_SatMut_Screen_Outer20230315.csv",header=T,stringsAsFactors = F)
bedata_parser=function(input_df){
  # Rearrange BE data, rename species to mutation
  bedata_outer=bedata_outer%>%
  rename(BE.Alt_Codon=Alt_Codon,SM.Alt_Codon=alt_codon,BE.LFC=BE_LFC,BE.pval=BE_p.value,BE.FDR=BE_FDR,SM.pval=pvalue,SM.padj=padj,SM.Significant=Significant,SM.ref=ref,SM.alt=alt,SM.LFC=log2FoldChange)%>%
  relocate(ref_aa,protein_start,alt_aa,species,ref_codon,BE.Alt_Codon,SM.Alt_Codon,alt_codon_shortest,n_nuc_min,AA_Number)
  bedata_outer=bedata_outer%>%dplyr::select(sgNAME,sgRNA.Seq,PAM,Type,sgRNA_Nuc,Target_Nuc,Ref_Codon,BE.Alt_Codon,BE.MAF.pDNA,Ref_AA,Alt_AA,ABL1_AA,BE.LFC,BE.pval,BE.FDR,BE_Screen,species=Mutation)
  
  # Filter out mutants only seen in the SM screen, not in the BE screen
  # bedata_outer=bedata_outer%>%filter(!BE_Screen%in%"",!species%in%"")
  # Filter for mutants in the kinase
  # bedata_outer=bedata_outer%>%filter(ABL1_AA%in%c(242:322))
  
  # length(unique(bedata_outer$species))
  bedata_inner_simple=bedata_outer
  #############Distances###########
  # Calculate the distance from PAM
  bedata_inner_simple=bedata_inner_simple%>%mutate(sgRNA_Nuc=gsub("\\[|\\]","",sgRNA_Nuc))
  # Figuring out which mutants are at the minimum distance
  
  # N_nuc tells you whether its a SNP or an MNV
  bedata_inner_simple=bedata_inner_simple%>%mutate(BE.n_nuc=str_count(sgRNA_Nuc,",")+1)
  # How do you calculate the distance for an MNV? Find distance from just one of it's nucleotides
  bedata_inner_simple=bedata_inner_simple%>%mutate(distance=case_when(BE.n_nuc%in%1~sgRNA_Nuc,
                                  BE.n_nuc>=2~strsplit(sgRNA_Nuc,",")[[1]][1]))
  
  bedata_inner_simple$distance=as.numeric(bedata_inner_simple$distance)
  bedata_inner_simple=bedata_inner_simple%>%mutate(distance_from_6=abs(6-distance))
  
  # For each sgRNA, figure out which one mutant is at the minimum distance
  # If there are multiple mutants at the minimum distance, note down both of them.
  bedata_sum=bedata_inner_simple%>%
    group_by(Type,sgRNA.Seq)%>%
    summarize(mutants_per_sgRNA=n(),
              mindist=min(distance_from_6),
              species.mindist=paste(species[which(distance_from_6==min(distance_from_6))],collapse=", "))
  # Sometimes a guide makes the same amino acid substitution two different ways (eg a snp and an mnv that make the same substitution). When this happens, the algorithm thinks that the guide is making two separate amino acid substitutions. this next conditional statement is going to remove those duplicates.
  bedata_sum=bedata_sum%>%
    rowwise()%>%
    mutate(species.mindist=case_when(
      strsplit(species.mindist,", ")[[1]][1]==
        strsplit(species.mindist,", ")[[1]][2]~strsplit(species.mindist,", ")[[1]][1],
                                     T~species.mindist))
  bedata_inner_simple=merge(bedata_inner_simple,bedata_sum,by=c("Type","sgRNA.Seq"))
  bedata_inner_simple
}
```



```{r}
netgr_wt=.06
sm_start=242
sm_end=322
# rm(list=ls())
# Adding function to merge SM and BE data
smdata=read.csv("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/baf3_IL3_rep1vsrep2_ft/screen_comparison_baf3_IL3_low_rep1vsrep2_ft.csv",header = T)
# smdata=read.csv("output/ABLEnrichmentScreens/IL3_Enrichment_bgmerged_2.20.23.csv",header = T)
smdata=smdata%>%rowwise%>%mutate(netgr_obs_mean=mean(netgr_obs_screen1,netgr_obs_screen2))
smdata=smdata%>%dplyr::select(species,ref_aa,protein_start,alt_aa,netgr_obs_mean)
# Read BE data
bedata_outer=read.csv("data/BE_ABL_Merged/BE_SatMut_Screen_Outer20230315.csv",header=T,stringsAsFactors = F)
# The following few lines of code helped me find if there are any sgRNA sequences on the + strand that are also the same sgRNA seq on the - strand? Answer is no
#### Adding baseline of BE screens
baseline=read.table("data/BE_ABL_Merged/BCRABL_ScreenMatrixE2_20220920BaF3.txt",sep = "\t",header = T)

baseline=baseline%>%mutate(BE.MAF.pDNA=(ABL1_C6+ABL1_G11)/sum(ABL1_C6+ABL1_G11))%>%dplyr::select(sgNAME=sgName,BE.MAF.pDNA)

bedata_outer=merge(bedata_outer,baseline,by="sgNAME")


# bedata_outer=bedata_outer%>%select(sgRNA.Seq,Strand)
# pos=bedata_outer%>%filter(Strand%in%"pos")
# pos=rbind(pos,c("ACTCGACGTTCACGTAGAAG","neg"))
# neg=bedata_outer%>%filter(Strand%in%"neg")
# posneg=merge(pos,neg,by="sgRNA.Seq")


#############Parsing BE data###########
bedata_simple=bedata_parser(bedata_outer)
#############Merging BE data and SM data###########
# Merge BE data and SM data by colname: species
# Note that when you merge BE data and SM data, you lose the synonymous SNPs, which the SM data does not have. To keep the synonymous SNPs, use all.x=T
bedata_simple=merge(bedata_simple,smdata,by="species",all.x = T)

# Filter out mutants only seen in the SM screen, not in the BE screen
  bedata_simple=bedata_simple%>%filter(!species%in%"")
  # Filter for mutants in the kinase
  # bedata_simple=bedata_simple%>%filter(ABL1_AA%in%c(sm_start:sm_end))
  # For synonymous substitutions, assume they take the WT net growth rate
  bedata_simple=bedata_simple%>%mutate(netgr_obs_mean=case_when(netgr_obs_mean%in%NA~netgr_wt,
                                                                T~netgr_obs_mean))
  # x=bedata_simple%>%filter(!netgr_obs_mean%in%NA,Type%in%"ABE",sgRNA_Nuc%in%c(3,4,5,6,7,8,9))
  # length(unique(x$species))
#############Filtering for SNPs###########
bedata_simple=bedata_simple%>%filter(BE.n_nuc%in%1)
# bedata_inner_simple2=bedata_inner_simple
# bedata_inner_simple=bedata_inner_simple%>%filter(mutants_per_sgRNA%in%1)
#############Method 1: Calculated mean net growth rates without weights###########  
#############Adding the tendency of each sgRNA, as predicted by SM###########
# bedata_simple=bedata_simple%>%filter(distance_from_6%in%c(0,1,2,3,4))%>%group_by(Type,sgRNA.Seq)%>%mutate(sgRNA.SM.Mean.Netgr=mean(netgr_obs_mean))

#############Grouping by Mutants, not sgRNAs###########
# bedata_bymutant=bedata_simple%>%filter(Type%in%"ABE",distance_from_6%in%c(0,1,2,3,4))%>%group_by(species)%>%summarize(SM.netgr_obs_mean=mean(netgr_obs_mean),
#                                         BE.LFC=mean(BE.LFC),
#                                         sgRNA.SM.Mean.Netgr=mean(sgRNA.SM.Mean.Netgr),
#                                         sgRNAs_per_mutant=n())
##############################################################################
#############Method 2: Calculated mean net growth rates as weighted means###########  
#############Adding the tendency of each sgRNA, as predicted by SM###########  
  distance=c(1:11)
  weight=c(.1,1.8,7.5,16.5,20,22,16,13.5,6,3.5,3.5)
  abe_weights=data.frame(cbind(distance,weight))
  bedata_simple=bedata_simple%>%filter(distance%in%c(1:11))%>%rowwise()%>%mutate(weight=abe_weights[abe_weights$distance==distance,"weight"])
  bedata_simple=bedata_simple%>%
    group_by(Type,sgRNA.Seq)%>%
    mutate(sgRNA.SM.Mean.Netgr=weighted.mean(netgr_obs_mean,weight),
           sgRNA.SM.Mean.Netgr=mean(netgr_obs_mean),
           mutants_per_sgRNA=n())
  # 4.11.23: the weighted mean actually seems to be making sgRNA.SM.Netgr.Mean worse at predicting BE data, so I'm taking it out
#############Grouping by Mutants, not sgRNAs###########
  # bedata_bymutant=bedata_simple%>%filter(Type%in%"ABE")%>%group_by(Type,species)%>%summarize(SM.netgr_obs_mean=weighted.mean(netgr_obs_mean,weight),
  #                                       BE.LFC=weighted.mean(BE.LFC,weight),
  #                                       sgRNA.SM.Mean.Netgr=weighted.mean(sgRNA.SM.Mean.Netgr,weight),
  #                                       sgRNAs_per_mutant=n())
  bedata_bymutant=bedata_simple%>%
    filter(Type%in%"ABE",distance_from_6%in%c(0,1,2,3,4))%>%
    # filter(mutants_per_sgRNA%in%1)%>%
    group_by(Type,species)%>%
    summarize(SM.netgr_obs_mean=mean(netgr_obs_mean),
                                        BE.LFC=mean(BE.LFC),
                                        sgRNA.SM.Mean.Netgr=mean(sgRNA.SM.Mean.Netgr),
                                        sgRNAs_per_mutant=n())
  ###Note, if you want to trust the sgRNAs with a lower baseline MAF less, then you can weight them down below.
  # In my experiene, that doesn't really help with scores,
# bedata_bymutant=bedata_simple%>%filter(Type%in%"ABE",distance_from_6%in%c(0,1,2,3,4))%>%group_by(species)%>%summarize(SM.netgr_obs_mean=weighted.mean(netgr_obs_mean,BE.MAF.pDNA),
#                                         BE.LFC=mean(BE.LFC),
#                                         sgRNA.SM.Mean.Netgr=weighted.mean(sgRNA.SM.Mean.Netgr,BE.MAF.pDNA),
#                                         sgRNAs_per_mutant=n())

##############################################################################  
  
  
  bedata_bymutant=bedata_bymutant%>%
    rowwise()%>%
    mutate(synonymous=case_when(SM.netgr_obs_mean==netgr_wt~T,
                                                                T~F),
           protein_start=as.numeric(gsub("^.(.*).$", "\\1", species)))
#############Calculating if mutants are significantly depleting###########
bedata_bymutant=bedata_bymutant%>%rowwise()%>%
mutate(BE.Significant=case_when(BE.LFC<=-.75~T,
                                BE.LFC>=1~T,
                                T~F),
       SM.Significant=case_when(SM.netgr_obs_mean<.045~T,
                                SM.netgr_obs_mean>.09~T,
                                T~F),
       significance_status=case_when((BE.Significant%in%T)&&(SM.Significant%in%F)~"BEOnly",
                                     (BE.Significant%in%F)&&(SM.Significant%in%T)~"SMOnly",
                                     (BE.Significant%in%T)&&(SM.Significant%in%T)~"Both",
                                     T~"Neither"))

# length(unique(x_sum$species))
# x_sum=bedata_inner_simple%>%filter(Type%in%"ABE",distance_from_6%in%c(0,1,2))%>%group_by(species)%>%summarize(SM.netgr_obs_mean=mean(netgr_obs_mean),
#                                         BE.LFC=mean(BE_LFC),
#                                         num_sgRNAs=n())
```
Does BE LFC predict the putative mutant data better or the admixture data better?
```{r}
ggplot(bedata_bymutant%>%filter(protein_start%in%c(242:322),synonymous%in%F,Type%in%"ABE"),aes(y=SM.netgr_obs_mean,x=BE.LFC,color=significance_status))+geom_point()
ggplot(bedata_bymutant%>%filter(protein_start%in%c(242:322),synonymous%in%F,Type%in%"ABE"),aes(y=sgRNA.SM.Mean.Netgr,x=BE.LFC,color=significance_status))+geom_point()
# ggplot(bedata_bymutant%>%filter(protein_start%in%c(242:322),synonymous%in%F,Type%in%"ABE"),aes(y=sgRNA.SM.Mean.Netgr,x=sgRNA.BE.Mean.LFC,color=significance_status))+geom_point()
bedata_roc=bedata_bymutant%>%filter(protein_start%in%c(242:322),Type%in%"ABE",synonymous%in%F)

# glm.fit.putative=glm(as.numeric(bedata_roc$SM.Significant)~bedata_roc$BE.LFC,family=binomial)
# bedata_roc$glm_fits=glm.fit.putative$fitted.values
# # ggplot(be_lfc.be,aes(x=BE.LFC,y=as.numeric(DSSP.Buried)))+geom_point()+geom_line(aes(x=BE.LFC,y=glm_fits))
# roc(as.numeric(bedata_roc$SM.Significant),bedata_roc$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T)
# 
# bedata_roc=bedata_roc%>%mutate(SM.Significant.admixture=case_when(sgRNA.SM.Mean.Netgr<=.045~T,
#                                                                   T~F))
# glm.fit.putative=glm(as.numeric(bedata_roc$SM.Significant.admixture)~bedata_roc$BE.LFC,family=binomial)
# bedata_roc$glm_fits=glm.fit.putative$fitted.values
# # ggplot(be_lfc.be,aes(x=BE.LFC,y=as.numeric(DSSP.Buried)))+geom_point()+geom_line(aes(x=BE.LFC,y=glm_fits))
# roc(as.numeric(bedata_roc$SM.Significant.admixture),bedata_roc$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T)


glm.fit.putative=glm(as.numeric(bedata_roc$BE.Significant)~bedata_roc$SM.netgr_obs_mean,family=binomial)
bedata_roc$glm_fits.putative=glm.fit.putative$fitted.values
# ggplot(be_lfc.be,aes(x=BE.LFC,y=as.numeric(DSSP.Buried)))+geom_point()+geom_line(aes(x=BE.LFC,y=glm_fits))
roc1=roc(as.numeric(bedata_roc$BE.Significant),bedata_roc$glm_fits.putative,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T)

glm.fit.admixture=glm(as.numeric(bedata_roc$BE.Significant)~bedata_roc$sgRNA.SM.Mean.Netgr,family=binomial)
bedata_roc$glm_fits.admixture=glm.fit.admixture$fitted.values
# ggplot(be_lfc.be,aes(x=BE.LFC,y=as.numeric(DSSP.Buried)))+geom_point()+geom_line(aes(x=BE.LFC,y=glm_fits))
roc2=roc(as.numeric(bedata_roc$BE.Significant),bedata_roc$glm_fits.admixture,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T)
roc.test(roc1,roc2,alternative="less")
# Note that the pvalue of this one-sided DeLong's non-parametric test of comparing two ROCs is 0.0843

par(pty="s")
roc(as.numeric(bedata_roc$BE.Significant),bedata_roc$glm_fits.putative,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#3937E3",lwd=4)
plot.roc(as.numeric(bedata_roc$BE.Significant),bedata_roc$glm_fits.admixture,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#378BE3",add=T,lwd=4,print.auc.y=45)
par(pty="m")
```

```{r}
# Overall Correlation
bedata_bymutant=bedata_bymutant%>%filter(synonymous%in%F,Type%in%"ABE")
ggplot(bedata_bymutant%>%filter(synonymous%in%F),aes(y=SM.netgr_obs_mean,x=BE.LFC,color=significance_status))+geom_point()
ggplot(bedata_bymutant,aes(y=SM.netgr_obs_mean,x=BE.LFC,color=significance_status,label=species))+geom_text()
x=bedata_bymutant%>%filter(synonymous%in%F,Type%in%"ABE")
cor(x$SM.netgr_obs_mean,x$BE.LFC)
cor(x$sgRNA.SM.Mean.Netgr,x$BE.LFC)
x=bedata_simple%>%filter(protein_start%in%c(242:322),!Ref_AA==Alt_AA,Type%in%"ABE")
cor(x$sgRNA.SM.Mean.Netgr,x$BE.LFC)

ggplot(bedata_bymutant,aes(y=SM.netgr_obs_mean,x=BE.LFC,label=species,color=significance_status))+geom_text()

########### Do false positives tend to be made by less sgRNAs? No###########
ggplot(bedata_bymutant,aes(x=significance_status,y=sgRNAs_per_mutant,fill=significance_status))+geom_boxplot()+
  geom_jitter(color="black", size=1,width=.1, alpha=0.9)
########### Does having more sgRNAs help at all? Not really#########
#Many sgRNAs create a single mutation. That helps create trust in a mutant's observed score. 
# In a way, a mutant is an admixture of admixtures, does having more sgRNAs help? Not really
ggplot(bedata_bymutant,aes(y=SM.netgr_obs_mean,x=BE.LFC,color=sgRNAs_per_mutant))+geom_point()

####### Is the depletion signal potentially masked by other mutants that an sgRNA makes? Yes#######
# Is the sgRNA.SM.Mean.Netgr of false negatives higher than that of true positives? That would indicate that false negative mutations look like they have a high LFC because they are created by sgRNAs that make neutral mutants. Ans: yes
bedata_bymutant2=bedata_bymutant
bedata_bymutant2$significance_status=factor(bedata_bymutant2$significance_status,levels=c("Neither","BEOnly","Both","SMOnly"))
ggplot(bedata_bymutant2,aes(x=significance_status,y=sgRNA.SM.Mean.Netgr,fill=significance_status))+geom_boxplot()+
  geom_jitter(color="black", size=1,width=.1, alpha=0.9)+scale_fill_manual(values=c("#00BFC4","#F8766D","#7CAE00","#C77CFF"))

# If we calculate the expected net growth rate of an sgRNA by simulating the admixture it creates,
# we get a decent correlation coefficient
ggplot(bedata_simple%>%filter(protein_start%in%c(242:322),!Ref_AA==Alt_AA,Type%in%"ABE"),aes(y=sgRNA.SM.Mean.Netgr,x=BE.LFC))+geom_point()+stat_cor(method="pearson")+scale_x_continuous("sgRNA BE LFC")+scale_y_continuous("Net growth rate mixture model")+cleanup
# ggsave("output/BE_SM_Plots/correlation_mixturemodel.pdf",width=3,height=3,units="in",useDingbats=F)
ggplot(bedata_simple%>%filter(protein_start%in%c(242:322),!Ref_AA==Alt_AA,Type%in%"ABE"),aes(y=netgr_obs_mean,x=BE.LFC))+geom_point()+stat_cor(method="pearson")+scale_x_continuous("sgRNA BE LFC")+scale_y_continuous("Net growth rate non-mixture model")+cleanup
# ggsave("output/BE_SM_Plots/correlation_nonmixturemodel.pdf",width=3,height=3,units="in",useDingbats=F)
# To do later: are your sgRNA predicted growth rates better if you consider all mutants, not just the putative mutant? Answer: the predicted growth rate of the mixture model doesn't really predict much better. However, it is just as good as the putative mutant, so it is still useful when considering all possible mutants that an sgRNA can create. 
# bedata_putative=bedata_simple%>%filter(protein_start%in%c(242:322),!Ref_AA==Alt_AA,Type%in%"ABE")%>%group_by(sgRNA.Seq)%>%summarize(netgr_obs_mean=netgr_obs_mean[min(distance_from_6)==T],BE.LFC=BE.LFC[min(distance_from_6)==T])
# ggplot(bedata_putative,aes(y=netgr_obs_mean,x=BE.LFC))+geom_point()+stat_cor(method="pearson")+scale_x_continuous("sgRNA BE LFC")+scale_y_continuous("Net growth rate putative mutant")+cleanup
# length(unique(x$sgRNA.Seq))

# Overall correlation for predicted admixtures
ggplot(bedata_bymutant,aes(y=sgRNA.SM.Mean.Netgr,x=BE.LFC,color=significance_status))+geom_point()
ggplot(bedata_bymutant,aes(y=sgRNA.SM.Mean.Netgr,x=BE.LFC,color=significance_status,label=species))+geom_text()
# cor(bedata_bymutant$SM.netgr_obs_mean,bedata_bymutant$BE.LFC)
# x=bedata_bymutant%>%filter(!SM.netgr_obs_mean%in%"0.060000000")

# color codes: red: #F8766D, green: #7CAE00, blue: #00BFC4, purple: #C77CFF
# x=bedata_dssp%>%filter(ACC>=50,protein_start%in%c(242:322))
```
Calculating residue-level enrichment scores from the BE screen
Sliding window variables:
1. Length of what you're trying to correlate the data with
2. Length of the prediction window: how much do you trust 
```{r}
smdata=read.csv("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/baf3_IL3_rep1vsrep2_ft/screen_comparison_baf3_IL3_low_rep1vsrep2_ft.csv",header = T)
smdata=smdata%>%rowwise%>%mutate(netgr_obs_mean=mean(netgr_obs_screen1,netgr_obs_screen2))
smdata=smdata%>%dplyr::select(species,ref_aa,protein_start,alt_aa,netgr_obs_mean)

dssp=read.csv("data/DSSP_SolventAccessibility_ABL/2hyy_dspp.csv",header = T)
dssp=dssp%>%mutate(RESIDUE=as.numeric(RESIDUE),ACC=as.numeric(ACC),AA=gsub("<ca>","",AA))
dssp=dssp%>%rename(protein_start=RESIDUE)

# x=bedata_outer%>%filter(ABL1_AA%in%c(242:322))%>%group_by(Type,sgRNA.Seq)%>%mutate(mutants_per_sgRNA=n())
# length(unique(x$sgRNA.Seq))

# x=x%>%filter(ABL1_AA%in%c(242:322),mutants_per_sgRNA%in%c(1))
# length(unique(x$sgRNA.Seq))

# x=bedata_simple%>%filter(Type%in%"ABE",ABL1_AA%in%c(242:322),distance_from_6%in%c(0,1))%>%dplyr::select(protein_start=ABL1_AA,BE.LFC,netgr_obs_mean,sgRNA.SM.Mean.Netgr,weight)%>%ungroup()%>%group_by(protein_start)%>%summarize(BE.LFC_atresidue=mean(BE.LFC),
                                                                                                                                                                                                              # BE.LFC_atresidue=weighted.mean(BE.LFC,weight),
                                                                                                                                                                                                              # SM.netgr_obs_mean=weighted.mean(netgr_obs_mean,weight),
                                                                                                                                                                                                              # SM.netgr_obs_mean=mean(netgr_obs_mean),
                                                                                                                                                                                                              # sgRNA.SM.Mean.Netgr=mean(sgRNA.SM.Mean.Netgr))
# y=bedata_outer%>%filter(sgRNA.Seq%in%"AGCTGGGCGGGGGCCAGTAC")
# rm(weight)
# bedata_dssp=merge(x,dssp,by="protein_start")
# bedata_dssp=bedata_dssp%>%mutate(exposed=case_when(ACC>=40~"Exposed",
#                                             T~"Buried"))

# ggplot(bedata_dssp,aes(x=ACC))+geom_histogram()
# 
# ggplot(bedata_dssp,aes(x=protein_start,y=BE.LFC_atresidue,color=exposed))+
#   geom_point()+
#   facet_wrap(~exposed,nrow=2)+
#   scale_y_continuous("Mean net growth rate at residue")+
#   scale_x_continuous("Residue on the ABL Kinase")+
#   labs(color="Solvent \nAccessibility")+cleanup
# 
# ggplot(bedata_dssp,aes(x=protein_start,y=SM.netgr_obs_mean,color=exposed))+
#   geom_point()+
#   facet_wrap(~exposed,nrow=2)+
#   scale_y_continuous("Mean net growth rate at residue")+
#   scale_x_continuous("Residue on the ABL Kinase")+
#   labs(color="Solvent \nAccessibility")+cleanup
# 
# ggplot(bedata_dssp,aes(x=protein_start,y=sgRNA.SM.Mean.Netgr,color=exposed))+
#   geom_point()+
#   facet_wrap(~exposed,nrow=2)+
#   scale_y_continuous("Mean net growth rate at residue")+
#   scale_x_continuous("Residue on the ABL Kinase")+
#   labs(color="Solvent \nAccessibility")+cleanup
# 
# ggplot(il3_dssp,aes(x=protein_start,y=netgr_obs_mean,color=exposed))+
#   geom_point()+
#   facet_wrap(~exposed,nrow=2)+
#   scale_y_continuous("Mean net growth rate at residue")+
#   scale_x_continuous("Residue on the ABL Kinase")+
#   labs(color="Solvent \nAccessibility")+cleanup

# first, I'm going to do a simple sliding window that jumps 1 amino acid at a time
bedata_byresidue=data.frame(matrix(ncol = 4, nrow=length(c(sm_start:sm_end))))
colnames(bedata_byresidue)=c("protein_start","BE.LFC.mean","BE.LFC.weighted.mean","SM.netgr.mean")

for(i in seq(sm_start:sm_end)){
  # aa_i=242
  aa_i=i+sm_start-1
  subset_i=bedata_simple%>%filter(Type%in%"ABE",ABL1_AA%in%aa_i)
  bedata_byresidue[i,"protein_start"]=aa_i
  bedata_byresidue[i,"BE.LFC.mean"]=mean(subset_i$BE.LFC)
  bedata_byresidue[i,"BE.LFC.weighted.mean"]=weighted.mean(subset_i$BE.LFC,subset_i$weight)
  bedata_byresidue[i,"SM.netgr.mean"]=mean(subset_i$netgr_obs_mean)
}
bedata_dssp=merge(bedata_byresidue,dssp,by="protein_start")

bedata_dssp=bedata_dssp%>%mutate(exposed=case_when(ACC>=40~"Exposed",
                                            T~"Buried"))

ggplot(bedata_dssp,aes(x=protein_start,y=BE.LFC.mean,color=exposed))+
  geom_point()+
  # facet_wrap(~exposed,nrow=2)+
  scale_y_continuous("Mean net growth rate at residue")+
  scale_x_continuous("Residue on the ABL Kinase")+
  labs(color="Solvent \nAccessibility")+cleanup

ggplot(bedata_dssp,aes(x=BE.LFC.mean,y=SM.netgr.mean))+geom_point()
cor(bedata_dssp[!bedata_dssp$BE.LFC.mean%in%NaN,"BE.LFC.mean"],bedata_dssp[!bedata_dssp$BE.LFC.mean%in%NaN,"SM.netgr.mean"])
ggplot(bedata_dssp,aes(x=BE.LFC.mean,y=ACC))+geom_point()
ggplot(bedata_dssp,aes(x=SM.netgr.mean,y=ACC))+geom_point()

subset_i=bedata_simple%>%filter(Type%in%"ABE",ABL1_AA%in%c(301:305))
# bedata_byresidue=bedata_bymutant%>%group_by(Type,protein_start)
```

Adding window length as a variable to the sliding window for loop
```{r}
# rm(list=ls())
smdata=read.csv("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/baf3_IL3_rep1vsrep2_ft/screen_comparison_baf3_IL3_low_rep1vsrep2_ft.csv",header = T)
smdata=smdata%>%rowwise%>%mutate(netgr_obs_mean=mean(netgr_obs_screen1,netgr_obs_screen2))
smdata=smdata%>%dplyr::select(species,ref_aa,protein_start,alt_aa,netgr_obs_mean)

dssp=read.csv("data/DSSP_SolventAccessibility_ABL/2hyy_dspp.csv",header = T)
dssp=dssp%>%mutate(RESIDUE=as.numeric(RESIDUE),ACC=as.numeric(ACC),AA=gsub("<ca>","",AA))
dssp=dssp%>%rename(protein_start=RESIDUE)

window_widths=c(1:5)
bedata_byresidue=data.frame(matrix(ncol = 4, nrow=length(c(sm_start:sm_end))))
colnames(bedata_byresidue)=c("window.width","protein_start","BE.LFC.mean","BE.LFC.weighted.mean")
for(j in 1:length(window_widths)){
  window_width_i=window_widths[j]
  # window_width_i=1
  for(i in seq(sm_start:sm_end)){
    # aa_i=242
    aa_i=i+sm_start-1
    subset_i=bedata_simple%>%filter(Type%in%"ABE",ABL1_AA%in%c(aa_i:(aa_i+window_width_i-1)))
    sm_subset_i=smdata%>%filter(protein_start%in%c(aa_i:(aa_i+window_width_i-1)))
    bedata_byresidue[i,"window.width"]=window_width_i
    bedata_byresidue[i,"protein_start"]=aa_i
    bedata_byresidue[i,"BE.LFC.mean"]=mean(subset_i$BE.LFC)
    bedata_byresidue[i,"BE.LFC.weighted.mean"]=weighted.mean(subset_i$BE.LFC,subset_i$weight)
    # bedata_byresidue[i,"SM.netgr.mean.BEsubset"]=mean(subset_i$netgr_obs_mean)
    # bedata_byresidue[i,"SM.netgr.mean"]=mean(sm_subset_i$netgr_obs_mean)
  }
  if(j%in%1){
    bedata_byresidue_bywindowwidth=bedata_byresidue
  } else {
  bedata_byresidue_bywindowwidth=rbind(bedata_byresidue_bywindowwidth,bedata_byresidue)  
  }
}
# bedata_byresidue=bedata_simple%>%filter(Type%in%"ABE",ABL1_AA%in%c(sm_start:sm_end))%>%group_by(ABL1_AA)%>%summarize(BE.LFC.mean=mean(BE.LFC),BE.LFC.weighted.mean=weighted.mean(BE.LFC,weight))%>%mutate(window.width=0)%>%dplyr::select(window.width,protein_start=ABL1_AA,BE.LFC.mean,BE.LFC.weighted.mean)
# bedata_byresidue_bywindowwidth=rbind(bedata_byresidue,bedata_byresidue_bywindowwidth)
smdata_byresidue=smdata%>%filter(protein_start%in%c(sm_start:sm_end))%>%group_by(protein_start)%>%summarize(SM.netgr.mean=mean(netgr_obs_mean))
bedata_byresidue_bywindowwidth=merge(bedata_byresidue_bywindowwidth,smdata_byresidue%>%dplyr::select(protein_start,SM.netgr.mean))
# x=bedata_byresidue_bywindowwidth%>%filter(protein_start%in%242)
bedata_dssp=merge(bedata_byresidue_bywindowwidth,dssp,by="protein_start")

bedata_dssp=bedata_dssp%>%mutate(exposed=case_when(ACC>=40~"Exposed",
                                            T~"Buried"))

ggplot(bedata_dssp,aes(x=protein_start,y=BE.LFC.mean,color=exposed))+
  geom_point()+
  # facet_wrap(~exposed,nrow=2)+
  # facet_grid(exposed~window.width)+
  facet_wrap(~window.width,nrow=5)+
  scale_y_continuous("Mean BE LFC at residue")+
  scale_x_continuous("Residue on the ABL Kinase")+
  labs(color="Solvent \nAccessibility")+cleanup+theme(legend.position = "none")
# ggsave("output/BE_SM_Plots/be_slidingwindow_all.pdf",width=4,height=5,units="in",useDingbats=F)
# ggplot(bedata_dssp,aes(x=protein_start,y=SM.netgr.mean,color=exposed))+
#   geom_point()+
#   # facet_wrap(~exposed,nrow=2)+
#   # facet_grid(exposed~window.width)+
#   facet_wrap(~window.width,nrow=5)+
#   scale_y_continuous("Mean net growth rate at residue")+
#   scale_x_continuous("Residue on the ABL Kinase")+
#   labs(color="Solvent \nAccessibility")+cleanup
alternatingregion=bedata_simple%>%filter(Type%in%"ABE",protein_start%in%293)
alternatingregion_sgRNA=bedata_simple%>%filter(Type%in%"ABE",sgRNA.Seq%in%"GAGATCAAACACCCTAACCT")
ggplot(bedata_dssp%>%filter(window.width==1),aes(x=protein_start,y=SM.netgr.mean,color=exposed))+
  geom_point()+
  facet_wrap(~exposed,nrow=2)+
  # facet_grid(exposed~window.width)+
  # facet_wrap(~window.width,nrow=5)+
  scale_y_continuous("Mean BE LFC at residue")+
  scale_x_continuous("Residue on the ABL Kinase")+
  labs(color="Solvent \nAccessibility")+cleanup


# x=bedata_dssp%>%filter(window.width==1)
ggplot(bedata_dssp%>%filter(window.width==1),aes(x=protein_start,y=BE.LFC.mean))+
  geom_point()+
  # facet_wrap(~exposed,nrow=2)+
  # facet_grid(exposed~window.width)+
  # facet_wrap(~window.width,nrow=5)+
  scale_y_continuous("Mean BE LFC at residue")+
  scale_x_continuous("Residue on the ABL Kinase")+
  labs(color="Solvent \nAccessibility")+cleanup
# ggsave("output/BE_SM_Plots/be_slidingwindow_1.pdf",width=3,height=2,units="in",useDingbats=F)



bedata_byresidue_bywindowwidth=bedata_byresidue_bywindowwidth%>%
  mutate(BE.Significant=case_when(abs(BE.LFC.mean)>0.5~T,
                                  T~F),
         SM.Significant=case_when(SM.netgr.mean<0.04~T,
                                  T~F))%>%
  rowwise()%>%
  mutate(significance_status=case_when((BE.Significant%in%T)&&(SM.Significant%in%F)~"BEOnly",
                                       (BE.Significant%in%F)&&(SM.Significant%in%T)~"SMOnly",
                                       (BE.Significant%in%T)&&(SM.Significant%in%T)~"Both",
                                       T~"Neither"),
         species=protein_start)
bedata_dssp=bedata_dssp%>%
  mutate(BE.Significant=case_when(abs(BE.LFC.mean)>0.5~T,
                                  T~F),
         SM.Significant=case_when(SM.netgr.mean<0.04~T,
                                  T~F))%>%
  rowwise()%>%
  mutate(significance_status=case_when((BE.Significant%in%T)&&(SM.Significant%in%F)~"BEOnly",
                                       (BE.Significant%in%F)&&(SM.Significant%in%T)~"SMOnly",
                                       (BE.Significant%in%T)&&(SM.Significant%in%T)~"Both",
                                       T~"Neither"),
         species=protein_start)

bedata_dssp_sum=bedata_dssp%>%filter(!BE.LFC.mean%in%NaN)%>%group_by(window.width)%>%summarize(cor.be.dssp=cor(BE.LFC.mean,ACC),
                                                                 cor.be.sm=cor(BE.LFC.mean,SM.netgr.mean),
                                                                 cor.sm.dssp=cor(SM.netgr.mean,ACC))

# Figuring out the FET p-values for BE vs SM for all 
# fisher.test(contab_maker(bedata_dssp%>%filter(!BE.LFC.mean%in%NaN,window.width==4)))$p.value
x=rbind(fisher.test(contab_maker(bedata_dssp%>%filter(window.width==1)))$p.value,fisher.test(contab_maker(bedata_dssp%>%filter(window.width==2)))$p.value,fisher.test(contab_maker(bedata_dssp%>%filter(window.width==3)))$p.value,fisher.test(contab_maker(bedata_dssp%>%filter(window.width==4)))$p.value,fisher.test(contab_maker(bedata_dssp%>%filter(window.width==5)))$p.value)

bedata_dssp_sum$fet.pval.be.sm=x[,1]
# Next thing: plot by nucleotide, and not just residue
```

Plotting buried hydrophobic vs exposed regions
```{r}
bedata_dssp=bedata_dssp%>%mutate(netgr_score=(SM.netgr.mean-.03)/.03,
                                 lfc_score=(BE.LFC.mean/1.5)+1,
                                 acc_score=ACC/100)
ggplot(bedata_dssp%>%filter(protein_start%in%c(290:297),window.width==1),aes(x=protein_start))+geom_line(color="red",aes(y=netgr_score))+geom_point(aes(y=netgr_score))+geom_line(color="green",aes(y=acc_score))+geom_point(aes(y=acc_score))+geom_line(color="blue",aes(y=lfc_score))+geom_point(aes(y=lfc_score))
ggplot(bedata_dssp%>%filter(protein_start%in%c(264:274),window.width==1),aes(x=protein_start))+geom_line(color="red",aes(y=netgr_score))+geom_point(aes(y=netgr_score))+geom_line(color="green",aes(y=acc_score))+geom_point(aes(y=acc_score))+geom_line(color="blue",aes(y=lfc_score))+geom_point(aes(y=lfc_score))


library(reshape2)

bedata_dssp_melt=melt(bedata_dssp%>%filter(window.width%in%1),id.vars="protein_start" ,measure.vars = c("netgr_score","lfc_score","acc_score"),variable.name = "measurement_type",value.name = "score")

bedata_dssp_melt$measurement_type=factor(bedata_dssp_melt$measurement_type,levels=c("netgr_score","lfc_score","acc_score"))


ggplot(bedata_dssp_melt%>%filter(protein_start%in%c(290:297)),aes(x=protein_start,y=measurement_type,fill=score))+
  geom_tile()+
  # scale_fill_gradient2(low ="darkblue",midpoint=0.5, high ="red",name="Score")+
  scale_fill_gradient(low ="darkblue", high ="red",name="Score")+
  scale_y_discrete("",expand=c(0,0),labels=c("SM\ndata","BE\ndata","Solvent\nAccessibility"))+scale_x_continuous("",expand=c(0,0),breaks=c(290:297),labels=c("290","291","292","293","294","295","296","297"))+theme(legend.position = "none",axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),axis.text.y=element_blank(),axis.ticks.y=element_blank())
# ggsave("output/BE_SM_Plots/heatmap_forBEComparisons_290.pdf",width=2,height = 2,units = "in",useDingbats=F)
# ggsave("output/BE_SM_Plots/heatmap_forBEComparisons_290.pdf",useDingbats=F)
#######Making heatmap for IL3 data
smdata=read.csv("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/baf3_IL3_rep1vsrep2_ft/screen_comparison_baf3_IL3_low_rep1vsrep2_ft.csv",header = T)
smdata=smdata%>%rowwise%>%mutate(netgr_obs_mean=mean(netgr_obs_screen1,netgr_obs_screen2))
# smdata=smdata%>%dplyr::select(species,ref_aa,protein_start,alt_aa,netgr_obs_mean)
heatmap_plotting_function(smdata,290,297,fill_variable = "netgr_obs_mean",fill_name = "Net growth rate")+
  scale_x_continuous(name="",expand=c(0,0),breaks=c(290:297),labels=c("290","291","292","293","294","295","296","297"))+
  scale_fill_gradient(low ="darkblue", high ="red",name="Net growth rate")+theme(legend.position = "none",axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
# ggsave("output/BE_SM_Plots/heatmap_forlollipop_290.pdf",width=2,height = 4,units = "in",useDingbats=F)


ggplot(bedata_dssp_melt%>%filter(protein_start%in%c(264:274)),aes(x=protein_start,y=measurement_type,fill=score))+
  geom_tile()+
  # scale_fill_gradient2(low ="darkblue",midpoint=0.5, high ="red",name="Score")+
  scale_fill_gradient(low ="darkblue", high ="red",name="Score")+
  scale_y_discrete("",expand=c(0,0),labels=c("SM\ndata","BE\ndata","Solvent\nAccessibility"))+scale_x_continuous("",expand=c(0,0),breaks=c(264:274),labels=c("264","265","266","267","268","269","270","271","272","273","274"))+theme(legend.position = "none",axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),axis.text.y=element_blank(),axis.ticks.y=element_blank())
# ggsave("output/BE_SM_Plots/heatmap_forBEComparisons_264.pdf",width=2,height = 2,units = "in",useDingbats=F)
# ggsave("output/BE_SM_Plots/heatmap_forBEComparisons_290.pdf",useDingbats=F)
#######Making heatmap for IL3 data
smdata=read.csv("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/baf3_IL3_rep1vsrep2_ft/screen_comparison_baf3_IL3_low_rep1vsrep2_ft.csv",header = T)
smdata=smdata%>%rowwise%>%mutate(netgr_obs_mean=mean(netgr_obs_screen1,netgr_obs_screen2))
# smdata=smdata%>%dplyr::select(species,ref_aa,protein_start,alt_aa,netgr_obs_mean)
heatmap_plotting_function(smdata,263,274,fill_variable = "netgr_obs_mean",fill_name = "Net growth rate")+
  scale_x_continuous(name="",expand=c(0,0),breaks=c(264:274),labels=c("264","265","266","267","268","269","270","271","272","273","274"))+
  scale_fill_gradient(low ="darkblue", high ="red",name="Net growth rate")+theme(legend.position = "none",axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
# ggsave("output/BE_SM_Plots/heatmap_forlollipop_264.pdf",width=2.4,height = 4,units = "in",useDingbats=F)
```
SLIDING WINDOW ANALYSES
One way to look at the utility of BE data is looking at a residue-by-residue basis rather than a mutant by mutant basis
Attempting to do an ROC curve for which sliding window model is the best at estimating SM true positives
1. BE data doesn't work well on a residue-by-residue basis. Build an ROC curve that shows that SM data works much better at predicting exposed vs buried than BE data
2. Some BE data models work better than others... i.e. the two residue model works better than the 1 or 5 residue model at predicting SM true positives
3. The types of effects that all these models miss are sandwich effects where there are + and - effects next to each other. Presumably because if an sgRNA makes both + and - mutants, their combined effect is a neutral effect
```{r}
library(pROC)
bedata_roc=bedata_bymutant%>%filter(!synonymous%in%T,protein_start%in%c(242:322))
# Doing the ROC curve by mutant, not by residue
be_lfc=bedata_roc%>%dplyr::select(BE.LFC,BE.Significant,SM.netgr_obs_mean,SM.Significant)
be_lfc$BE.Significant=as.numeric(be_lfc$BE.Significant)
ggplot(be_lfc,aes(x=BE.LFC,y=BE.Significant))+geom_point()
ggplot(be_lfc,aes(x=BE.LFC,y=SM.Significant))+geom_point()
glm.fit=glm(as.numeric(be_lfc$SM.Significant)~be_lfc$BE.LFC,family=binomial)
be_lfc$glm_fits=glm.fit$fitted.values
# Note that the variable used for the threshold for significance is the BE LFC. 

ggplot(be_lfc,aes(x=BE.LFC,y=as.numeric(SM.Significant)))+geom_point()+geom_line(aes(x=BE.LFC,y=glm_fits))

par(pty="s")
roc(as.numeric(be_lfc$SM.Significant),be_lfc$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T)
par(pty="m")

# Doing the ROC curve by residue, not by mutant

# Assuming that each residue that is not seen by the BE data is a neutral residue
# bedata_byresidue_bywindowwidth2=bedata_byresidue_bywindowwidth
# bedata_byresidue_bywindowwidth=bedata_byresidue_bywindowwidth2
# bedata_byresidue_bywindowwidth=bedata_byresidue_bywindowwidth%>%dplyr::select(-BE.LFC.mean)%>%dplyr::rename(BE.LFC.mean=BE.LFC.weighted.mean)
bedata_roc=bedata_byresidue_bywindowwidth%>%
  # filter(window.width%in%1,!BE.LFC.mean%in%NaN)%>%
  filter(!BE.LFC.mean%in%NaN)%>%
  mutate(BE.LFC=BE.LFC.mean,
         SM.netgr_obs_mean=SM.netgr.mean,
         BE.LFC=case_when(BE.LFC%in%NaN~0,
                          T~BE.LFC))
be_lfc=bedata_roc%>%dplyr::select(window.width,BE.LFC,BE.Significant,SM.netgr_obs_mean,SM.Significant)
be_lfc.1=be_lfc%>%filter(window.width%in%1)
be_lfc.2=be_lfc%>%filter(window.width%in%2)
be_lfc.3=be_lfc%>%filter(window.width%in%3)
be_lfc.4=be_lfc%>%filter(window.width%in%4)
be_lfc.5=be_lfc%>%filter(window.width%in%5)

# be_lfc$BE.Significant=as.numeric(be_lfc$BE.Significant)
# ggplot(be_lfc,aes(x=BE.LFC,y=BE.Significant))+geom_point()
# ggplot(be_lfc,aes(x=BE.LFC,y=SM.Significant))+geom_point()

glm.fit.1=glm(as.numeric(be_lfc.1$SM.Significant)~be_lfc.1$BE.LFC,family=binomial)
glm.fit.2=glm(as.numeric(be_lfc.2$SM.Significant)~be_lfc.2$BE.LFC,family=binomial)
glm.fit.3=glm(as.numeric(be_lfc.3$SM.Significant)~be_lfc.3$BE.LFC,family=binomial)
glm.fit.4=glm(as.numeric(be_lfc.4$SM.Significant)~be_lfc.4$BE.LFC,family=binomial)
glm.fit.5=glm(as.numeric(be_lfc.5$SM.Significant)~be_lfc.5$BE.LFC,family=binomial)

be_lfc.1$glm_fits=glm.fit.1$fitted.values
be_lfc.2$glm_fits=glm.fit.2$fitted.values
be_lfc.3$glm_fits=glm.fit.3$fitted.values
be_lfc.4$glm_fits=glm.fit.4$fitted.values
be_lfc.5$glm_fits=glm.fit.5$fitted.values

ggplot(be_lfc.2,aes(x=BE.LFC,y=as.numeric(SM.Significant)))+geom_point()+geom_line(aes(x=BE.LFC,y=glm_fits))
# roc(as.numeric(be_lfc$SM.Significant),be_lfc$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T)
par(pty="s")
roc(as.numeric(be_lfc.1$SM.Significant),be_lfc.1$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#3937E3",lwd=4)
plot.roc(as.numeric(be_lfc.2$SM.Significant),be_lfc.2$glm_fits,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#378BE3",add=T,lwd=4,print.auc.y=45)
# plot.roc(as.numeric(be_lfc.3$SM.Significant),be_lfc.3$glm_fits,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#37E1E3",add=T,lwd=4,print.auc.y=40)
# plot.roc(as.numeric(be_lfc.4$SM.Significant),be_lfc.4$glm_fits,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#37E38F",add=T,lwd=4,print.auc.y=35)
plot.roc(as.numeric(be_lfc.5$SM.Significant),be_lfc.5$glm_fits,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#37E339",add=T,lwd=4,print.auc.y=40)
par(pty="m")
```
```{r}
# Demonstrating that the SM data works better at predicting residue exposure than the BE data
# bedata_roc=bedata_dssp%>%filter(window.width%in%1)

bedata_roc=bedata_dssp%>%
  # filter(window.width%in%1)%>%
  filter(window.width%in%1,!BE.LFC.mean%in%NaN)%>%
  mutate(BE.LFC=BE.LFC.mean,
         SM.netgr_obs_mean=SM.netgr.mean,
         BE.LFC=case_when(BE.LFC%in%NaN~0,
                          T~BE.LFC),
         SM.netgr_obs_mean=case_when(SM.netgr_obs_mean%in%NaN~0.05,
                          T~SM.netgr_obs_mean),
         DSSP.Buried=case_when(ACC>=50~F,
                               T~T))


# be_lfc$BE.Significant=as.numeric(be_lfc$BE.Significant)
# ggplot(be_lfc,aes(x=BE.LFC,y=BE.Significant))+geom_point()
# ggplot(be_lfc,aes(x=BE.LFC,y=SM.Significant))+geom_point()
# BE predicting DSSP:
be_lfc.be=bedata_roc%>%
  # filter(!BE.LFC%in%0)%>%
  dplyr::select(BE.LFC,BE.Significant,SM.netgr_obs_mean,SM.Significant,exposed,DSSP.Buried)
glm.fit.be=glm(as.numeric(be_lfc.be$DSSP.Buried)~be_lfc.be$BE.LFC,family=binomial)
be_lfc.be$glm_fits=glm.fit.be$fitted.values
ggplot(be_lfc.be,aes(x=BE.LFC,y=as.numeric(DSSP.Buried)))+geom_point()+geom_line(aes(x=BE.LFC,y=glm_fits))
roc(as.numeric(be_lfc.be$DSSP.Buried),be_lfc.be$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T)
# SM predicting DSSP:
be_lfc.sm=bedata_roc%>%dplyr::select(BE.LFC,BE.Significant,SM.netgr_obs_mean,SM.Significant,exposed,DSSP.Buried)
glm.fit.sm=glm(as.numeric(be_lfc.sm$DSSP.Buried)~be_lfc.sm$SM.netgr_obs_mean,family=binomial)
be_lfc.sm$glm_fits=glm.fit.sm$fitted.values
ggplot(be_lfc.sm,aes(x=SM.netgr_obs_mean,y=as.numeric(DSSP.Buried)))+geom_point()+geom_line(aes(x=SM.netgr_obs_mean,y=glm_fits))
roc(as.numeric(be_lfc.sm$DSSP.Buried),be_lfc.sm$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T)

par(pty="s")
roc(as.numeric(be_lfc.be$DSSP.Buried),be_lfc.be$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#3937E3",lwd=4)
plot.roc(as.numeric(be_lfc.sm$DSSP.Buried),be_lfc.sm$glm_fits,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#378BE3",add=T,lwd=4,print.auc.y=45)
par(pty="m")
# rocdata=roc(as.numeric(be_lfc$DSSP.Buried),be_lfc$glm_fits)
# rocdata_df=data.frame(rocdata)
# plot(1-rocdata$specificities,rocdata$sensitivities)
# ggplot(rocdata,aes(x=1-speci))
```



Making lollipop plot with heatmap overlayed underneath it
```{r}
bedata_bymutant2=bedata_bymutant%>%filter(protein_start%in%c(242:322),Type%in%"ABE",synonymous%in%F)
bedata_bymutant2$significance_status=factor(bedata_bymutant2$significance_status,levels=c("Neither","Both","SMOnly","BEOnly"))
source("code/plotting/heatmap_plotting_function.R")
ggplot(bedata_byresidue_bywindowwidth%>%filter(window.width%in%1),aes(x=protein_start,y=BE.LFC.mean))+
  geom_point()+geom_segment( aes(x=protein_start, xend=protein_start, y=-.5, yend=BE.LFC.mean))+
  scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,322),expand=c(0,0))+
  scale_y_continuous(name="BE LFC")+
  theme(panel.grid.major = element_blank(),
        panel.grid.minor.x = element_blank())
# ggsave("output/BE_SM_Plots/lollipop.pdf",width=6,height = 2,units = "in",useDingbats=F)


# ggplot(bedata_byresidue_bywindowwidth%>%filter(window.width%in%1),aes(x=protein_start,y=SM.netgr.mean))+
#   geom_point()+geom_segment( aes(x=protein_start, xend=protein_start, y=.045, yend=SM.netgr.mean))

#######Making heatmap for IL3 data
smdata=read.csv("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/baf3_IL3_rep1vsrep2_ft/screen_comparison_baf3_IL3_low_rep1vsrep2_ft.csv",header = T)
smdata=smdata%>%rowwise%>%mutate(netgr_obs_mean=mean(netgr_obs_screen1,netgr_obs_screen2))
# smdata=smdata%>%dplyr::select(species,ref_aa,protein_start,alt_aa,netgr_obs_mean)
heatmap_plotting_function(smdata,242,321,fill_variable = "netgr_obs_mean",fill_name = "Net growth rate")+
  scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,322),expand=c(0,0))+
  scale_fill_gradient(low ="darkblue", high ="red",name="Net growth rate")
# ggsave("output/BE_SM_Plots/heatmap_forlollipop.pdf",width=6,height = 4,units = "in",useDingbats=F)

#######Making heatmap for Imatinib data
smdata=read.csv("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/k562_imat_high_rep1vsrep2/screen_comparison_k562_imat_high_rep1vsrep2.csv",header = T)
smdata=smdata%>%rowwise%>%mutate(netgr_obs_mean=mean(netgr_obs_screen1,netgr_obs_screen2))
# smdata=smdata%>%dplyr::select(species,ref_aa,protein_start,alt_aa,netgr_obs_mean)
heatmap_plotting_function(smdata,242,321,fill_variable = "netgr_obs_mean",fill_name = "Net growth rate")+
  scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,322),expand=c(0,0))+
  # scale_fill_gradient(low ="darkblue", high ="red",name="Net growth rate")+
  theme(legend.position = "none")
# ggsave("output/BE_SM_Plots/heatmap_forlollipop.pdf",width=6,height = 4,units = "in",useDingbats=F)
#######


# heatmap_plotting_function(smdata,242,321,fill_variable = "netgr_obs_mean",fill_name = "Net growth rate")+scale_fill_gradient2(low ="darkblue",mid="white",midpoint = .045, high ="red")

# To merge plots, you can use the code in the following URL: https://felixfan.github.io/stacking-plots-same-x/

# Correlation Plots for the BE data and the SM data:
ggplot(smdata,aes(x=netgr_obs_screen1,y=netgr_obs_screen2))+
  geom_point()+
  geom_abline()+
  geom_hline(yintercept = .065,linetype="dashed")+
  geom_vline(xintercept = .065,linetype="dashed")+
  scale_x_continuous("SM Net growth rate R1")+
  scale_y_continuous("SM Net growth rate R2")+
  stat_cor(method = "pearson")+cleanup
# ggsave("output/BE_SM_Plots/correlation_sm.pdf",width=3,height = 3,units = "in",useDingbats=F)

# ABL1_C6 and g11 are plasmid baseline
# I63Baf3ABL14 are two replicates after IL3 removal
# 
# Correlation plot for BE vs SM
ggplot(bedata_simple%>%filter(protein_start%in%c(sm_start:sm_end)),aes(x=netgr_obs_mean,y=BE.LFC))+geom_point()
ggplot(bedata_bymutant%>%filter(!synonymous%in%T,protein_start%in%c(sm_start:sm_end)),aes(y=SM.netgr_obs_mean,x=BE.LFC,label=species))+
  geom_text()+
  facet_wrap(~Type)+
  stat_cor(method = "pearson")+
  scale_y_continuous("SM net growth rate")+
  scale_x_continuous("BE LFC")+
  geom_hline(yintercept = .065,linetype="dashed")
# ggsave("output/BE_SM_Plots/correlation_smvsbe_abecbe.pdf",width=8,height = 4,units = "in",useDingbats=F)

ggplot(bedata_bymutant2,aes(y=SM.netgr_obs_mean,x=BE.LFC))+
  geom_point()+
  # facet_wrap(~Type)+
  stat_cor(method = "pearson")+
  scale_y_continuous("SM net growth rate")+
  scale_x_continuous("BE LFC")+
  geom_hline(yintercept = .065,linetype="dashed")+cleanup
# ggsave("output/BE_SM_Plots/correlation_smvsbe.pdf",width=3,height = 3,units = "in",useDingbats=F)

ggplot(bedata_bymutant2,aes(y=SM.netgr_obs_mean,x=BE.LFC,label=species,color=significance_status))+
  # stat_cor(method = "pearson")+
  geom_text(size=3)+
  # geom_hline(yintercept = .065,linetype="dashed")
  # facet_wrap(~Type)+
  scale_y_continuous("SM net growth rate")+
  scale_x_continuous("BE LFC")+cleanup+theme(legend.position = "none")+
  scale_color_manual(values=c("#00BFC4","#7CAE00","#C77CFF","#F8766D"))
# ggsave("output/BE_SM_Plots/correlation_smvsbe_colored.pdf",width=3,height = 3,units = "in",useDingbats=F)
# nrow(bedata_bymutant%>%filter(protein_start%in%c(242:322),Type%in%"ABE",significance_status%in%"SMOnly"))
# xx=bedata_bymutant%>%filter(protein_start%in%c(242:322),Type%in%"ABE",significance_status%in%"BEOnly")
  # source("code/plotting/cleanup.R")
ggplot(bedata_bymutant2,aes(y=sgRNA.SM.Mean.Netgr,x=BE.LFC,label=species,color=significance_status))+
  # stat_cor(method = "pearson")+
  geom_text(size=3)+
  # geom_hline(yintercept = .065,linetype="dashed")
  # facet_wrap(~Type)+
  scale_y_continuous("SM net growth rate")+
  scale_x_continuous("BE LFC")+cleanup+theme(legend.position = "none")+
  scale_color_manual(values=c("#00BFC4","#7CAE00","#C77CFF","#F8766D"))
# ggsave("output/BE_SM_Plots/correlation_smvsbe_colored_corrected.pdf",width= 3,height = 3,units = "in",useDingbats=F)


ggplot(bedata_bymutant2,aes(x=significance_status,y=sgRNAs_per_mutant,fill=significance_status))+geom_boxplot()+
  geom_jitter(color="black", size=1,width=.1, alpha=0.9)+
  scale_x_discrete("Depletion status",labels=c("Neither","Both","SM \nOnly","BE \nOnly"))+
  scale_y_continuous("sgRNAs per mutant",breaks=c(1:7))+
  cleanup+
  theme(legend.position = "none")+
  scale_fill_manual(values=c("#00BFC4","#7CAE00","#C77CFF","#F8766D"))
# ggsave("output/BE_SM_Plots/sgRNAspermutant_boxplot.pdf",width=4,height = 3,units = "in",useDingbats=F)


ggplot(bedata_bymutant2,aes(x=significance_status,y=sgRNA.SM.Mean.Netgr,fill=significance_status))+geom_boxplot()+
  geom_jitter(color="black", size=1,width=.1, alpha=0.9)+
  scale_x_discrete("Depletion status",labels=c("Neither","Both","SM \nOnly","BE \nOnly"))+
  scale_y_continuous("Growth Rate of Mutant Admixture",breaks=c(1:7))+
  cleanup+
  theme(legend.position = "none")+
  scale_fill_manual(values=c("#00BFC4","#7CAE00","#C77CFF","#F8766D"))
# ggsave("output/BE_SM_Plots/admixture_boxplot.pdf",width=4,height = 3,units = "in",useDingbats=F)

ggplot(bedata_bymutant2%>%filter(significance_status%in%c("Both","SMOnly")),aes(x=significance_status,y=sgRNA.SM.Mean.Netgr,fill=significance_status))+geom_boxplot()+
  geom_jitter(color="black", size=1,width=.1, alpha=0.9)+
  scale_x_discrete("Depletion status",labels=c("Both","SM \nOnly"))+
  scale_y_continuous("Growth Rate of Mutant Admixture",breaks=c(1:7))+
  cleanup+
  theme(legend.position = "none")+
  scale_fill_manual(values=c("#7CAE00","#C77CFF"))
ggsave("output/BE_SM_Plots/admixture_boxplot_depleting.pdf",width=2,height = 3,units = "in",useDingbats=F)

ggplot(bedata_bymutant2%>%filter(significance_status%in%c("Neither","BEOnly")),aes(x=significance_status,y=sgRNA.SM.Mean.Netgr,fill=significance_status))+geom_boxplot()+
  geom_jitter(color="black", size=1,width=.1, alpha=0.9)+
  scale_x_discrete("Depletion status",labels=c("Neither","BE \nOnly"))+
  scale_y_continuous("Growth Rate of Mutant Admixture",breaks=c(1:7))+
  cleanup+
  theme(legend.position = "none")+
  scale_fill_manual(values=c("#00BFC4","#F8766D"))
# ggsave("output/BE_SM_Plots/admixture_boxplot_neutral.pdf",width=2,height = 3,units = "in",useDingbats=F)

########Making Pie Charts that show that there are a lot of neutral mutants present in the purple region
blank_theme <- theme_minimal()+
  theme(
    legend.position="none",
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  panel.border = element_blank(),
  panel.grid=element_blank(),
  axis.ticks = element_blank(),
  plot.title=element_text(size=14, face="bold")
  )


# Note that I got the numbers below manually.. I.e. I looked at each mutant in the green and purple region of the plot and figured out how many also had mutants that wre potentially neutral
df_forpiecharts=data.frame(c("Both","Both","SMOnly","SMOnly"),c("neutralpresent","noneutral","neutralpresent","noneutral"),cbind(c(4,14,9,1)))
colnames(df_forpiecharts)=c("significance_status","neutral_status","n")

ggplot(df_forpiecharts%>%filter(significance_status%in%"Both"),aes(y=n,x="",fill=neutral_status))+geom_bar(stat="identity")+ coord_polar("y", start=0)+scale_fill_manual(values=c("#AEF3AB", "#7CAE00"))+blank_theme+theme(axis.text.x=element_blank())
# ggsave("output/BE_SM_Plots/both_piechart.pdf",width=1,height=1,units = "in",useDingbats=F)

ggplot(df_forpiecharts%>%filter(significance_status%in%"SMOnly"),aes(y=n,x="",fill=neutral_status))+geom_bar(stat="identity")+ coord_polar("y", start=0)+scale_fill_manual(values=c("#DDB0FF", "#C77CFF"))+blank_theme+theme(axis.text.x=element_blank())
# ggsave("output/BE_SM_Plots/smonly_piechart.pdf",width=1,height=1,units = "in",useDingbats=F)

t.test(bedata_bymutant2[bedata_bymutant2$significance_status%in%"Both","sgRNA.SM.Mean.Netgr"],
       bedata_bymutant2[bedata_bymutant2$significance_status%in%"SMOnly","sgRNA.SM.Mean.Netgr"])

t.test(bedata_bymutant2[bedata_bymutant2$significance_status%in%"Neither","sgRNA.SM.Mean.Netgr"],
       bedata_bymutant2[bedata_bymutant2$significance_status%in%"BEOnly","sgRNA.SM.Mean.Netgr"])

# x=bedata_bymutant2[bedata_bymutant2$significance_status%in%"Both","sgRNA.SM.Mean.Netgr"]
# x=bedata_bymutant2%>%filter(significance_status%in%"Both")
```

```{r}
# Trying to plot out correlations for the BE data:
bematrix=read.table("data/BE_ABL_Merged/BCRABL_ScreenMatrixE2_20220920BaF3.txt",sep="\t",header = T,stringsAsFactors = F)
# D and E are ABE and F and G are CBE.
bematrix=bematrix%>%dplyr::select(sgName,gene,rep1_before=ABL1_C6,rep2_before=ABL1_G11,rep1_after=I63DBaF3ABL14,rep2_after=I63EBaF3ABL14)
bematrix=bematrix%>%mutate(BE.LFC.rep1=log2(rep1_after/rep1_before),BE.LFC.rep2=log2(rep2_after/rep2_before))
ggplot(bematrix%>%filter(BE.LFC.rep1<=0,BE.LFC.rep2<=0),aes(x=BE.LFC.rep1,y=BE.LFC.rep2))+geom_point()+geom_abline()+stat_cor(method = "pearson")+scale_x_continuous("BE LFC Replicate 1")+scale_y_continuous("BE LFC Replicate 2")+cleanup
# ggsave("output/BE_SM_Plots/correlation_be.pdf",width=3,height = 3,units = "in",useDingbats=F)
```

Looking at net growth rates in the old IL3 data and in the new IL3 data to ask the question: When one looks at the same SM residues with the old data what is different? Ie how much does SM quality affect the conclusions.
Answer:
When one looks at the same SM residues with the old data what is different? Ie how much does SM quality affect the conclusions.
All mutants that scored as deleterious by SM now also scored as deleterious by SM in our old screen (mutants in the neither and SMonly case). There is a mutant in the BEOnly data that scored by SM before, but isnt scoring by SM now (F283L, which is in a buried residue, but we had good counts on it before and after the screen).

```{r,include=F,eval=F}
# Both
c("E255G", "F283S", "F311S", "I313T", "K271E", "K271R", "K274E", "L248P", "Q252R", "T272A", "V270A",
"Y253C", "Y257C", "Y312H")
# BEOnly
c("E258G","F283L","F311L","K274R","L284L","T306A","Y312Y")
# SMOnly
c("C305R","E286G","E316G","I293T","I313V","I314T","K247E","M278V","M318T","Y312C")
oldil3=read.csv("output/ABLEnrichmentScreens/IL3_Enrichment_bgmerged_2.20.23.csv")
x=oldil3%>%filter(species%in%c("C305R","E286G","E316G","I293T","I313V","I314T","K247E","M278V","M318T","Y312C"))
x=oldil3%>%filter(species%in%c("E258G","F283L","F311L","K274R","L284L","T306A","Y312Y"))

smdata=read.csv("output/ABLEnrichmentScreens/ABL_Region1_Lane18_Comparisons/cross-replicate/baf3_IL3_rep1vsrep2_ft/screen_comparison_baf3_IL3_low_rep1vsrep2_ft.csv",header = T)
# smdata=read.csv("output/ABLEnrichmentScreens/IL3_Enrichment_bgmerged_2.20.23.csv",header = T)
smdata=smdata%>%rowwise%>%mutate(netgr_obs_mean=mean(netgr_obs_screen1,netgr_obs_screen2))
x=smdata%>%filter(species%in%c("E258G","F283L","F311L","K274R","L284L","T306A","Y312Y"))

# Choosing a set of guides for Ivan to make and sequence at baseline and after sequencing
# Trying to choose 1 guide per mutant. Trying not to choose mutants that are too close together
# Ended up removing I313V because I314T is pretty close to it
# One concern is that the sgRNA for Y312C also makes I313V with low probability. Cheeck that this does not overlap with I314
purple=bedata_simple%>%filter(species%in%c("C305R","E286G","E316G","I293T","I313V","I314T","K247E","M278V","M318T","Y312C"),
                         sgRNA_Nuc%in%c(2,3,4,5,6,7,8,9),
                         !sgRNA.Seq%in%c("CTTGAAAGAAGCTGCAGTCA","AGAAGCTGCAGTCATGAAAG","CACTGAGTTCATGACCTACG","TCACTGAGTTCATGACCTAC","TGTTTGATCTCTTTCATGAC","CTATATCATCACTGAGTTCA","TGAAGCACAAGCTGGGCGGG","CATGGAGGTGGAAGAGTTCT","CTATATCATCACTGAGTTCA","ACTCAGTGATGATATAGAAC","TCAGTGATGATATAGAACGG"))%>%mutate(significance_status="SMOnly")

green=bedata_simple%>%filter(species%in%c("F283S", "K271E", "K271R", "K274E", "L248P", "Q252R", "T272A", "V270A",
"Y253C", "Y257C"),
                         sgRNA_Nuc%in%c(2,3,4,5,6,7,8,9),
                         !sgRNA.Seq%in%c("GCCGTGAAGACCTTGAAGGA","GGAGGTGTACGAGGGCGTGT","GAACTCTTCCACCTCCATGG","GACCTTGAAGGAGGACACCA","ACCTTGAAGGAGGACACCAT","GAAGGAGGACACCATGGAGG","TTGAAGGAGGACACCATGGA","CCAGTACGGGGAGGTGTACG","CAGTACGGGGAGGTGTACGA","CGGGGGCCAGTACGGGGAGG","AGGTCTTCACGGCCACCGTC","TACGGGGAGGTGTACGAGGG","AGTACGGGGAGGTGTACGAG"))%>%mutate(significance_status="Both")

red=bedata_simple%>%filter(species%in%c("K274R","T306A","E258G","F283L","F311L"),
                         sgRNA_Nuc%in%c(2,3,4,5,6,7,8,9),
                         !sgRNA.Seq%in%c("AAGGAGGACACCATGGAGGT","GAAGGAGGACACCATGGAGG","GACCTTGAAGGAGGACACCA","TTGAAGGAGGACACCATGGA","ACCTTGAAGGAGGACACCAT","ATATAGAACGGGGGCTCCCG
","GAACGGGGGCTCCCGGGTGC","GATATAGAACGGGGGCTCCC","ATATAGAACGGGGGCTCCCG","GAACTCTTCCACCTCCATGG","AAGAACTCTTCCACCTCCAT"))%>%mutate(significance_status="BEOnly")

blue=bedata_simple%>%filter(species%in%c("Y264C","E279G","Y320C"),
                         sgRNA_Nuc%in%c(2,3,4,5,6,7,8,9),
                         !sgRNA.Seq%in%c("GAAGAAATACAGCCTGACGG","ATACAGCCTGACGGTGGCCG","AAGAAATACAGCCTGACGGT","TGGAAGAAATACAGCCTGAC","CATGGAGGTGGAAGAGTTCT","CATGACCTACGGGAACCTCC"))%>%mutate(significance_status="Neither")

guides_for_verification=rbind(purple,green,red,blue)%>%dplyr::select(PAM,significance_status,"species",protein_start,SM.netgr_obs_mean=netgr_obs_mean,BE.LFC,distance,mutants_per_sgRNA)
# write.csv(guides_for_verification,"output/BE_SM_Plots/sgRNAs_for_verification.csv")


y=bedata_simple%>%filter(sgRNA.Seq%in%c("GCACAAGCTGGGCGGGGGCC",""),Type%in%"ABE")
y=bedata_simple%>%filter(sgRNA.Seq%in%c("CATGAAGCACAAGCTGGGCG",""),Type%in%"ABE")

y=bedata_simple%>%filter(species%in%c("K247E",""),Type%in%"ABE")
y=smdata%>%filter(species%in%c("K245G","K245R","K245E"))
# y=bedata_simple%>%filter(species%in%c("Y320C"),Type%in%"ABE")
# x=bedata_bymutant%>%rowwise()%>%mutate(protein_start=as.numeric(substr(species,2,4)))
# length(unique(guides_for_verification$sgRNA.Seq))
# Blue:
# Y264C
# T267A
# Red:
# 274
# 306
# 311
# 
# Green:
# 252/253
# 257
# 283
# 271
# 274
# 
# Purple:
# 247
# 278
# 286
# 293
# 305
# 312
# 316
# 318

```


```{r,eval=F,include=F}
bedata_bymutant%>%filter(significance_status%in%"Both")
bedata_bymutant[bedata_bymutant$significance_status%in%"Both","species"]
bedata_bymutant[bedata_bymutant$significance_status%in%"BEOnly","species"]
bedata_bymutant[bedata_bymutant$significance_status%in%"SMOnly","species"]
greenmutants$species
# Both
c("E255G", "F283S", "F311S", "I313T", "K271E", "K271R", "K274E", "L248P", "Q252R", "T272A", "V270A",
"Y253C", "Y257C", "Y312H")
# BEOnly
c("E258G","F283L","F311L","K274R","L284L","T306A","Y312Y")
# SMOnly
c("C305R","E286G","E316G","I293T","I313V","I314T","K247E","M278V","M318T","Y312C")

# K247E=GCACAAGCTGGGCGGGGGCC
# Y312C=CTATATCATCACTGAGTTCA and CCCGTTCTATATCATCACTG
# I313V=CTATATCATCACTGAGTTCA
# C305R=GGGTGCAGACCCCAAGGAGC
# E286G=AGAAGCTGCAGTCATGAAAG and AAAGAAGCTGCAGTCATGAA
# E316G= CACTGAGTTCATGACCTACG and TCACTGAGTTCATGACCTAC and ATCACTGAGTTCATGACCTA
# I293T= TTGATCTCTTTCATGACTGC and TGTTTGATCTCTTTCATGAC
# I314T= TCAGTGATGATATAGAACGG and ACTCAGTGATGATATAGAAC
# M278V= CATGGAGGTGGAAGAGTTCT and GGACACCATGGAGGTGGAAG and GAGGACACCATGGAGGTGGA
# M316T= 
  
  
  # F283S= AAGAACTCTTCCACCTCCAT
# I313T= TGATATAGAACGGGGGCTCC
# I314T= TCAGTGATGATATAGAACGG and ACTCAGTGATGATATAGAAC
# K271E= GCCGTGAAGACCTTGAAGGA and GGCCGTGAAGACCTTGAAGG
# L248P= GCCCAGCTTGTGCTTCATGG
# Y257C= GGTGTACGAGGGCGTGTGGA

# E255G= CGGGGAGGTGTACGAGGGCG and GGGAGGTGTACGAGGGCGTG and TACGGGGAGGTGTACGAGGG
# Y312H= GATATAGAACGGGGGCTCCC and TGATATAGAACGGGGGCTCC and ATATAGAACGGGGGCTCCCG and ATAGAACGGGGGCTCCCGGG
# F311S=ATATAGAACGGGGGCTCCCG and ATAGAACGGGGGCTCCCGGG and GATATAGAACGGGGGCTCCC
# T272A=GCCGTGAAGACCTTGAAGGA and GACCTTGAAGGAGGACACCA
# Y253C= CCAGTACGGGGAGGTGTACG and CAGTACGGGGAGGTGTACGA and GGCCAGTACGGGGAGGTGTA and AGTACGGGGAGGTGTACGAG

x=bedata_simple%>%filter(species%in%"Y312H")
x=bedata_simple%>%filter(Type%in%"ABE",sgRNA.Seq%in%c("AGAAGCTGCAGTCATGAAAG",
                                                      "AAAGAAGCTGCAGTCATGAA"))
```


Looking at individual mutants in the red green and purple zones
```{r, eval=F, include=F}

ggplot(bedata_bymutant,aes(y=SM.netgr_obs_mean,x=BE.LFC,label=species))+geom_text()
ggplot(bedata_bymutant,aes(y=sgRNA.SM.Mean.Netgr,x=BE.LFC,label=species))+geom_text()
ggplot(bedata_bymutant,aes(y=sgRNA.SM.Mean.Netgr,x=BE.LFC,label=species))+geom_text()
purple=bedata_bymutant%>%filter(BE.LFC>=-.75,SM.netgr_obs_mean<.04)
mean(purple$num_sgRNAs)
green=bedata_bymutant%>%filter(BE.LFC<=-.75,SM.netgr_obs_mean<.04)
mean(green$num_sgRNAs)
red=bedata_bymutant%>%filter(BE.LFC>=-.75,SM.netgr_obs_mean>.04)
mean(green$num_sgRNAs)
cor(bedata_bymutant$BE.LFC,bedata_bymutant$SM.netgr_obs_mean)
cor(bedata_bymutant$BE.LFC,bedata_bymutant$sgRNA.SM.Mean.Netgr)
y=bedata_bymutant%>%filter(!species%in%c("C305R","I313V","K247E","Y312C","E286G","E316G","I293T"))
cor(y$BE.LFC,y$SM.netgr_obs_mean)
purple_sgRNAs=bedata_outer%>%filter(Mutation%in%c("C305R","I313V","K247E","Y312C","E286G","E316G","I293T"))
purple_sgRNAs=bedata_outer%>%filter(species%in%c("I313T"))

# K247E=GCACAAGCTGGGCGGGGGCC
# Y312C=CTATATCATCACTGAGTTCA and CCCGTTCTATATCATCACTG
# I313V=CTATATCATCACTGAGTTCA
# C305R=GGGTGCAGACCCCAAGGAGC
# E286G=AGAAGCTGCAGTCATGAAAG and AAAGAAGCTGCAGTCATGAA
# E316G= CACTGAGTTCATGACCTACG and TCACTGAGTTCATGACCTAC and ATCACTGAGTTCATGACCTA
# I293T= TTGATCTCTTTCATGACTGC and TGTTTGATCTCTTTCATGAC
x=bedata_simple%>%filter(sgRNA.Seq%in%c("ATCACTGAGTTCATGACCTA"))
x=bedata_outer%>%filter(sgRNA.Seq%in%c("TGTTTGATCTCTTTCATGAC"))


green_sgRNAs=bedata_outer%>%filter(species%in%c("F283S","I313T","I314T","K271E","L248P","","Y257C"))
green_sgRNAs=bedata_outer%>%filter(species%in%c("Y257C"))
F283S= AAGAACTCTTCCACCTCCAT
# I313T= TGATATAGAACGGGGGCTCC
# I314T= TCAGTGATGATATAGAACGG and ACTCAGTGATGATATAGAAC
# K271E= GCCGTGAAGACCTTGAAGGA and GGCCGTGAAGACCTTGAAGG
# L248P= GCCCAGCTTGTGCTTCATGG
# Y257C= GGTGTACGAGGGCGTGTGGA
x=bedata_simple%>%filter(!Type%in%"CBE",sgRNA.Seq%in%c("ACTCAGTGATGATATAGAAC"))
x=bedata_outer%>%filter(!Type%in%"CBE",sgRNA.Seq%in%c("GCCGTGAAGACCTTGAAGGA"))

# bedata_inner_simple=merge(bedata_inner_simple,x_sum,by="species")
# x_sum=x_sum%>%
#   mutate(BE.Significant=case_when(abs(BE.LFC)>0.5~T,
#                                   T~F),
#          SM.Significant=case_when(SM.padj<0.05~T,
#                                   T~F))%>%
#   rowwise()%>%
#   mutate(significance_status=case_when((BE.Significant%in%T)&&(SM.Significant%in%F)~"BEOnly",
#                                        (BE.Significant%in%F)&&(SM.Significant%in%T)~"SMOnly",
#                                        (BE.Significant%in%T)&&(SM.Significant%in%T)~"Both",
#                                        T~"Neither"))

# bedata_inner_simple_filtered=bedata_inner_simple%>%rowwise()%>%filter(species%in%species.mindist)

```

Exactly what % of the mutants by BE are double mutants?
```{r,eval=F,include=F}
be_duplex=read.table("data/Consensus_Data/novogene_lane16b/Sample8_combined/duplex/duplex_sorted_filtered.tsv",header = F)
colnames(be_duplex)=c("pos","cigar","tlen","seq","mdz")
be_duplex=be_duplex%>%mutate(mdz=gsub("MD:Z:","",mdz))
be_duplex=be_duplex%>%filter(!mdz%in%c("133","AS:i:0","XS:i:0"))
be_duplex=be_duplex%>%mutate(n_nuc=str_count(mdz,"A|T|C|G"))

be_duplex_sum=be_duplex%>%filter(!n_nuc%in%0)%>%group_by(n_nuc)%>%summarize(ct=n())
be_duplex2=be_duplex%>%filter(n_nuc%in%c(2,3))
be_duplex2=be_duplex%>%filter(n_nuc%in%c(1),!grepl("N",seq)%in%T)%>%mutate(alt_nuc=gsub("\\d+", "", mdz))
be_duplex2_sum=be_duplex2%>%group_by(alt_nuc)%>%summarize(ct=n())
# Out of 5021 reads with mutations, 3796 have 1 mutant, and 1225 have more than 1 mutant:
# 1.  3796
# 2.  860
# 3.  248
# 4.  98
# 5.  18
# 6.  1
# The majority of the edits seem to be happening on the A or T nucleotides, which makes sense for ABEs. However a good portion also seems to happen on C and G nucleotidess
# However, if I take out reads that have Ns in them, then there is an even spread of ATGC nucleotides? v confusing.
```



