---
title: "ABL_cosmic_analysis_fullkinase"
author: "Haider Inam"
date: '2024-04-28'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# knitr::opts_knit$set(root.dir = normalizePath(".."))
# rm(list=ls())

library(stringr)
library(dplyr)
library(ggplot2)
library(plotly)
library(tictoc)
library(doParallel)
library(foreach)
library(RColorBrewer)
library("ggrepel")
library(reshape2)
#Cleanup code for plotting
# cleanup=theme_bw() +
#   theme(plot.title = element_text(hjust=.5),
#         panel.grid.major = element_blank(),
#         panel.grid.major.y = element_blank(),
#         panel.background = element_blank(),
#         # axis.line = element_line(color = "black"),
#         axis.text = element_text(face="bold",color="black",size="11"),
#         text=element_text(size=11,face="bold"),
#         axis.title=element_text(face="bold",size="11"))

```
For the density plots:
```{r}
library(viridisLite)
library("MASS")

theme_set(theme_bw(base_size = 16))

# Get density of points in 2 dimensions.
# @param x A numeric vector.
# @param y A numeric vector.
# @param n Create a square n by n grid to compute density.
# @return The density within each square.
get_density <- function(x, y, ...) {
  dens <- MASS::kde2d(x, y, ...)
  ix <- findInterval(x, dens$x)
  iy <- findInterval(y, dens$y)
  ii <- cbind(ix, iy)
  return(dens$z[ii])
}

set.seed(1)
dat <- data.frame(
  x = c(
    rnorm(1e4, mean = 0, sd = 0.1),
    rnorm(1e3, mean = 0, sd = 0.1)
  ),
  y = c(
    rnorm(1e4, mean = 0, sd = 0.1),
    rnorm(1e3, mean = 0.1, sd = 0.2)
  )
)
```

ABL_cosmic_analysis had a comparison of sanger data vs pooled screening data FOR REGION 1 ONLY

This chunk of code is pretty similar to ABL_cosmic_analysis, except that the comparisons are done across the full ABL kinase (residues 242-512)

Seeing how well our imatinib resistance mutants work with cosmic (sanger institute) mutants

Unseen mutants: This code also checks whether single nucleotide substitution mutants that are strong hits for imatinib resistance in our screens are not seen in the Sanger database because they are just low probability mutants. Short answer: we don't see a lot of evidence that resistant mutants are unseen in the sanger database because of their low probability
```{r}
# source("code")
source("code/res_residues_adder.R")
source("code/variants_parser.R")
source("code/compare_samples.R")
source("code/resmuts_adder.R")
source("code/cosmic_data_adder.R")
source("code/plotting/cleanup.R")
source("code/plotting/heatmap_plotting_function.R")

```


Checking region 2 3 4 plots

```{r}
#######################Inputting Data#######################
data=read.csv("output/ABLEnrichmentScreens/ABL_Region234_Comparisons/ABLfullkinase_allconditions_growthrates_1.2.23.csv",header = T,stringsAsFactors = F)
data=data%>%rowwise()%>%mutate(netgr_mean=mean(netgr.imat.high.1,netgr.imat.high.1),
                               score_mean=mean(score.imat.high.1,score.imat.high.1))
# class(data)
data=resmuts_adder(data)
data=res_residues_adder(data)
data=cosmic_data_adder(data)
# data=data%>%rowwise()%>%mutate(netgr_mean=mean(netgr_obs.x,netgr_obs.y))
# data=data%>%rowwise()%>%mutate(netgr_mean=netgr_obs_mean)
data$resmut_cosmic="neither"
data[data$cosmic_present%in%TRUE,"resmut_cosmic"]="cosmic"
data[data$resmuts%in%TRUE,"resmut_cosmic"]="resmut"
data$resmut_cosmic=factor(data$resmut_cosmic,levels=c("neither","cosmic","resmut"))

#######################Waterfall plots#######################

ggplot(data%>%filter(protein_start>=242,protein_start<=494,n_nuc_min%in%1),aes(x=reorder(species,-score_mean),y=score_mean,fill=resmut_cosmic))+geom_col()+theme(axis.text.x=element_text(angle=90, hjust=1))+scale_y_continuous(name="Enrichcment Score")+scale_x_discrete(name="Mutant")+guides(fill = guide_legend(title = "Clinically\n Observed \n Resmut"))+scale_fill_manual(values=c("gray","orange","red"))


ggplot(data%>%filter(protein_start>=242,protein_start<=494,n_nuc_min%in%1,score_mean>=0),aes(x=reorder(species,-netgr_mean),y=netgr_mean,fill=resmut_cosmic))+geom_col()+
  scale_y_continuous(name=bquote('Net Growth Rate '(Hours^-1)))+
  scale_x_discrete(name="Mutant")+
  scale_fill_manual(name="Resistance Status",labels=c("Never Seen","Sanger Mutant","Known Resistant Mutant"),values=c("gray","orange","red"))+
  theme(axis.text.x=element_text(angle=90, hjust=1),
        axis.text=element_text(size=6),
        panel.grid.major = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.background = element_blank())+
  theme(legend.position = "none")

# ggsave("output/ABLEnrichmentScreens/ABL_Region234_Comparisons/sm_imatinib_plots/ImatinibEnrichment_Netgr_Distribution_zoom.pdf",width=8,height=4,units="in",useDingbats=F)

#######################Scatter Plots#######################


# Scatterplot with all mutants in the library
ggplot(data%>%filter(!species%in%c("V299L"),protein_start>=242,protein_start<=494)%>%arrange(resmuts),aes(x=netgr.imat.high.1,y=netgr.imat.high.2))+
  geom_point(color="black",shape=21,aes(fill=factor(resmut_cosmic)))+
  # geom_text_repel(aes(label=ifelse(resmuts%in%T,species,"")))+
  geom_label_repel(size=2,aes(label=ifelse(resmuts%in%T,species,"")))+
  geom_abline(linetype="dashed")+
  scale_x_continuous(name="Net Growth Rate Rep A",limits=c(-.025,.06))+
  scale_y_continuous(name="Net Growth Rate Rep B",limits=c(-.025,.06))+
  labs(fill="Sanger\n Mutant")+
  scale_fill_manual(values=c("gray90","orange","red"))+
  coord_equal()+
  cleanup+theme(legend.position = "none")
# ggsave("output/ABLEnrichmentScreens/ABL_Region234_Comparisons/sm_imatinib_plots/ImatinibEnrichment_Plot_netgr_4.28.24.pdf",width=4,height=4,units="in",useDingbats=F)

# Scatterplot with only single nucleotide (i.e. likely) substitutions.
# Notice how there seem to be less "false positives" for resistance in this plot vs the first plot
ggplot(data%>%filter(!species%in%c("V299L"),protein_start>=242,protein_start<=494,n_nuc_min%in%1)%>%arrange(resmuts),aes(x=netgr.imat.high.1,y=netgr.imat.high.2))+
  geom_point(color="black",shape=21,aes(fill=factor(resmut_cosmic)))+
  geom_label_repel(size=2,aes(label=ifelse(resmuts%in%T,species,"")))+
  geom_abline(linetype="dashed")+
  scale_x_continuous(name="Net Growth Rate Rep A",limits=c(-.025,.06))+
  scale_y_continuous(name="Net Growth Rate Rep B",limits=c(-.025,.06))+
  labs(fill="Sanger\n Mutant")+
  scale_fill_manual(values=c("gray90","orange","red"))+
  coord_equal()+
  cleanup+theme(legend.position = "none")
# ggsave("output/ABLEnrichmentScreens/ABL_Region234_Comparisons/sm_imatinib_plots/ImatinibEnrichment_Plot_netgr_onesitesonly_4.28.24.pdf",width=4,height=4,units="in",useDingbats=F)

# Plotly plot for visualizing which mutants are present
plotly=ggplot(data%>%filter(protein_start>=242,protein_start<=494),aes(x=netgr.imat.high.1,y=netgr.imat.high.2,label=species,color=resmuts))+
  geom_text(size=3)+
  # geom_point(color="black",shape=21,aes(fill=resmut_cosmic))+
  geom_abline()+
  scale_x_continuous(name="Net Growth Rate D0 D2")+
  scale_y_continuous(name="Net Growth Rate D0 D4")+
  labs(fill="Sanger\n Mutant")+
  # scale_fill_manual(values=c("gray90","orange","red"))+
  cleanup
  # +theme(legend.position = "none")
ggplotly(plotly)



#######################Boxplots#######################
ggplot(data%>%filter(!species%in%c("V299L"),protein_start>=242,protein_start<=494),aes(x=resmut_cosmic,y=netgr_mean,fill=resmut_cosmic))+
  # geom_violin(color="black")+
  geom_boxplot(color="black")+
  # geom_jitter(color="black", size=.5,width=.1, alpha=0.9)+
  scale_fill_manual(values=c("gray90","orange","red"))+
  scale_y_continuous(name=bquote('Net Growth Rate '(Hours^-1)))+
  scale_x_discrete("Resistance Status",labels=c("Never\n Seen","Rare\n VUDR","Known\n Resistant\n Variant"))+
  cleanup+
  theme(legend.position = "none",axis.title.x = element_blank())
# ggsave("output/ABLEnrichmentScreens/ABL_Region234_Comparisons/sm_imatinib_plots/ImatinibEnrichment_netgr_BoxPlot_4.28.24.pdf",width=2.5,height=3,units="in",useDingbats=F)


#######################Distribution plots#######################
ggplot(data%>%filter(!species%in%c("V299L"),protein_start>=242,protein_start<=512),aes(x=netgr_mean,fill=resmuts))+
  geom_density(alpha=.7)+
  scale_x_continuous(bquote('Net Growth Rate '(Hours^-1)))+
  scale_y_continuous("Density")+
  scale_fill_manual(values=c("gray","red"))+
  cleanup+
  theme(legend.position = "none")


#######################Heatmaps#######################
heatmap_plotting_function(data%>%mutate(netgr_mean=case_when(netgr_mean>=.055~.055,
                                                                 netgr_mean<=0~0,
                                                    T~netgr_mean)),242,494,fill_variable = "netgr_mean",fill_name = "Net growth rate")+
  scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,494),expand=c(0,0))+
  # scale_fill_gradient2(low ="darkblue",mid = "white",midpoint=0.04, high ="red",name="Net growth rate")+
  # scale_fill_gradient(low ="lightblue", high ="red",name="Net growth rate")+
  scale_fill_gradient2(low="blue",mid ="#9999FF",midpoint=.03, high ="red",name="Net growth rate")+theme(legend.position = "none")
# ggsave("output/ABLEnrichmentScreens/ABL_Region234_Comparisons/sm_imatinib_plots/baf3_imatinib_heatmap_1200nM_score_medianadjusted.pdf",width=10,height = 4,units = "in",useDingbats=F)


# Heatmap with legend for adobe illustrator
heatmap_plotting_function(data%>%mutate(netgr_mean=case_when(netgr_mean>=.055~.055,
                                                                 netgr_mean<=0~0,
                                                    T~netgr_mean)),242,494,fill_variable = "netgr_mean",fill_name = "Net growth rate")+
  scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,494),expand=c(0,0))+
  # scale_fill_gradient2(low ="darkblue",mid = "white",midpoint=0.04, high ="red",name="Net growth rate")+
  # scale_fill_gradient(low ="lightblue", high ="red",name="Net growth rate")+
  scale_fill_gradient2(low="blue",mid ="#9999FF",midpoint=.03, high ="red",name="Net growth rate")
# ggsave("output/ABLEnrichmentScreens/ABL_Region234_Comparisons/sm_imatinib_plots/baf3_imatinib_heatmap_1200nM_score_medianadjusted_withlegend.pdf",width=12,height = 4,units = "in",useDingbats=F)

```

Pooled vs spike-in:
Comparing the fitness of pooled mutants to mutant standards
```{r}
library("ggpubr")
###Downloading mutation bias data. Source of this data: https://github.com/pritchardlabatpsu/SurvivalOfTheLikeliest/tree/master/Figures/Figure3

# compicmut=read.csv("data/Combined_data_frame_IC_Mutprob_abundance.csv",header=T,stringsAsFactors=F)

#############################300 nM Imatinib#############################
standards=read.csv("output/ic50data_all_conc.csv",header = T,stringsAsFactors = F)
standards=standards%>%filter(conc%in%0.3)
standards=merge(data,standards,by.x="species",by.y="mutant")

ggplot(standards,aes(x=.055-drug_effect,y=netgr.imat.low.2,label=species))+
  geom_text(size=2)+
  # scale_x_continuous("Predicted Net Growth Rate (Hrs-1) \nClonal IC50",limits=c(-0.01,.06))+
  # scale_y_continuous("Observed Net Growth Rate (Hrs-1) \nDMS Screen",limits=c(-0.01,.06))+
  scale_x_continuous(element_blank(),limits=c(-0.01,.065))+
  scale_y_continuous(element_blank(),limits=c(-0.01,.065))+
  stat_cor(method="pearson")+
  geom_abline(linetype="dashed")+
  cleanup
# ggsave("output/ABLEnrichmentScreens/ABL_Region234_Comparisons/sm_imatinib_plots/mutantstandards_dms_vs_ic50s_300nM.pdf",width=3,height = 3,units="in",useDingbats=F)

 #############################600 nM Imatinib#############################
standards=read.csv("output/ic50data_all_conc.csv",header = T,stringsAsFactors = F)
standards=standards%>%filter(conc%in%0.6)
standards=merge(data,standards,by.x="species",by.y="mutant")

ggplot(standards,aes(x=.055-drug_effect,y=netgr.imat.mid.1,label=species))+
  geom_text(size=2)+
  # scale_x_continuous("Predicted Net Growth Rate (Hrs-1) \nClonal IC50",limits=c(-0.01,.06))+
  # scale_y_continuous("Observed Net Growth Rate (Hrs-1) \nDMS Screen",limits=c(-0.01,.06))+
  scale_x_continuous(element_blank(),limits=c(-0.01,.065))+
  scale_y_continuous(element_blank(),limits=c(-0.01,.065))+
  stat_cor(method="pearson")+
  geom_abline(linetype="dashed")+
  cleanup
# ggsave("output/ABLEnrichmentScreens/ABL_Region234_Comparisons/sm_imatinib_plots/mutantstandards_dms_vs_ic50s_600nM.pdf",width=3,height = 3,units="in",useDingbats=F)

#############################1200 nM Imatinib#############################
standards=read.csv("output/ic50data_all_conc.csv",header = T,stringsAsFactors = F)
standards=standards%>%filter(conc%in%0.6)
standards=merge(data,standards,by.x="species",by.y="mutant")

ggplot(standards,aes(x=.055-drug_effect,y=netgr.imat.high.1,label=species))+
  # geom_text(size=2)+
  # geom_point()+
  geom_text(size=2)+
  # scale_x_continuous("Predicted Net Growth Rate (Hrs-1) \nClonal IC50",limits=c(-0.01,.06))+
  # scale_y_continuous("Observed Net Growth Rate (Hrs-1) \nDMS Screen",limits=c(-0.01,.06))+
  scale_x_continuous(element_blank(),limits=c(-0.01,.065))+
  scale_y_continuous(element_blank(),limits=c(-0.01,.065))+
  stat_cor(method="pearson")+
  geom_abline(linetype="dashed")+
  cleanup

# ggsave("output/ABLEnrichmentScreens/ABL_Region234_Comparisons/sm_imatinib_plots/mutantstandards_dms_vs_ic50s_1200nM.pdf",width=3,height = 3,units="in",useDingbats=F)

```


```{r}
twinstrand_maf_merge=read.csv("output/twinstrand_maf_merge.csv",header = T,stringsAsFactors = F)
twinstrand_maf_merge=twinstrand_maf_merge%>%
  mutate(Spike_in_freq=as.numeric(Spike_in_freq))%>%
  mutate(Spike_in_freq=case_when(experiment=="Enu_4"~2000,
                                 experiment==experiment~Spike_in_freq))%>%
  mutate(actualDepth=Depth*3)%>% #To account for 2 mouse 1 human reads
  mutate(expectedAltDepth=case_when(time_point=="D0"&Spike_in_freq==1000~Depth/1000,
                                    time_point=="D0"&Spike_in_freq==5000~Depth/5000,
                                    time_point=="D0"&Spike_in_freq==2000~Depth/2000,
                                    time_point==time_point~NaN))

a=twinstrand_maf_merge%>%filter(time_point=="D0",experiment%in%c("M3","M6")&tki_resistant_mutation=="True"|experiment%in%"Enu_4",!mutant=="NA",!mutant=="D276G",!mutant=="V280syn")%>%
  mutate(expectedAltDepth=case_when(experiment=="Enu_4"&mutant=="F311L"~expectedAltDepth,
                                   experiment=="Enu_4"&mutant=="T315I"~expectedAltDepth*55,
                                   experiment=="Enu_4"&mutant=="F317L"~expectedAltDepth*6,
                                   experiment=="Enu_4"&mutant=="E355G"~expectedAltDepth*3,
                                   experiment=="Enu_4"&mutant=="F359V"~expectedAltDepth*13,
                                   experiment=="Enu_4"&mutant=="F359C"~expectedAltDepth*5,
                                   experiment=="Enu_4"&mutant=="H396P"~expectedAltDepth*17,
                                   experiment=="Enu_4"&mutant=="A397P"~expectedAltDepth*12,
                                   experiment=="Enu_4"&mutant=="Y253H"~expectedAltDepth*63,
                                   experiment=="Enu_4"&mutant=="Q252H"~expectedAltDepth*5,
                                   experiment=="Enu_4"&mutant=="G250E"~expectedAltDepth*11,
                                   experiment=="Enu_4"&mutant=="L248V"~expectedAltDepth*6,
                                   experiment=="Enu_4"&mutant=="H214R"~expectedAltDepth*4,
                                   experiment=="Enu_4"&mutant=="K285N"~expectedAltDepth*5,
                                   experiment=="Enu_4"&mutant=="L324R"~expectedAltDepth*7,
                                   mutant==mutant~expectedAltDepth))

twinstrand_simple_melt_merge=read.csv("output/twinstrand_simple_melt_merge.csv",header = T,stringsAsFactors = F)
a=twinstrand_simple_melt_merge%>%
  filter(!experiment%in%c("Enu_4","Enu_3"),duration%in%"d3d6",conc=="0.8")%>%
  mutate(netgr_obs=case_when(experiment=="M5"~netgr_obs+.015,
                                   experiment%in%c("M6","M3","M5","M4","M7")~netgr_obs))

a=a%>%filter(experiment%in%c("M3","M5"),!mutant%in%"D276G")
spikeins_cast=dcast(a,formula = mutant~experiment,value.var = "netgr_obs")


standards=merge(data,spikeins_cast,by.x="species",by.y="mutant")

ggplot(standards,aes(x=M5,y=netgr.imat.mid.1,label=species))+geom_text()
ggplot(standards,aes(x=M3,y=netgr.IL3.1,label=species))+geom_text()

# As you can see, there isn't a strong correlation between the mutant standards measured in DMS DS and in the 20-mutant spike-in screen. Part of the reason is probably the absense of high sensitivity mutants (such as V299L) from the spike-in data
```

