---
title: "BCRABL_FunctionalKinaseAnalysis"
author: "Haider Inam"
date: '2022-11-02'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath(".."))

library(stringr)
library(dplyr)
library(ggplot2)
library(plotly)
library(tictoc)
library(doParallel)
library(foreach)
library(RColorBrewer)
#Cleanup code for plotting
cleanup=theme_bw() +
  theme(plot.title = element_text(hjust=.5), 
        panel.grid.major = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(color = "black"),
        axis.text = element_text(face="bold",color="black",size="11"),
        text=element_text(size=11,face="bold"),
        axis.title=element_text(face="bold",size="11"))

```


Plotting the correlation of allele frequencies of stuff that I see in two samples
```{r}
# samplex=read.csv("data/Consensus_Data/Novogene_lane14/sample10_combined/sscs/variant_caller_outputs/variants_unique_ann.csv",header=T,stringsAsFactors = F)
samplex=read.csv("data/Consensus_Data/Novogene_lane15/sample_6/sscs/variant_caller_outputs/variants_unique_ann.csv",header=T,stringsAsFactors = F)
# "Novogene_lane15/sample_6/sscs","Novogene_lane14/sample12/sscs"
samplex=samplex%>%
  rowwise()%>%
  mutate(ref_aa=strsplit(amino_acids,"/")[[1]][1],
         alt_aa=strsplit(amino_acids,"/")[[1]][2])
samplex=samplex%>%mutate(maf=ct/depth)
samplex_simple=samplex%>%dplyr::select(alt_start_pos,protein_start,ref,alt,ref_aa,alt_aa,consequence_terms,ct,depth,maf)

# sampley=read.csv("data/Consensus_Data/Novogene_lane15/sample_3/sscs/variant_caller_outputs/variants_unique_ann.csv",header=T,stringsAsFactors = F)
sampley=read.csv("data/Consensus_Data/Novogene_lane14/sample12/sscs/variant_caller_outputs/variants_unique_ann.csv",header=T,stringsAsFactors = F)
# sample7=read.csv("data/Consensus_Data/Novogene_lane15/sample_7/sscs/variant_caller_outputs/variants_unique_ann.csv",header=T,stringsAsFactors = F)

sampley=sampley%>%
  rowwise()%>%
  mutate(ref_aa=strsplit(amino_acids,"/")[[1]][1],
         alt_aa=strsplit(amino_acids,"/")[[1]][2])
sampley=sampley%>%mutate(maf=ct/depth)
sampley_simple=sampley%>%dplyr::select(alt_start_pos,protein_start,ref,alt,ref_aa,alt_aa,consequence_terms,ct,depth,maf)

samples_xy=merge(samplex_simple%>%filter(consequence_terms%in%"missense_variant"),sampley_simple%>%filter(consequence_terms%in%"missense_variant"),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all = T)

plotly=ggplot(samples_xy%>%filter(protein_start>=242,protein_start<=492),aes(x=maf.x,y=maf.y))+
  geom_point(color="black",shape=21,aes(fill=ct.x))+
    scale_x_continuous(trans="log")+
    scale_y_continuous(trans="log")+
  geom_abline()+
  theme_bw()
ggplotly(plotly)
# cor(samples_xy$maf.x,samples_xy$maf.y)
```

Background samples: Lane13 Sample 7, Lane 14 Sample 10, 11?
D2 Post iL3 withdrawal: Lane 13 Sample 9,10
D4 Post iL3 withdrawal: Lane 13 Sample 11,12
D2 Post Imatinib treatment: Lane 14 Sample 12

SSCS
```{r}
source("code/merge_samples.R")
source("code/depth_finder.R")
il3all=merge_samples("Novogene_lane14/Sample10_combined/sscs","Novogene_lane13/sample7/sscs")
il3all=merge_samples(il3all,"Novogene_lane14/sample11/sscs")
il3all=merge_samples(il3all,"Novogene_lane15/sample_3/sscs")
il3all=merge_samples(il3all,"Novogene_lane13/Sample10/sscs")
il3all=merge_samples(il3all,"Novogene_lane13/Sample9/sscs")
il3all=merge_samples(il3all,"Novogene_lane15/sample_4/sscs")
il3all=merge_samples(il3all,"Novogene_lane15/sample_5/sscs")
il3all=merge_samples(il3all,"Novogene_lane15/sample_6/sscs")
il3all=merge_samples(il3all,"Novogene_lane15/sample_7/sscs")
# a=il3all%>%filter(protein_start%in%c(242:494),consequence_terms%in%"missense_variant")

# il3D0=merge_samples("Novogene_lane14/Sample10_combined/sscs","Novogene_lane15/sample_3/sscs")
# a=il3D0%>%filter(!protein_start%in%c(242:494),type%in%"mnv")
# a=il3D0%>%filter(protein_start%in%c(242:494),consequence_terms%in%"missense_variant")
il3D0=merge_samples("Novogene_lane14/Sample10_combined/sscs","Novogene_lane13/sample7/sscs")
il3D0=merge_samples(il3D0,"Novogene_lane14/sample11/sscs")
il3D0=merge_samples(il3D0,"Novogene_lane15/sample_3/sscs")

il3D0=il3D0%>%
  rowwise()%>%
  mutate(ref_aa=strsplit(amino_acids,"/")[[1]][1],
         alt_aa=strsplit(amino_acids,"/")[[1]][2])
il3D0=il3D0%>%mutate(maf=ct/depth)
il3D0=il3D0%>%mutate(totalcells=370,totalmutant=maf*totalcells)
il3D0_simple=il3D0%>%dplyr::select(alt_start_pos,protein_start,ref,alt,ref_aa,alt_aa,consequence_terms,ct,depth,maf,totalcells,totalmutant)

# a=il3D0%>%filter(protein_start%in%c(242:494),consequence_terms%in%"missense_variant")
# a=il3D0.D2.merge%>%filter(consequence_terms%in%"missense_variant",protein_start%in%c(242:494))
il3D2=merge_samples("Novogene_lane13/Sample9/sscs","Novogene_lane13/Sample10/sscs")
il3D2=merge_samples(il3D2,"Novogene_lane15/Sample_4/sscs")
il3D2=il3D2%>%
  rowwise()%>%
  mutate(ref_aa=strsplit(amino_acids,"/")[[1]][1],
         alt_aa=strsplit(amino_acids,"/")[[1]][2])
il3D2=il3D2%>%mutate(maf=ct/depth)
il3D2=il3D2%>%mutate(totalcells=1515,totalmutant=maf*totalcells)
il3D2_simple=il3D2%>%dplyr::select(alt_start_pos,protein_start,ref,alt,ref_aa,alt_aa,consequence_terms,ct,depth,maf,totalcells,totalmutant)


# imatD2=read.csv("data/Consensus_Data/Novogene_lane14/sample12/variant_caller_outputs/variants_unique_ann.csv",stringsAsFactors = F)
imatD2=merge_samples("Novogene_lane15/sample_6/sscs","Novogene_lane14/sample12/sscs")
# imatD2=merge_samples(imatD2,"Novogene_lane15/Sample4")
imatD2=imatD2%>%
  rowwise()%>%
  mutate(ref_aa=strsplit(amino_acids,"/")[[1]][1],
         alt_aa=strsplit(amino_acids,"/")[[1]][2])
imatD2=imatD2%>%mutate(maf=ct/depth)
imatD2=imatD2%>%mutate(totalcells=1192,totalmutant=maf*totalcells)
imatD2_simple=imatD2%>%dplyr::select(alt_start_pos,protein_start,ref,alt,ref_aa,alt_aa,consequence_terms,ct,depth,maf,totalcells,totalmutant)



il3D4=read.csv("data/Consensus_Data/novogene_lane15/sample_5/sscs/variant_caller_outputs/variants_unique_ann.csv")
# imatD2=merge_samples(imatD2,"Novogene_lane15/Sample4")
il3D4=il3D4%>%
  rowwise()%>%
  mutate(ref_aa=strsplit(amino_acids,"/")[[1]][1],
         alt_aa=strsplit(amino_acids,"/")[[1]][2])
il3D4=il3D4%>%mutate(maf=ct/depth)
il3D4=il3D4%>%mutate(totalcells=17239,totalmutant=maf*totalcells)
il3D4_simple=il3D4%>%dplyr::select(alt_start_pos,protein_start,ref,alt,ref_aa,alt_aa,consequence_terms,ct,depth,maf,totalcells,totalmutant)


imatD4=read.csv("data/Consensus_Data/novogene_lane15/sample_7/sscs/variant_caller_outputs/variants_unique_ann.csv")
# imatD2=merge_samples(imatD2,"Novogene_lane15/Sample4")
imatD4=imatD4%>%
  rowwise()%>%
  mutate(ref_aa=strsplit(amino_acids,"/")[[1]][1],
         alt_aa=strsplit(amino_acids,"/")[[1]][2])
imatD4=imatD4%>%mutate(maf=ct/depth)
imatD4=imatD4%>%mutate(totalcells=10397,totalmutant=maf*totalcells)
imatD4_simple=imatD4%>%dplyr::select(alt_start_pos,protein_start,ref,alt,ref_aa,alt_aa,consequence_terms,ct,depth,maf,totalcells,totalmutant)


##########IL3 Day 2 vs IL3 D0############
il3D0.D2=merge(il3D0_simple%>%filter(consequence_terms%in%"missense_variant")%>%mutate(totalmutant=totalmutant*107/370),il3D2_simple%>%filter(consequence_terms%in%"missense_variant"),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all.x = T)
il3D0.D2$alt_aa=factor(il3D0.D2$alt_aa,levels=c("P","G","Y","W","F","V","L","I","A","T","S","Q","N","M","C","E","D","R","K","H"))
il3D0.D2[il3D0.D2$ct.y%in%NA,"ct.y"]=.5

il3D0.D2=depth_finder(il3D0.D2,"depth.y")
il3D0.D2=depth_finder(il3D0.D2,"totalcells.y")
il3D0.D2=il3D0.D2%>%mutate(maf.y=ct.y/depth.y,
                           totalmutant.y=maf.y*totalcells.y)

il3D0.D2[il3D0.D2$ct.x%in%NA,"ct.x"]=.5

il3D0.D2=il3D0.D2%>%mutate(score=log2(maf.y/maf.x))
# il3D0.D2[il3D0.D2$score%in%NA,"score"]=-6

il3D0.D2[il3D0.D2$totalmutant.y%in%NA,"totalmutant.y"]=0
il3D0.D2=il3D0.D2%>%mutate(netgr_obs=log(totalmutant.y/totalmutant.x)/48)
il3D0.D2[il3D0.D2$netgr_obs%in%NA,"netgr_obs"]=-.055


##########IL3 Day 4 vs IL3 D2############
il3D2.D4=merge(il3D2_simple%>%filter(consequence_terms%in%"missense_variant"),il3D4_simple%>%filter(consequence_terms%in%"missense_variant"),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all.x = T)
il3D2.D4[il3D2.D4$ct.y%in%NA,"ct.y"]=.5

il3D2.D4=depth_finder(il3D2.D4,"depth.y")
il3D2.D4=depth_finder(il3D2.D4,"totalcells.y")
il3D2.D4=il3D2.D4%>%mutate(maf.y=ct.y/depth.y,
                           totalmutant.y=maf.y*totalcells.y)

il3D2.D4[il3D2.D4$ct.x%in%NA,"ct.x"]=.5
il3D2.D4=il3D2.D4%>%mutate(score=log2(maf.y/maf.x))
# il3D2.D4[il3D2.D4$score%in%NA,"score"]=-6

il3D2.D4[il3D2.D4$totalmutant.y%in%NA,"totalmutant.y"]=0
il3D2.D4=il3D2.D4%>%mutate(netgr_obs=log(totalmutant.y/totalmutant.x)/48)
il3D2.D4[il3D2.D4$netgr_obs%in%NA,"netgr_obs"]=-.055

##########IL3 Day 4 vs Imat D0############
il3D0.D4=merge(il3D0_simple%>%filter(consequence_terms%in%"missense_variant")%>%mutate(totalmutant=totalmutant*107/370),il3D4_simple%>%filter(consequence_terms%in%"missense_variant"),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all.x = T)
# il3D0.D4[il3D0.D4$ct.y%in%NA,"ct.y"]=0
il3D0.D4[il3D0.D4$ct.y%in%NA,"ct.y"]=.5

il3D0.D4=depth_finder(il3D0.D4,"depth.y")
il3D0.D4=depth_finder(il3D0.D4,"totalcells.y")
il3D0.D4=il3D0.D4%>%mutate(maf.y=ct.y/depth.y,
                           totalmutant.y=maf.y*totalcells.y)

il3D0.D4[il3D0.D4$ct.x%in%NA,"ct.x"]=.5
il3D0.D4=il3D0.D4%>%mutate(score=log2(maf.y/maf.x))
# il3D0.D4[il3D0.D4$score%in%NA,"score"]=-6

il3D0.D4[il3D0.D4$totalmutant.y%in%NA,"totalmutant.y"]=0
il3D0.D4=il3D0.D4%>%mutate(netgr_obs=log(totalmutant.y/totalmutant.x)/96)
il3D0.D4[il3D0.D4$netgr_obs%in%NA,"netgr_obs"]=-.055

##########Imatinib Day 2 vs Imat D0############
imatD0.D2=merge(il3D0_simple%>%filter(consequence_terms%in%"missense_variant"),imatD2_simple%>%filter(consequence_terms%in%"missense_variant"),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all.x = T)
# imatD0.D2[imatD0.D2$ct.y%in%NA,"ct.y"]=0
imatD0.D2[imatD0.D2$ct.y%in%NA,"ct.y"]=.5

imatD0.D2=depth_finder(imatD0.D2,"depth.y")
imatD0.D2=depth_finder(imatD0.D2,"totalcells.y")
imatD0.D2=imatD0.D2%>%mutate(maf.y=ct.y/depth.y,
                           totalmutant.y=maf.y*totalcells.y)

imatD0.D2[imatD0.D2$ct.x%in%NA,"ct.x"]=.5
imatD0.D2=imatD0.D2%>%mutate(score=log2(maf.y/maf.x))
# imatD0.D2[imatD0.D2$score%in%NA,"score"]=-6

imatD0.D2[imatD0.D2$totalmutant.y%in%NA,"totalmutant.y"]=0
imatD0.D2=imatD0.D2%>%mutate(netgr_obs=log(totalmutant.y/totalmutant.x)/48)
imatD0.D2[imatD0.D2$netgr_obs%in%NA,"netgr_obs"]=-.055


##########Imatinib Day 4 vs Imat D2############
imatD2.D4=merge(imatD2_simple%>%filter(consequence_terms%in%"missense_variant"),imatD4_simple%>%filter(consequence_terms%in%"missense_variant"),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all.x = T)
# imatD2.D4[imatD2.D4$ct.y%in%NA,"ct.y"]=0
imatD2.D4[imatD2.D4$ct.y%in%NA,"ct.y"]=.5

imatD2.D4=depth_finder(imatD2.D4,"depth.y")
imatD2.D4=depth_finder(imatD2.D4,"totalcells.y")
imatD2.D4=imatD2.D4%>%mutate(maf.y=ct.y/depth.y,
                           totalmutant.y=maf.y*totalcells.y)

imatD2.D4[imatD2.D4$ct.x%in%NA,"ct.x"]=.5
imatD2.D4=imatD2.D4%>%mutate(score=log2(maf.y/maf.x))
# imatD2.D4[imatD2.D4$score%in%NA,"score"]=-6

imatD2.D4[imatD2.D4$totalmutant.y%in%NA,"totalmutant.y"]=0
imatD2.D4=imatD2.D4%>%mutate(netgr_obs=log(totalmutant.y/totalmutant.x)/48)
imatD2.D4[imatD2.D4$netgr_obs%in%NA,"netgr_obs"]=-.055

##########Imatinib Day 4 vs Imat D0############
imatD0.D4=merge(il3D0_simple%>%filter(consequence_terms%in%"missense_variant"),imatD4_simple%>%filter(consequence_terms%in%"missense_variant"),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all.x = T)
# imatD0.D4[imatD0.D4$ct.y%in%NA,"ct.y"]=0
imatD0.D4[imatD0.D4$ct.y%in%NA,"ct.y"]=.5

imatD0.D4=depth_finder(imatD0.D4,"depth.y")
imatD0.D4=depth_finder(imatD0.D4,"totalcells.y")
imatD0.D4=imatD0.D4%>%mutate(maf.y=ct.y/depth.y,
                           totalmutant.y=maf.y*totalcells.y)

imatD0.D4[imatD0.D4$ct.x%in%NA,"ct.x"]=.5
imatD0.D4=imatD0.D4%>%mutate(score=log2(maf.y/maf.x))
# imatD0.D4[imatD0.D4$score%in%NA,"score"]=-6

imatD0.D4[imatD0.D4$totalmutant.y%in%NA,"totalmutant.y"]=0
imatD0.D4=imatD0.D4%>%mutate(netgr_obs=log(totalmutant.y/totalmutant.x)/96)
imatD0.D4[imatD0.D4$netgr_obs%in%NA,"netgr_obs"]=-.055

######Plotting imat D0 D2 Heatmap####
ggplot(imatD0.D2%>%filter(nchar(as.character(alt_aa))%in%1,protein_start>=242,protein_start<=494),aes(x=protein_start,y=alt_aa,fill=score))+
  geom_tile()+
  theme(panel.background=element_rect(fill="white", colour="black"))+
  scale_fill_gradient2(low ="darkblue",midpoint=-1,mid="white", high ="red",name="Enrichment Score")+
  scale_color_manual(values=c("black"))+scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,493),expand=c(0,0),breaks = c(250,253,255,276,299,315,317,351,355,359,396,459,486))+
  ylab("Mutant Amino Acid")
# ggsave("BCRABL_imatinib_D2.pdf",height=6,width=24,units="in",useDingbats=F)

######Plotting resistant imat D0 D2 Heatmap####
ggplot(imatD0.D2%>%filter(nchar(as.character(alt_aa))%in%1,protein_start%in%c(250,253,255,276,299,315,317,351,355,359,396,459,486)),aes(x=protein_start,y=alt_aa,fill=score))+
  geom_tile()+
  theme(panel.background=element_rect(fill="white", colour="black"))+
  scale_fill_gradient2(low ="darkblue",midpoint=-1,mid="white", high ="red",name="Enrichment Score")+
  scale_color_manual(values=c("black"))+scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,493),expand=c(0,0),breaks =c(250,253,255,276,299,315,317,351,355,359,396,459,486))+
  ylab("Mutant Amino Acid")
# ggsave("BCRABL_imatinib_D2_resistantresidues.pdf",height=6,width=24,units="in",useDingbats=F)

######Plotting iL3 D0 D2 Heatmap####
il3D0.D2$alt_aa=factor(il3D0.D2$alt_aa,levels=c("P","G","Y","W","F","V","L","I","A","T","S","Q","N","M","C","E","D","R","K","H"))
ggplot(il3D0.D2%>%filter(nchar(as.character(alt_aa))%in%1,protein_start>=242,protein_start<=494),aes(x=protein_start,y=alt_aa,fill=score))+
  geom_tile()+
  theme(panel.background=element_rect(fill="white", colour="black"))+
  scale_fill_gradient2(low ="darkblue",midpoint=-1,mid="white", high ="red",name="Enrichment Score")+
  scale_color_manual(values=c("black"))+scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,493),expand=c(0,0),breaks = c(248,250,256, 271,275,300,325,350,363,375,381,400,405,425,450,475))+
  ylab("Mutant Amino Acid")
# a=il3D0.D2%>%filter(netgr_obs%in%"-Inf",consequence_terms%in%"missense_variant",protein_start%in%c(242:494))

# ggsave("BCRABL_iL3Independence_D2.pdf",height=6,width=24,units="in",useDingbats=F)
# write.csv(il3D0.D2,"BCRABL_Il3Independence_D2.csv")

#####Focusing on conserved residues#####
ggplot(il3D0.D4%>%filter(nchar(as.character(alt_aa))%in%1,protein_start%in%c(271,363,381:383)),aes(x=protein_start,y=alt_aa,fill=netgr_obs))+
  geom_tile()+
  theme(panel.background=element_rect(fill="white", colour="black"))+
  scale_fill_gradient2(low ="blue",midpoint=0,mid="white", high ="red",name="Enrichment Score")+
  scale_color_manual(values=c("black"))+scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,493),expand=c(0,0),breaks = c(248,250,256, 271,275,300,325,350,363,375,381,400,405,425,450,475))+
  ylab("Mutant Amino Acid")
# ggsave("BCRABL_iL3Independence_D2_essential.pdf",height=6,width=24,units="in",useDingbats=F)


#####Graying out unseen residues#####
###D0 D2 IL3
df_grid  = expand.grid(protein_start = c(242:493),alt_aa = unique(il3D0.D2$alt_aa))

il3D0.D2.merge=merge(df_grid,il3D0.D2,by=c("protein_start","alt_aa"),all=T)
il3D0.D2.merge=il3D0.D2.merge%>%filter(nchar(as.character(alt_aa))%in%1,protein_start>=242,protein_start<=494)
###Trying to add in the score as WT for residues that are wt
reference_seq=read.table("data/Refs/ABL/abl_cds_translation.txt")
il3D0.D2.merge=il3D0.D2.merge%>%
  rowwise()%>%
  # group_by(protein_start)%>%
  mutate(ref_aa=case_when(ref_aa%in%NA~substr(reference_seq,protein_start,protein_start),
                          T~ref_aa))%>%
  mutate(wt=case_when(ref_aa==alt_aa~T,
                      T~F))



##########Plotting gray and yellow heatmap
il3D0.D2.merge$alt_aa=factor(il3D0.D2.merge$alt_aa,levels=c("P","G","Y","W","F","V","L","I","A","T","S","Q","N","M","C","E","D","R","K","H"))
il3D0.D2.merge.filtered=il3D0.D2.merge%>%filter(!protein_start%in%c(290:305))
a=il3D0.D2.merge.filtered%>%filter(wt%in%T)
ggplot(il3D0.D2.merge.filtered,aes(x=protein_start,y=alt_aa))+
  geom_tile(data=subset(il3D0.D2.merge.filtered,!is.na(score)),aes(fill=score))+
  scale_fill_gradient2(low ="darkblue",midpoint=-1,mid="white", high ="red",name="Enrichment Score")+
  geom_tile(data=subset(il3D0.D2.merge.filtered,is.na(score)&wt%in%F),aes(color="white"),linetype = "solid",color="white", fill = "gray90", alpha = 0.8)+
  geom_tile(data=subset(il3D0.D2.merge.filtered,is.na(score)&wt%in%T),aes(color="white"),linetype = "solid",color="white", fill = "yellow", alpha = 0.4)+
  theme(panel.background=element_rect(fill="white", colour="black"))+scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,493),expand=c(0,0))+
  ylab("Mutant Amino Acid")
# ggsave("BCRABL_il3_D2_gray.pdf",height=6,width=24,units="in",useDingbats=F)



###D0 D4 IL3
#####Graying out unseen residues#####
df_grid  = expand.grid(protein_start = c(242:493),alt_aa = unique(il3D0.D4$alt_aa))

il3D0.D4.merge=merge(df_grid,il3D0.D4,by=c("protein_start","alt_aa"),all=T)
il3D0.D4.merge=il3D0.D4.merge%>%filter(nchar(as.character(alt_aa))%in%1,protein_start>=242,protein_start<=494)
###Trying to add in the score as WT for residues that are wt
reference_seq=read.table("data/Refs/ABL/abl_cds_translation.txt")
il3D0.D4.merge=il3D0.D4.merge%>%
  rowwise()%>%
  # group_by(protein_start)%>%
  mutate(ref_aa=case_when(ref_aa%in%NA~substr(reference_seq,protein_start,protein_start),
                          T~ref_aa))%>%
  mutate(wt=case_when(ref_aa==alt_aa~T,
                      T~F))

##########Plotting gray and yellow heatmap
il3D0.D4.merge$alt_aa=factor(il3D0.D4.merge$alt_aa,levels=c("P","G","Y","W","F","V","L","I","A","T","S","Q","N","M","C","E","D","R","K","H"))
il3D0.D4.merge.filtered=il3D0.D4.merge%>%filter(!protein_start%in%c(290:305))
# a=il3D0.D4.merge.filtered%>%filter(wt%in%T)
ggplot(il3D0.D4.merge.filtered,aes(x=protein_start,y=alt_aa))+
  geom_tile(data=subset(il3D0.D4.merge.filtered,!is.na(score)),aes(fill=netgr_obs))+
  scale_fill_gradient2(low ="darkblue",midpoint=.04,mid="white", high ="red",name="Net growth rate")+
  geom_tile(data=subset(il3D0.D4.merge.filtered,is.na(score)&wt%in%F),aes(color="white"),linetype = "solid",color="white", fill = "gray90", alpha = 0.8)+
  geom_tile(data=subset(il3D0.D4.merge.filtered,is.na(score)&wt%in%T),aes(color="white"),linetype = "solid",color="white", fill = "yellow", alpha = 0.4)+
  theme(panel.background=element_rect(fill="white", colour="black"))+scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,493),expand=c(0,0))+
  ylab("Mutant Amino Acid")


ggplot(il3D0.D4.merge.filtered,aes(x=protein_start,y=alt_aa))+
  geom_tile(data=subset(il3D0.D4.merge.filtered,!is.na(score)),aes(fill=score))+
  scale_fill_gradient2(low ="darkblue",midpoint=0,mid="white", high ="red",name="Enrichment Score")+
  geom_tile(data=subset(il3D0.D4.merge.filtered,is.na(score)&wt%in%F),aes(color="white"),linetype = "solid",color="white", fill = "gray90", alpha = 0.8)+
  geom_tile(data=subset(il3D0.D4.merge.filtered,is.na(score)&wt%in%T),aes(color="white"),linetype = "solid",color="white", fill = "yellow", alpha = 0.4)+
  theme(panel.background=element_rect(fill="white", colour="black"))+scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,493),expand=c(0,0))+
  ylab("Mutant Amino Acid")

# ggsave("BCRABL_il3_D4_scores_updated.pdf",height=6,width=24,units="in",useDingbats=F)


###D0 D4 Imat
#####Graying out unseen residues#####
df_grid  = expand.grid(protein_start = c(242:493),alt_aa = unique(imatD0.D4$alt_aa))

imatD0.D4.merge=merge(df_grid,imatD0.D4,by=c("protein_start","alt_aa"),all=T)
imatD0.D4.merge=imatD0.D4.merge%>%filter(nchar(as.character(alt_aa))%in%1,protein_start>=242,protein_start<=494)
###Trying to add in the score as WT for residues that are wt
reference_seq=read.table("data/Refs/ABL/abl_cds_translation.txt")
imatD0.D4.merge=imatD0.D4.merge%>%
  rowwise()%>%
  # group_by(protein_start)%>%
  mutate(ref_aa=case_when(ref_aa%in%NA~substr(reference_seq,protein_start,protein_start),
                          T~ref_aa))%>%
  mutate(wt=case_when(ref_aa==alt_aa~T,
                      T~F))

##########Plotting gray and yellow heatmap
imatD0.D4.merge$alt_aa=factor(imatD0.D4.merge$alt_aa,levels=c("P","G","Y","W","F","V","L","I","A","T","S","Q","N","M","C","E","D","R","K","H"))
imatD0.D4.merge.filtered=imatD0.D4.merge%>%filter(!protein_start%in%c(290:305))
# a=il3D0.D4.merge.filtered%>%filter(wt%in%T)
ggplot(imatD0.D4.merge.filtered,aes(x=protein_start,y=alt_aa))+
  geom_tile(data=subset(imatD0.D4.merge.filtered,!is.na(score)),aes(fill=netgr_obs))+
  scale_fill_gradient2(low ="darkblue",midpoint=.03,mid="white", high ="red",name="Net growth rate")+
  geom_tile(data=subset(imatD0.D4.merge.filtered,is.na(score)&wt%in%F),aes(color="white"),linetype = "solid",color="white", fill = "gray90", alpha = 0.8)+
  geom_tile(data=subset(imatD0.D4.merge.filtered,is.na(score)&wt%in%T),aes(color="white"),linetype = "solid",color="white", fill = "yellow", alpha = 0.4)+
  theme(panel.background=element_rect(fill="white", colour="black"))+scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,493),expand=c(0,0))+
  ylab("Mutant Amino Acid")
ggsave("BCRABL_imat_D4_gray_updated.pdf",height=6,width=24,units="in",useDingbats=F)


ggplot(imatD0.D4.merge.filtered,aes(x=protein_start,y=alt_aa))+
  geom_tile(data=subset(imatD0.D4.merge.filtered,!is.na(score)),aes(fill=score))+
  scale_fill_gradient2(low ="darkblue",midpoint=0,mid="white", high ="red",name="Enrichment Score")+
  geom_tile(data=subset(imatD0.D4.merge.filtered,is.na(score)&wt%in%F),aes(color="white"),linetype = "solid",color="white", fill = "gray90", alpha = 0.8)+
  geom_tile(data=subset(imatD0.D4.merge.filtered,is.na(score)&wt%in%T),aes(color="white"),linetype = "solid",color="white", fill = "yellow", alpha = 0.4)+
  theme(panel.background=element_rect(fill="white", colour="black"))+scale_x_continuous(name="Residue on the ABL Kinase",limits=c(242,493),expand=c(0,0))+
  ylab("Mutant Amino Acid")
# ggsave("BCRABL_imat_D4_scores.pdf",height=6,width=24,units="in",useDingbats=F)

b=imatD0.D4%>%filter(consequence_terms%in%"missense_variant",protein_start>=242,protein_start<=494)
```

Plotting enrichment scores
```{r}
####Plotting Imatinib D4 D2 scores vs Imatinib D2 D0 scores
#First merging IL3 D0 D2 to IL3 D2 D4
scores_all=merge(il3D0.D2%>%dplyr::select(-ct.x,-depth.x,-maf.x,-totalmutant.x,-ct.y,-depth.y,-maf.y,-totalmutant.y,-totalcells.x,-totalcells.y),il3D2.D4%>%dplyr::select(-ct.x,-depth.x,-maf.x,-totalmutant.x,-ct.y,-depth.y,-maf.y,-totalmutant.y,-totalcells.x,-totalcells.y),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),suffixes = c(".il3d0.d2",".il3d2.d4"),all = T)
#Second, adding Imat D0 D2 to dataframe
scores_all=merge(scores_all,imatD0.D2%>%dplyr::select(-ct.x,-depth.x,-maf.x,-totalmutant.x,-ct.y,-depth.y,-maf.y,-totalmutant.y,-totalcells.x,-totalcells.y),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all = T)
scores_all=scores_all%>%rename(score.imatd0.d2=score,netgr_obs.imatd0.d2=netgr_obs)
#Third, adding Imat D2 D4 to dataframe
scores_all=merge(scores_all,imatD2.D4%>%dplyr::select(-ct.x,-depth.x,-maf.x,-totalmutant.x,-ct.y,-depth.y,-maf.y,-totalmutant.y,-totalcells.x,-totalcells.y),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all = T)
scores_all=scores_all%>%rename(score.imatd2.d4=score,netgr_obs.imatd2.d4=netgr_obs)
#Fourth, adding IL3 D4 D0 to dataframe
scores_all=merge(scores_all,il3D0.D4%>%dplyr::select(-ct.x,-depth.x,-maf.x,-totalmutant.x,-ct.y,-depth.y,-maf.y,-totalmutant.y,-totalcells.x,-totalcells.y),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all = T)
scores_all=scores_all%>%rename(score.il3d0.d4=score,netgr_obs.il3d0.d4=netgr_obs)
#Fifth, adding Imatinib D4 D0 to dataframe
scores_all=merge(scores_all,imatD0.D4%>%dplyr::select(-ct.x,-depth.x,-maf.x,-totalmutant.x,-ct.y,-depth.y,-maf.y,-totalmutant.y,-totalcells.x,-totalcells.y),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"),all = T)
scores_all=scores_all%>%rename(score.imatd0.d4=score,netgr_obs.imatd0.d4=netgr_obs)

scores_all=scores_all%>%mutate(resmuts=case_when(protein_start%in%253&alt_aa%in%"H"~T,
                                                         protein_start%in%255&alt_aa%in%"V"~T,
                                                         protein_start%in%486&alt_aa%in%"S"~T,
                                                         protein_start%in%396&alt_aa%in%"P"~T,
                                                         protein_start%in%255&alt_aa%in%"K"~T,
                                                         protein_start%in%315&alt_aa%in%"I"~T,
                                                         protein_start%in%252&alt_aa%in%"H"~T,
                                                         protein_start%in%253&alt_aa%in%"F"~T,
                                                         protein_start%in%250&alt_aa%in%"E"~T,
                                                         protein_start%in%359&alt_aa%in%"C"~T,
                                                         protein_start%in%351&alt_aa%in%"T"~T,
                                                         protein_start%in%355&alt_aa%in%"G"~T,
                                                         protein_start%in%317&alt_aa%in%"L"~T,
                                                         protein_start%in%359&alt_aa%in%"I"~T,
                                                         protein_start%in%355&alt_aa%in%"A"~T,
                                                         protein_start%in%459&alt_aa%in%"K"~T,
                                                         protein_start%in%276&alt_aa%in%"G"~T,
                                                         protein_start%in%299&alt_aa%in%"L"~T,
                                                         
                                                         T~F))





ggplot(scores_all%>%filter(consequence_terms%in%"missense_variant",protein_start%in%c(242:494),!score.il3d2.d4%in%-6),aes(x=score.il3d0.d2,y=score.il3d2.d4))+geom_point()+geom_abline()

ggplot(scores_all%>%filter(consequence_terms%in%"missense_variant",protein_start%in%c(242:494),!score.il3d2.d4%in%-6),aes(x=score.il3d0.d2,y=score.il3d2.d4))+geom_point()+geom_abline()

ggplot(scores_all%>%filter(consequence_terms%in%"missense_variant",protein_start%in%c(242:494),!score.imatd2.d4%in%-6),aes(x=score.imatd0.d2,y=score.imatd0.d4))+geom_point()+geom_abline()

ggplot(scores_all%>%filter(consequence_terms%in%"missense_variant",protein_start%in%c(242:494),!score.imatd2.d4%in%-6),aes(x=netgr_obs.imatd0.d2,y=netgr_obs.imatd0.d4))+geom_point()+geom_abline()

ggplot(scores_all%>%filter(consequence_terms%in%"missense_variant",protein_start%in%c(242:494),!score.imatd2.d4%in%-6),aes(x=score.il3d0.d2,y=score.il3d0.d4))+geom_point()+geom_abline()

ggplot(scores_all%>%filter(consequence_terms%in%"missense_variant",protein_start%in%c(242:494),!score.imatd2.d4%in%-6),aes(x=score.imatd0.d2,y=score.imatd2.d4))+geom_point()+geom_abline()

ggplot(scores_all%>%filter(consequence_terms%in%"missense_variant",protein_start%in%c(242:494),!score.imatd2.d4%in%-6),aes(x=score.il3d0.d2,y=score.il3d2.d4))+geom_point()+geom_abline()

ggplot(scores_all%>%filter(!score.imatd2.d4%in%-6),aes(x=score.imatd2.d4))+geom_histogram()
ggplot(scores_all%>%filter(!score.imatd0.d4%in%-6),aes(x=score.imatd0.d4))+geom_histogram()
ggplot(scores_all%>%filter(!score.imatd0.d2%in%-6),aes(x=score.imatd0.d2))+geom_histogram()

ggplot(scores_all%>%filter(!score.il3d0.d4%in%-6),aes(x=score.il3d0.d4))+geom_histogram()
ggplot(scores_all%>%filter(!score.il3d2.d4%in%-6),aes(x=score.il3d2.d4))+geom_histogram()
ggplot(scores_all%>%filter(!score.il3d0.d2%in%-6),aes(x=score.il3d0.d2))+geom_histogram()

ggplot(il3D0_simple,aes(x=maf))+geom_density()+scale_x_continuous(trans = "log10")
ggplot(il3D2_simple,aes(x=maf))+geom_density()+scale_x_continuous(trans = "log10")
ggplot(imatD2_simple,aes(x=maf))+geom_density()+scale_x_continuous(trans = "log10")
ggplot(imatD4_simple,aes(x=maf))+geom_density()+scale_x_continuous(trans = "log10")


a=imatD0.D4%>%filter(protein_start%in%c(253,252,459,276))
```



Plotting specific resistant mutants
```{r}
resmuts=scores_all%>%filter(resmuts%in%T)%>%mutate(species=paste(ref_aa,protein_start,alt_aa,sep = ""))

ggplot(resmuts%>%filter(!protein_start%in%317),aes(x=score.imatd0.d2,y=score.imatd2.d4,label=species))+geom_text()+geom_abline()
# ggplot(resmuts%>%filter(!protein_start%in%317),aes(x=netgr_obs.imatd0.d2,y=netgr_obs.imatd2.d4,label=species))+geom_text()+geom_abline()


ic50data_all_sum=read.csv("output/ic50data_all_confidence_intervals_raw_data.csv",row.names = 1)
resmuts_merged=merge(resmuts,ic50data_all_sum,by="species")


#Plotting Imatinib D0 vs D2
ggplot(resmuts_merged,aes(x=netgr_pred_model,y=score.imatd0.d2,label=species))+geom_text()
ggplot(resmuts_merged,aes(x=netgr_pred_model,y=netgr_obs.imatd0.d2,label=species))+geom_text()
#Plotting Imatinib D2 vs D4
ggplot(resmuts_merged%>%filter(!protein_start%in%317),aes(x=netgr_pred_model,y=score.imatd2.d4,label=species))+geom_text()

#Plotting Imatinib D0 vs D4
ggplot(resmuts_merged%>%filter(!protein_start%in%317),aes(x=netgr_pred_model,y=score.imatd0.d4,label=species))+geom_text()

#Plotting Imatinib D0 vs D2 Adjusted
ggplot(resmuts_merged%>%filter(!protein_start%in%317),aes(x=netgr_pred_model,y=score.imatd0.d2-score.il3d0.d2,label=species))+geom_text()

#Plotting Imatinib D2 vs D4 Adjusted
ggplot(resmuts_merged%>%filter(!protein_start%in%317),aes(x=netgr_pred_model,y=score.imatd2.d4-score.il3d2.d4,label=species))+geom_text()
ggplot(resmuts_merged%>%filter(!protein_start%in%317),aes(x=netgr_pred_model,y=-netgr_obs.imatd2.d4+netgr_obs.il3d2.d4+.055,label=species))+geom_text()+geom_abline()

#Plotting Imatinib D0 vs D4 Adjusted
ggplot(resmuts_merged%>%filter(!protein_start%in%317),aes(x=netgr_pred_model,y=score.imatd0.d4-score.il3d0.d4,label=species))+geom_text()
ggplot(resmuts_merged%>%filter(!protein_start%in%317),aes(x=netgr_pred_model,y=netgr_obs.imatd0.d4-netgr_obs.il3d0.d4,label=species))+geom_text()+geom_abline()
# Also plot IL3 D4 D2 scores vs iL3 D2 D0 scores
# Also update imatininb scores by considering without drug (fitness in the absence of iL3). Do these scores match up better with the IC50 predictions?
# Also, when looking at the distribution of enrichment scores, color the multiple nucleotide variants

```

Looking at how well the observed and predicted growth rates match up
```{r}
ic50data_all_sum2=read.csv("output/ic50data_all_conc.csv",row.names = 1)
        ic50data_all_sum2=ic50data_all_sum2%>%filter(conc%in%.7)%>%rename(species=mutant)%>%dplyr::select(species,drug_effect)
        resmuts_merged=merge(resmuts,ic50data_all_sum2,by="species")
        #Looking at D2 D4
        ggplot(resmuts_merged%>%filter(!protein_start%in%317,!species%in%"V299L"),aes(x=.055-drug_effect,y=.055-(-netgr_obs.imatd2.d4+netgr_obs.il3d2.d4),label=species))+geom_text()+geom_abline()
        ggplot(resmuts_merged%>%filter(!protein_start%in%317,!species%in%"V299L"),aes(x=.055-drug_effect,y=netgr_obs.imatd2.d4,label=species))+geom_text()+geom_abline()
        # ggplot(resmuts_merged%>%filter(!protein_start%in%317,!species%in%"V299L"),aes(x=netgr_obs.il3d0.d2-drug_effect,y=netgr_obs.imatd0.d2,label=species))+geom_text()+geom_abline()
        
        #Looking at D0 D4
        ggplot(resmuts_merged%>%filter(!protein_start%in%317,!species%in%"V299L"),aes(x=.055-drug_effect,y=.055-(-netgr_obs.imatd0.d4+netgr_obs.il3d0.d4),label=species))+geom_text()+geom_abline()
        ggplot(resmuts_merged%>%filter(!protein_start%in%317,!species%in%"V299L"),aes(x=.055-drug_effect,y=netgr_obs.imatd0.d4,label=species))+geom_text()+geom_abline()
        # resmuts_merged[netgr_obs.imat]=resmuts_merged%>%mutate(drugfit=)
        ggplot(resmuts_merged%>%filter(!protein_start%in%317,!species%in%"V299L"),aes(x=netgr_obs.il3d0.d4-drug_effect,y=netgr_obs.imatd0.d4,label=species))+geom_text()+geom_abline()
        # ggplot(resmuts_merged%>%filter(!protein_start%in%317),aes(x=.025-drug_effect,y=netgr_obs.imatd2.d4,label=species))+geom_text()+geom_abline()
        
        ggplot(resmuts,aes(x=species,y=netgr_obs.il3d0.d2))+geom_col()
        ggplot(resmuts,aes(x=species,y=netgr_obs.il3d2.d4))+geom_col()
        ggplot(resmuts,aes(x=species,y=netgr_obs.imatd0.d2))+geom_col()
        ggplot(resmuts%>%filter(!(protein_start%in%252&alt%in%"T")),aes(x=species,y=netgr_obs.imatd2.d4))+geom_col()
        
        
        ggplot(resmuts,aes(x=species,y=score.il3d0.d2))+geom_col()
        ggplot(resmuts,aes(x=species,y=score.il3d2.d4))+geom_col()
        ggplot(resmuts,aes(x=species,y=score.imatd0.d2))+geom_col()
        ggplot(resmuts%>%filter(!protein_start%in%317),aes(x=species,y=score.imatd2.d4))+geom_col()
        
a=resmuts_merged%>%filter(!netgr_obs.imatd0.d4%in%c(NA,"-Inf"),!netgr_obs.il3d0.d4%in%c(NA,"-Inf"))
a=a%>%mutate(netgr_pred_mean=.055-drug_effect,netgr_obs=.055-(netgr_obs.il3d0.d4-netgr_obs.imatd0.d4))
a=a%>%mutate(netgr_pred_mean=.055-drug_effect,netgr_obs=netgr_obs.imatd0.d4)
a=a%>%mutate(netgr_pred_mean=netgr_obs.il3d0.d4-drug_effect,netgr_obs=netgr_obs.imatd0.d4)
cor(a$netgr_pred_mean,a$netgr_obs)
```


How many unique mutants are there per residue in the IL3D0 library
```{r}
calls_sum=il3D0%>%filter(consequence_terms%in%"missense_variant")%>%group_by(protein_start)%>%summarize(unique_mutants=n(),count=sum(ct))

calls_sum=calls_sum%>%mutate(region=case_when(protein_start<250&protein_start>=242~1,
                                              protein_start<258&protein_start>=250~2,
                                              protein_start<266&protein_start>=258~3,
                                              protein_start<274&protein_start>=266~4,
                                              protein_start<282&protein_start>=274~5,
                                              protein_start<290&protein_start>=282~6,
                                              protein_start<298&protein_start>=290~7,
                                              protein_start<306&protein_start>=298~8,
                                              protein_start<314&protein_start>=306~9,
                                              protein_start<322&protein_start>=314~10,
                                              protein_start<330&protein_start>=322~11,
                                              protein_start<338&protein_start>=330~12,
                                              protein_start<346&protein_start>=338~13,
                                              protein_start<354&protein_start>=346~14,
                                              protein_start<362&protein_start>=354~15,
                                              protein_start<370&protein_start>=362~16,
                                              protein_start<378&protein_start>=370~17,
                                              protein_start<386&protein_start>=378~18,
                                              protein_start<394&protein_start>=386~19,
                                              protein_start<402&protein_start>=394~20,
                                              protein_start<410&protein_start>=402~21,
                                              protein_start<418&protein_start>=410~22,
                                              protein_start<426&protein_start>=418~23,
                                              protein_start<434&protein_start>=426~24,
                                              protein_start<442&protein_start>=434~25,
                                              protein_start<450&protein_start>=442~26,
                                              protein_start<458&protein_start>=450~27,
                                              protein_start<466&protein_start>=458~28,
                                              protein_start<474&protein_start>=466~29,
                                              protein_start<482&protein_start>=474~30,
                                              protein_start<490&protein_start>=482~31,
                                              protein_start<498&protein_start>=490~32,
                                              T~0))
getPalette = colorRampPalette(brewer.pal(33, "Set2"))
plotly=ggplot(calls_sum%>%filter(protein_start>=242,protein_start<=494),aes(x=protein_start,y=unique_mutants))+geom_col(color="black",aes(fill=factor(region)))+cleanup+scale_fill_manual(values=getPalette(33))
ggplotly(plotly)

getPalette = colorRampPalette(brewer.pal(33, "Set2"))
plotly=ggplot(calls_sum%>%filter(protein_start>=242,protein_start<=494),aes(x=protein_start,y=count))+geom_col(color="black",aes(fill=factor(region)))+cleanup+scale_fill_manual(values=getPalette(33))
ggplotly(plotly)


il3D0=il3D0%>%mutate(region=case_when(protein_start<250&protein_start>=242~1,
                                              protein_start<258&protein_start>=250~2,
                                              protein_start<266&protein_start>=258~3,
                                              protein_start<274&protein_start>=266~4,
                                              protein_start<282&protein_start>=274~5,
                                              protein_start<290&protein_start>=282~6,
                                              protein_start<298&protein_start>=290~7,
                                              protein_start<306&protein_start>=298~8,
                                              protein_start<314&protein_start>=306~9,
                                              protein_start<322&protein_start>=314~10,
                                              protein_start<330&protein_start>=322~11,
                                              protein_start<338&protein_start>=330~12,
                                              protein_start<346&protein_start>=338~13,
                                              protein_start<354&protein_start>=346~14,
                                              protein_start<362&protein_start>=354~15,
                                              protein_start<370&protein_start>=362~16,
                                              protein_start<378&protein_start>=370~17,
                                              protein_start<386&protein_start>=378~18,
                                              protein_start<394&protein_start>=386~19,
                                              protein_start<402&protein_start>=394~20,
                                              protein_start<410&protein_start>=402~21,
                                              protein_start<418&protein_start>=410~22,
                                              protein_start<426&protein_start>=418~23,
                                              protein_start<434&protein_start>=426~24,
                                              protein_start<442&protein_start>=434~25,
                                              protein_start<450&protein_start>=442~26,
                                              protein_start<458&protein_start>=450~27,
                                              protein_start<466&protein_start>=458~28,
                                              protein_start<474&protein_start>=466~29,
                                              protein_start<482&protein_start>=474~30,
                                              protein_start<490&protein_start>=482~31,
                                              protein_start<498&protein_start>=490~32,
                                              T~0))

getPalette = colorRampPalette(brewer.pal(33, "Set2"))
plotly=ggplot(il3D0%>%filter(protein_start>=242,protein_start<=494),aes(x=protein_start,y=ct))+geom_col(color="black",aes(fill=factor(region)))+cleanup+scale_fill_manual(values=getPalette(33))
ggplotly(plotly)
```


#Next: plot iL3 D2 D4 scores
#Correlate Il3 D2 D4 scores with il3 D0 D2 scores

Are any of the enriched mutants in the cosmic somatic database?
```{r}
# rm(list=ls())
cosmic_data=read.table("data/Cosmic_ABL/ABL_Cosmic_Gene_mutations.tsv",sep="\t",header = T,stringsAsFactors = )
cosmic_data=cosmic_data%>%mutate(AA.Mutation=gsub("p.","",AA.Mutation))
cosmic_data=cosmic_data[!grepl("ins|del",cosmic_data$CDS.Mutation),]
cosmic_data=cosmic_data[grepl("Missense",cosmic_data$Type),]
cosmic_data=cosmic_data%>%filter(!Type%in%"Substitution - coding silent")
cosmic_data=cosmic_data%>%filter(Position<=500,Position>=64,Count>=2)
# write.csv(cosmic_data,"cosmic_abl.csv")
cosmic_simple=cosmic_data%>%dplyr::select(subs_name=AA.Mutation,cosmic_count=Count)
cosmic_simple=cosmic_simple%>%group_by(subs_name)%>%summarize(cosmic_count=sum(cosmic_count))
cosmic_simple$cosmic_present=T
```


```{r}
source("code/merge_samples.R")
sample10=read.csv(file = "data/Consensus_Data/Novogene_lane14/sample10_combined/sscs/variant_caller_outputs/variants_unique_ann.csv",header=T,stringsAsFactors = F)

sample10=sample10%>%
  rowwise()%>%
  mutate(ref_aa=strsplit(amino_acids,"/")[[1]][1],
         alt_aa=strsplit(amino_acids,"/")[[1]][2])
sample10=sample10%>%mutate(maf=ct/depth)
sample10_simple=sample10%>%dplyr::select(alt_start_pos,protein_start,ref,alt,ref_aa,alt_aa,consequence_terms,ct,depth,maf)
sample10=il3D0
sample12=read.csv(file = "data/Consensus_Data/Novogene_lane14/sample12/sscs/variant_caller_outputs/variants_unique_ann.csv",header=T,stringsAsFactors = F)

sample12=sample12%>%
  rowwise()%>%
  mutate(ref_aa=strsplit(amino_acids,"/")[[1]][1],
         alt_aa=strsplit(amino_acids,"/")[[1]][2])
sample12=sample12%>%mutate(maf=ct/depth)
sample12=imatD4
sample12_simple=sample12%>%dplyr::select(alt_start_pos,protein_start,ref,alt,ref_aa,alt_aa,consequence_terms,ct,depth,maf)

samples_10.12=merge(sample10_simple%>%filter(consequence_terms%in%"missense_variant"),sample12_simple%>%filter(consequence_terms%in%"missense_variant"),by=c("ref_aa","protein_start","alt_aa","ref","alt","alt_start_pos","consequence_terms"))

samples_10.12=samples_10.12%>%mutate(score=log2(maf.y/maf.x),score2=log2(ct.y/ct.x))
samples_10.12_simple=samples_10.12%>%dplyr::select(ref_aa,protein_start,alt_aa,consequence_terms,ct.x,score)

ggplot(samples_10.12,aes(x=score))+geom_density()

# ggplot(samples_14.16,aes(x=score))+geom_histogram(bins=100)

samples_10.12=samples_10.12%>%mutate(resmuts=case_when(protein_start%in%253&alt_aa%in%"H"~T,
                                                         protein_start%in%255&alt_aa%in%"V"~T,
                                                         protein_start%in%486&alt_aa%in%"S"~T,
                                                         protein_start%in%396&alt_aa%in%"P"~T,
                                                         protein_start%in%255&alt_aa%in%"K"~T,
                                                         protein_start%in%315&alt_aa%in%"I"~T,
                                                         protein_start%in%252&alt_aa%in%"H"~T,
                                                         protein_start%in%253&alt_aa%in%"F"~T,
                                                         protein_start%in%250&alt_aa%in%"E"~T,
                                                         protein_start%in%359&alt_aa%in%"C"~T,
                                                         protein_start%in%351&alt_aa%in%"T"~T,
                                                         protein_start%in%355&alt_aa%in%"G"~T,
                                                         protein_start%in%317&alt_aa%in%"L"~T,
                                                         protein_start%in%359&alt_aa%in%"I"~T,
                                                         protein_start%in%355&alt_aa%in%"A"~T,
                                                         protein_start%in%459&alt_aa%in%"K"~T,
                                                         protein_start%in%276&alt_aa%in%"G"~T,
                                                         protein_start%in%299&alt_aa%in%"L"~T,
                                                         
                                                         T~F))

samples_10.12=samples_10.12%>%mutate(resresids=case_when(protein_start%in%253~T,
                                                         protein_start%in%255~T,
                                                         protein_start%in%486~T,
                                                         protein_start%in%396~T,
                                                         protein_start%in%255~T,
                                                         protein_start%in%315~T,
                                                         protein_start%in%252~T,
                                                         protein_start%in%253~T,
                                                         protein_start%in%250~T,
                                                         protein_start%in%359~T,
                                                         protein_start%in%351~T,
                                                         protein_start%in%355~T,
                                                         protein_start%in%317~T,
                                                         protein_start%in%359~T,
                                                         protein_start%in%355~T,
                                                         protein_start%in%459~T,
                                                         protein_start%in%276~T,
                                                         protein_start%in%299~T,
                                                         
                                                         T~F))

highscore=samples_10.12%>%filter(!ct.x%in%1,protein_start>=242,protein_start<=494)
highscore=highscore%>%mutate(subs_name=paste(ref_aa,protein_start,alt_aa,sep = ""))
ggplot(highscore,aes(x=reorder(subs_name,-score),y=score,fill=resmuts))+geom_col()+theme(axis.text.x=element_text(angle=90, hjust=1))+scale_y_continuous(name="Enrithcment Score")+scale_x_discrete(name="Mutant")+guides(fill = guide_legend(title = "Clinically\n Observed \n Resmut"))+scale_fill_manual(values=c("gray90","red"))
highscore2=highscore%>%group_by(alt_aa,protein_start,subs_name,resmuts,resresids)%>%summarize(score=mean(score))

ggplot(highscore2,aes(x=reorder(subs_name,-score),y=score,fill=resmuts))+geom_col()+theme(axis.text.x=element_blank(),axis.ticks.x=element_blank())+scale_y_continuous(name="Enrithcment Score")+scale_x_discrete(name="Mutant")+guides(fill = guide_legend(title = "Clinically\n Observed \n Resistant Mutant"))+scale_fill_manual(values=c("gray90","red"))
# ggsave("BCRABL_Imatinib_Scores_Resmuts.pdf",height=4,width=8,units="in",useDingbats=F)

ggplot(highscore2,aes(x=reorder(subs_name,-score),y=score,fill=resresids))+geom_col()+theme(axis.text.x=element_blank(),axis.ticks.x=element_blank())+scale_y_continuous(name="Enrithcment Score")+scale_x_discrete(name="Mutant")+guides(fill = guide_legend(title = "Clinically\n Observed \n Resistant Residue"))+scale_fill_manual(values=c("gray90","red"))
# ggsave("BCRABL_Imatinib_Scores_Resresids.pdf",height=4,width=8,units="in",useDingbats=F)

###Merging cosmic data and highscore
# highscore$cosmic_present=F
highscore_cosmic=merge(highscore,cosmic_simple,by="subs_name",all.x = T)
highscore_cosmic[highscore_cosmic$cosmic_present%in%NA,"cosmic_present"]=F
highscore_cosmic[highscore_cosmic$cosmic_count%in%NA,"cosmic_count"]=0

plotly=ggplot(highscore_cosmic,aes(x=reorder(subs_name,-score),y=score,fill=cosmic_present))+geom_col()+theme(axis.text.x=element_text(angle=90, hjust=1))+scale_y_continuous(name="Enrichment Score")+scale_x_discrete(name="Mutant")+guides(fill = guide_legend(title = "Cosmic\n Observed"))
ggplotly(plotly)
a=cosmic_data%>%filter(AA.Mutation%in%"L248V")
a=highscore%>%filter(subs_name%in%"M388L")


plotly=ggplot(highscore_cosmic%>%filter(cosmic_present%in%T),aes(x=reorder(subs_name,-score),y=score,fill=cosmic_count))+geom_col()+theme_bw()
ggplotly(plotly)
# a=highscore_cosmic%>%filter(cosmic_present%in%T)
# a=highscore2%>%filter(resmuts%in%T)
```


