---
title: "growthrate_plots"
author: "Haider Inam"
date: "4/2/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir=rprojroot::find_rstudio_root_file())
```

####Please change required directories this chunk if compiling in R rather than RmD
```{r}
#Inputs:
conc_for_predictions=0.8
net_gr_wodrug=0.05
#Reading required tables
ic50data=read.csv("data/heatmap_concat_data.csv",header = T,stringsAsFactors = F)
# ic50data=read.csv("../data/heatmap_concat_data.csv",header = T,stringsAsFactors = F)

twinstrand_maf=read.table("data/Twinstrand/prj00053-2019-12-02.deliverables/all.mut",sep="\t",header = T,stringsAsFactors = F)
# twinstrand_maf=read.table("../data/Twinstrand/prj00053-2019-12-02.deliverables/all.mut",sep="\t",header = T,stringsAsFactors = F)

names=read.table("data/Twinstrand/prj00053-2019-12-02.deliverables/manifest.tsv",sep="\t",header = T,stringsAsFactors = F)
# names=read.table("../data/Twinstrand/prj00053-2019-12-02.deliverables/manifest.tsv",sep="\t",header = T,stringsAsFactors = F)

# twinstrand_maf_merge=read.csv("../output/twinstrand_maf_merge.csv",header = T,stringsAsFactors = F)
twinstrand_maf_merge=read.csv("output/twinstrand_maf_merge.csv",header = T,stringsAsFactors = F)

# twinstrand_simple_melt_merge=read.csv("../output/twinstrand_simple_melt_merge.csv",header = T,stringsAsFactors = F)
twinstrand_simple_melt_merge=read.csv("output/twinstrand_simple_melt_merge.csv",header = T,stringsAsFactors = F)

# ic50data_long=read.csv("../output/ic50data_all_conc.csv",header = T,stringsAsFactors = F)
ic50data_long=read.csv("output/ic50data_all_conc.csv",header = T,stringsAsFactors = F)
ic50data_long$netgr_pred=net_gr_wodrug-ic50data_long$drug_effect
```

```{r,include=F}
# rm(list=ls())
library(knitr)
library(tictoc)
library(workflowr)
library(VennDiagram)
library(dplyr)
library(foreach)
library(doParallel)
library(ggplot2)
library(reshape2)
library(RColorBrewer)
library(devtools)
library(ggsignif)
library(plotly)
library(BiocManager)
library(drc)
# library(ensembldb) #Loading this with Dplyr commands seems to throw an error in Rmd
# library(EnsDb.Hsapiens.v86) #Loading this with Dplyr commands seems to throw an error in Rmd

######################Cleanup for GGPlot2#########################################
cleanup=theme_bw() +
  theme(plot.title = element_text(hjust=.5),
        panel.grid.major = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(color = "black"),
        axis.text = element_text(face="bold",color="black",size="11"),
        text=element_text(size=11,face="bold"),
        axis.title=element_text(face="bold",size="11"))
```



###Plotting 1:1000 data
```{r}
ggplot(twinstrand_simple_melt_merge%>%filter(experiment%in%c("M3","M5","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(mutant),label=mutant))+geom_text()+geom_abline()+facet_wrap(~"experiment")+cleanup

plotly=ggplot(twinstrand_simple_melt_merge%>%filter(experiment%in%c("M3","M5","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(experiment),label=factor(mutant)))+geom_text()+geom_abline()+cleanup
ggplotly(plotly)

#trying out the dotplot verison to show the individual datapoints
plotly=ggplot(twinstrand_simple_melt_merge%>%filter(experiment%in%c("M3","M5","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(mutant)))+geom_boxplot(position=position_dodge(1))+geom_jitter(shape=16, position=position_jitter())+geom_abline()+cleanup
ggplotly(plotly)

###Replicate M5 seems to be a little underapproximative. Only M7 M3 look better.
plotly=ggplot(twinstrand_simple_melt_merge%>%filter(experiment%in%c("M3","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(mutant)))+geom_boxplot(position=position_dodge(1))+geom_jitter(shape=16, position=position_jitter())+geom_abline()+cleanup
ggplotly(plotly)

```
Analyzing the %age of reads used up by the most resistant mutants
```{r}
t315=twinstrand_maf_merge%>%filter(mutant%in%c("T315I","Y253H","E255V","G250E"))%>%mutate(percentage=AltDepth*100/Depth)
wasted_reads=t315%>%group_by(experiment,time_point,mutant)%>%summarize(percentage=percentage)
ggplot(wasted_reads%>%filter(!time_point%in%"D0"),aes(x=mutant,y=percentage,fill=mutant))+geom_col(position=position_dodge(1))+facet_grid(time_point~experiment)+cleanup+theme(axis.text.x=element_text(angle=90))
# ggsave("wasted_reads_bymutant.pdf",width=6,height=3,units="in")

wasted_reads_sum_top4=wasted_reads%>%group_by(experiment,time_point)%>%summarize(percentage=sum(percentage))
ggplot(wasted_reads_sum_top4,aes(x=time_point,y=percentage,fill=time_point))+geom_col(position=position_dodge(1))+facet_wrap(~experiment)+cleanup+scale_y_continuous(name = "Percentage of reads taken \n up by top 4 mutants")
# ggsave("wasted_reads_total_top4.pdf",width=4,height=3,units="in")

wasted_reads_sum_top2=wasted_reads%>%filter(mutant%in%c("T315I","Y253H"))%>%group_by(experiment,time_point)%>%summarize(percentage=sum(percentage))
ggplot(wasted_reads_sum_top2,aes(x=time_point,y=percentage,fill=time_point))+geom_col(position=position_dodge(1))+facet_wrap(~experiment)+cleanup+scale_y_continuous(name = "Percentage of reads taken \n up by top 2 mutants")
ggsave("wasted_reads_total_top2.pdf",width=4,height=3,units="in")


######Looking at the scenario where the top 4 mutants weren't there in the ENU experiments:
###Percentage of wasted reads (top 4 mutants) for both enu experiments were ~60% for D3 and ~83% for D6
# a=twinstrand_maf_merge%>%filter(experiment%in%)
maf_without_wasted_reads=twinstrand_maf_merge%>%filter(!mutant%in%c(NA,"T315I","Y253H","G250E"),experiment%in%"Enu_4")%>%group_by(time_point)%>%mutate(Percentage_new=AltDepth*100/sum(AltDepth))
ggplot(maf_without_wasted_reads,aes(x=mutant,y=Percentage_new,fill=mutant))+geom_col(position=position_dodge(1))+facet_wrap(~time_point,ncol=1)+cleanup+theme(axis.text.x=element_text(angle=90))

# a=twinstrand_maf_merge%>%filter(!mutant%in%c(NA,"T315I","Y253H","G250E"),experiment%in%"Enu_4",time_point=="D0")
# sum(a$AltDepth)

maf_without_wasted_reads$mutant=factor(maf_without_wasted_reads$mutant,levels = as.character(unique(maf_without_wasted_reads$mutant[order((maf_without_wasted_reads$AltDepth),decreasing = T)])))

getPalette = colorRampPalette(brewer.pal(9, "Spectral"))


ggplot(maf_without_wasted_reads%>%filter(time_point=="D3"),aes(x="",y=Percentage_new,fill=mutant))+geom_bar(width=1,stat="identity")+cleanup+theme(axis.text.x=element_text(angle=90),axis.title = element_blank())+coord_polar("y",start=0)+scale_fill_manual(values = getPalette(length(unique(maf_without_wasted_reads$mutant))))+ggtitle("With Mutants Removed")
ggsave("maf_piechart_new.pdf",width=4,height=4,units="in")




maf_with_wasted_reads=twinstrand_maf_merge%>%filter(experiment%in%"Enu_4",!mutant%in%NA)%>%group_by(time_point)%>%mutate(Percentage=AltDepth*100/sum(AltDepth))
# a=twinstrand_maf_merge%>%filter(experiment%in%"Enu_4",!mutant%in%NA,time_point=="D3")


maf_with_wasted_reads$mutant=factor(maf_with_wasted_reads$mutant,levels = as.character(unique(maf_with_wasted_reads$mutant[order((maf_with_wasted_reads$AltDepth),decreasing = T)])))

ggplot(maf_with_wasted_reads%>%filter(time_point=="D3"),aes(x="",y=Percentage,fill=mutant))+geom_bar(width=1,stat="identity")+cleanup+theme(axis.text.x=element_text(angle=90),axis.title = element_blank())+coord_polar("y",start=0)+scale_fill_manual(values = getPalette(length(unique(maf_with_wasted_reads$mutant))))+ggtitle("Without Removing Mutants")
ggsave("maf_piechart_old.pdf",width=4,height=4,units="in")

maf_before_after=merge(maf_with_wasted_reads%>%filter(time_point=="D3")%>%ungroup()%>%dplyr::select(mutant,Percentage),maf_without_wasted_reads%>%filter(time_point=="D3")%>%ungroup()%>%dplyr::select(mutant,Percentage_new),by="mutant",all.x = T,all.y = T)
maf_before_after=maf_before_after%>%mutate(Percentage_new=case_when(Percentage_new%in%NA~0,
                                                     T~Percentage_new))
maf_before_after=maf_before_after%>%dplyr::select(mutant,NotSkipped=Percentage,Skipped=Percentage_new)
maf_before_after_melt=melt(maf_before_after,id.vars = "mutant",measure.vars = c("NotSkipped","Skipped"),variable.name = "Status",value.name = "Percentage")


ggplot(maf_before_after_melt%>%filter(!mutant%in%c("T315I","Y253H","G250E")),aes(x=mutant,y=Percentage,fill=Status))+geom_col(position=position_dodge(1))+cleanup+theme(axis.text.x = element_text(angle=90),legend.position = "bottom")
ggsave("Residue_Skipping_Comparison.pdf",width=4,height=3,units="in")


```
```{r}
twinstrand_maf=read.table("data/Twinstrand/prj00053-2019-12-02.deliverables/all.mut",sep="\t",header = T,stringsAsFactors = F)
# wt=twinstrand_maf_merge%>%filter(Sample%in%c("dna00762"))
wt=twinstrand_maf%>%filter(Sample%in%c("dna00762"))
wt_snps=read.table("/Volumes/MyPassportforMac/duplexsequencing/twinstrand_september2020/wt2/variants_SNPs_filtered_sorted.tsv",header = F,stringsAsFactors = F)

ggplot(wt_snps,aes(x=V1,y=V2*100/20000))+geom_col(width=3,position = position_dodge(1),fill="red")+cleanup+scale_x_continuous(name="Position in ABL1 Kinase Domain")+scale_y_continuous(name="Mutant Fraction %",limits=c(0,.15))+ggtitle("Minor Alleles Standard Sequencing")
ggsave("wt_standardsequencing.pdf",width=3,height=2,units="in",useDingbats=F)

ggplot(twinstrand_maf%>%filter(Sample=="dna00762",!VariationType%in%c("indel"),!Start%in%c("130872199")),aes(x=Start,y=AltDepth*100/Depth))+geom_col(width=10,position=position_dodge(1),fill="red")+cleanup+scale_x_continuous(name="Position in ABL1 Kinase Domain")+scale_y_continuous(name="Mutant Fraction %",limits=c(0,.15))+ggtitle("Minor Alleles Duplex Sequencing")
ggsave("wt_duplexconsensus.pdf",width=3,height=2,units="in",useDingbats=F)
# a=twinstrand_maf%>%filter(Sample=="dna00762",!VariationType%in%c("indel"))

ggplot(twinstrand_maf%>%filter(Sample=="dna00763",!VariationType%in%c("indel"),!Start%in%c("130872199")),aes(x=Start,y=AltDepth*100/Depth))+geom_col(width=10,position=position_dodge(1),fill="red")+cleanup+scale_x_continuous(name="Position in ABL1 Kinase Domain")+scale_y_continuous(name="Mutant Fraction %",limits=c(0,.15))+ggtitle("Minor Alleles Duplex Sequencing")
ggsave("m3d0_duplexconsensus.pdf",width=3,height=2,units="in",useDingbats=F)
# a=twinstrand_maf%>%filter(Sample=="dna00762",!VariationType%in%c("indel"))
# a=twinstrand_maf%>%filter(Sample=="dna00763",!VariationType%in%c("indel"),!Start%in%c("130872199"))


```

###Plotting 1 in 5,000
```{r}
ggplot(twinstrand_simple_melt_merge%>%filter(experiment%in%c("M3","M4","M5","M6","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(mutant),label=mutant))+geom_text()+geom_abline()+facet_wrap(~"experiment")+cleanup

plotly=ggplot(twinstrand_simple_melt_merge%>%filter(experiment%in%c("M3","M4","M5","M6","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(experiment),label=factor(mutant)))+geom_text()+geom_abline()+cleanup
ggplotly(plotly)

#trying out the dotplot verison to show the individual datapoints
plotly=ggplot(twinstrand_simple_melt_merge%>%filter(experiment%in%c("M3","M4","M5","M6","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(mutant)))+geom_boxplot(position=position_dodge(1))+geom_jitter(shape=16, position=position_jitter())+geom_abline()+cleanup
ggplotly(plotly)

#Removing M5 and M6 makes these look better. My assumption is that things were growing slower for the M5, M6 conditions because of culture conditions.
#This may look bad. But this also represents an opportunity: You don't need to estimate the approximate growth rate in the absence of drug because it is dynamic (changing culture conditions etc). You could use counts from a no-drug control but even that isn't the best control because of plate to plate variability in growth rates. You could just look at the most resistant mutants and see how fast they're growing. From their growth rates, you can back-calculate what the exact w/o drug growth rate is.
plotly=ggplot(twinstrand_simple_melt_merge%>%filter(experiment%in%c("M3","M4","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(mutant)))+geom_boxplot(position=position_dodge(1))+geom_jitter(shape=16, position=position_jitter())+geom_abline()+cleanup
ggplotly(plotly)

#As suggested above, I will modify the observed growth rate so that the w/o drug growth rate essentially derives from the observed growth rate of the most resistant cell line. Please note that I am doing this in a very roundabout way here because I am looking at the growth rate of T315I in the slow-growing cell lines, comparing it to how slow it is growing vs the mean growth rate of T315I from the other experimental days, and adding that difference in growth rate to that slow-growing experimental day.
# library(dplyr)
a=twinstrand_simple_melt_merge%>%
  mutate(netgr_obs=case_when(experiment=="M5"~netgr_obs+.015,
                                   experiment%in%c("M3","M6","M5","M4","M7")~netgr_obs))

plotly=ggplot(a%>%filter(experiment%in%c("M3","M4","M5","M6","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(experiment),label=factor(mutant)))+geom_text()+geom_abline()+cleanup
ggplotly(plotly)
a_new=a%>%filter(experiment%in%c("M3"),duration%in%("d3d6"))
plotly=ggplot(a%>%filter(experiment%in%c("M3","M4","M5","M6","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(mutant)))+geom_boxplot(position=position_dodge(1))+geom_jitter(shape=16, position=position_jitter())+geom_abline()+cleanup
ggplotly(plotly)

plotly=ggplot(a%>%filter(experiment%in%c("M3","M4","M5","M6","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(mutant)))+geom_boxplot(position=position_dodge(1))+geom_jitter(shape=16, position=position_jitter())+geom_abline()+cleanup
ggplotly(plotly)
```
##Plotting 1 in 5000 data
Note that this is somewhat similar to what twinstrand sent to us.
```{r}

ggplot(twinstrand_maf_merge%>%filter(tki_resistant_mutation=="True",!mutant=="NA",Spike_in_freq==1000,!experiment%in%c("Enu_3","Enu_4")),aes(x=time_point,y=AltDepth,color=factor(experiment)))+geom_point()+facet_wrap(~mutant)+cleanup

ggplot(twinstrand_maf_merge%>%filter(tki_resistant_mutation=="True",!mutant=="NA",Spike_in_freq==1000,!experiment%in%c("Enu_3","Enu_4")),aes(x=time_point,y=totalmutant,color=factor(experiment)))+geom_point()+facet_wrap(~mutant)+scale_y_continuous(trans = "log10")+cleanup


ggplot(twinstrand_maf_merge%>%filter(tki_resistant_mutation=="True",!mutant=="NA",Spike_in_freq==1000,!experiment%in%c("Enu_3","Enu_4")),aes(x=time_point,y=totalmutant,color=factor(experiment)))+geom_point()+facet_wrap(~mutant)+cleanup

#Combining 1 to 1,000 and 1 to 5,000
ggplot(twinstrand_maf_merge%>%filter(tki_resistant_mutation=="True",!mutant=="NA",!experiment%in%c("Enu_3","Enu_4")),aes(x=time_point,y=totalmutant,color=factor(experiment),shape=factor(Spike_in_freq)))+geom_point()+facet_wrap(~mutant)+scale_y_continuous(trans = "log10")+cleanup

# Looking at why 1 in 5,000 mutants look a little undersampled
#Turns out they look different because they started off with 5x less cells. Duh.
# a=twinstrand_maf_merge%>%filter(tki_resistant_mutation=="True",Spike_in_freq=="5000",experiment=="M6",mutant%in%c("T315I","G250E"))
```


Looking at whether removing ANY variants with less than 5 ALT reads makes the correlation plots look better
Answer: Yes it makes it better but mutants that defy predicitons are still there.
```{r}
twinstrand_maf_merge=twinstrand_maf_merge%>%filter(!AltDepth<=5)

twinstrand_simple=twinstrand_maf_merge%>%filter(tki_resistant_mutation=="True",!is.na(mutant),!is.na(experiment))
twinstrand_simple=twinstrand_simple%>%dplyr::select("mutant","experiment","Spike_in_freq","time_point","totalmutant")
twinstrand_simple_cast=dcast(twinstrand_simple,mutant+experiment+Spike_in_freq~time_point,value.var="totalmutant")

twinstrand_simple_cast$d0d3=log(twinstrand_simple_cast$D3/twinstrand_simple_cast$D0)/72
twinstrand_simple_cast$d3d6=log(twinstrand_simple_cast$D6/twinstrand_simple_cast$D3)/72
twinstrand_simple_cast$d0d6=log(twinstrand_simple_cast$D6/twinstrand_simple_cast$D0)/144
#Check if ln(final/initial)/time is the correct formula. Also notice how I'm using days not hours
twinstrand_simple_melt=melt(twinstrand_simple_cast[,-c(4:6)],id.vars=c("mutant","experiment","Spike_in_freq"),variable.name = "duration",value.name = "netgr_obs") 
twinstrand_simple_melt$drug_effect_obs=net_gr_wodrug-twinstrand_simple_melt$netgr_obs

# twinstrand_simple_melt_merge=merge(twinstrand_simple_melt,ic50data_formerge,"mutant")
twinstrand_simple_melt_merge=merge(twinstrand_simple_melt,ic50data_long,"mutant")
twinstrand_simple_melt_merge=twinstrand_simple_melt_merge%>%filter(conc%in%0.8)

a=twinstrand_simple_melt_merge%>%
  mutate(netgr_obs=case_when(experiment=="M5"~netgr_obs+.015,
                                   experiment%in%c("M3","M5","M6","M4","M7")~netgr_obs))


plotly=ggplot(a%>%filter(experiment%in%c("M3","M4","M5","M6","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(mutant)))+geom_boxplot(position=position_dodge(1))+geom_jitter(shape=16, position=position_jitter())+geom_abline()+cleanup
ggplotly(plotly)

plotly=ggplot(a%>%filter(experiment%in%c("M3","M4","M5","M6","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(mutant),label=mutant))+geom_text()+geom_abline()+facet_wrap(~"experiment")+cleanup
ggplotly(plotly)

plotly=ggplot(a%>%filter(experiment%in%c("M3","M4","M5","M6","M7"),duration=="d3d6"),aes(x=netgr_pred,y=netgr_obs,color=factor(experiment),label=factor(mutant)))+geom_text()+geom_abline()+cleanup
ggplotly(plotly)
```

```{r}

names=read.table("data/Twinstrand/prj00053-2019-12-02.deliverables/manifest.tsv",sep="\t",header = T,stringsAsFactors = F)
twinstrand_maf_names=merge(twinstrand_maf,names,by.x="Sample",by.y="TwinstrandId")
a=twinstrand_maf_names%>%filter(tki_resistant_mutation=="True",CustomerName%in%c("M3D0","M3D3","M3D6"))
```

