---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

# **Duplex Sequencing Spike-Ins**  

<img src="assets/ds_coverages.png" width="400" height="300"  alt="">  

* Click [here](dose_response_curve_fitting.html) for the script that reads BCRABL dose response data, and fits it to a 4-parameter dose response curve.  
<img src="assets/4param_fit.png" width="400" height="300"  alt="">    

* Click [here](twinstrand_spikeins_data_generation.html) for the script that reads mutation annotation data and generates dataframes with annotated variants and the correct cell counts  

* Click [here](growthrate_plots.html) for initial growth rate plots  
<img src="assets/growthrates.png" width="400" height="300"  alt="">  

* Click [here](growthrate_plots_ENU.html) for growth rate plots for ENU mutants  
<img src="assets/growthrates_ENU.png" width="400" height="300"  alt="">  

* Click [here](spikeins_depthofcoverages.html) for analysis looking at whether we achieved our desired depth of coverages
<img src="assets/coverages.png" width="400" height="300"  alt="">  

* Click [here](clinical_abundance_predictions.html) for analysis looking at how well our data predicts BCRABL clinical abundance compared to conventional IC50s
<img src="assets/clinical_abundance_predictions.png" width="600" height="400"  alt="">  

* Click [here](misc.html) for method to obtain confidence intervals on growth rates from IC50 measurements. Plotted alongside observed growth rate data, these confidence intervals show how well our pooled data matches predictions based off of IC50s. (Needs updates in methodology)  
<img src="assets/pooled_trajectories_confidence.png" width="400" height="300"  alt=""> 


* Click [here](4_7_20_update.html) for analyses of replicate to replicate agreement between the spike in replicates. Also includes this analysis for the mutagenesis replicates. Also includes the agreement in observed vs predicted depth of coverages for all sequencing pools. Note: this analysis does not include dosage corrections that improves replicate to replicate heterogeneity.  
<img src="assets/mutagenesis_replicates_correlation.png" width="400" height="300"  alt="">  

* Click [here](nonlinear_growth_analysis.html) to see analysis of predictions based on Enrich2 vs Shendure vs our method

* Click [here](dosing_normalization.html) for our dosing normalization strategy  
<img src="assets/dosing_normalization.png" width="330" height="250"  alt="">  

* Click [here](enrichment_simulations.html) for simulations showing enrichment and depletion events in a 3 mutant pool.
<img src="assets/enrichment_simulations_3mutants.png" width="330" height="250"  alt="">  

* Click [here](E255K_alphas_figure.html) showing the conformity between the growth rates in the presence of drug measured at a low MAF via FACs, sequencing, and via IC50 studies for a single mutant.  
<img src="assets/e255k_alphas.png" width="330" height="250"  alt="">  
  

* Click [here](ABL_SM_CRISPR_Cut_Analyses.html) to view BCRABL Imatinib heatmaps  generated using CRISPR-cut libraries
<img src="assets/ABL_SM_Crispr_heatmap.png" width="400" height="283"  alt="">  

* Click [here](lane14_comparisons.html) to view preliminary BCRABL Imatinib and Asciminib heatmaps  made with enzymatically sheared libraries
<img src="assets/bcrabl_heatmap.png" width="400" height="250"  alt="">  
  
* Click [here](DupSeq_QC_Lane11.html) and to view iL3 independence data and [here](BCRABL_FunctionalKinaseAnalysis.html) to view iL3 independence heatmap and  [here](ABL_sheared_libraries.html) to view imatinib heatmaps on enzymatically sheared data
<img src="assets/il3heatmap.png" width="800" height="200"  alt="">  
  
* Click [here](ErrorRate_Analysis.html) to view error rates without barcodes, with single strand consensi, and with duplex consensus sequencing.
<img src="assets/Error_rates.png" width="420" height="300"  alt="">  

* Click [here](K562_BCRABL_Analysis.html) to view the results of the BCRABL Library transduced in K562 Cells
<img src="assets/k562_plot.png" width="800" height="222"  alt="">  
  
* Click [here](CDF-analysis-library-evenness.html) to view analyses of the CDFs of our saturating mutagenesis libraries. These show how even or uneven the libraries are at different stages of the screening process.      
<img src="assets/unevenness_cdfs.png" width="387" height="300"  alt="">  
  
        
* Click [here](ABL_BaseEditor_Analyses.html) to view analyses in which we compared base editing and saturating mutagenesis data on ABL. This is followed by analysis in which we decide on filter cutoffs for enrichment scores to detect signficantly depleting mutants that drop out for IL3 independence in BaF3s.     
<img src="assets/abl_baseeditor-analysis.png" width="500" height="293"  alt="">  
  
    
* Click [here](ABL_BaseEditor_Analyses_v2.html) to view analyses in which we predicted admixtures of mutants made by sgRNAs in Base Editing Screens. We then used the saturating mutagenesis data to predict the fitness of these sgRNAs.     
<img src="assets/abl_baseeditor-analysis-v2.png" width="300" height="320"  alt="">  
  
* Click [here](ABL_kinasefunction_data_parser.html) to view initial kinase function analyses.  
<img src="assets/kinase_function_dist.png" width="300" height="213"  alt="">  


* Click [here](ABL_cosmic_analysis.html) to view our analyses of whether mutants in the cosmic/sanger database are resistant according to our screens.  
<img src="assets/abl_cosmic_analysis.png" width="300" height="216"  alt="">.  

  
* Click [here](ABL_cosmic_analysis_fullkinase.html) to view our analyses of imatinib resistance in the full ABL kinase. This includes heatmaps, scatterplots, boxplots, waterfall plots, all analyzing the imatinib resistance profile of the ABL kinase.  
<img src="assets/fullkinase_scatter.png" width="300" height="281"  alt="">.  


* Click [here](ABL_Gnomad_analysis.html) to view our analyses of how SNPs in Gnomad enable potentially drug resistant mutants, necessitating the study of multi-nucleotide variants in DMS screens.  
<img src="assets/gnomad_enabled.png" width="600" height="300"  alt="">.  





#### Click [here](https://github.com/haiderinam/duplex_sequencing_screen) to be redirected to the github page that contains all the data and analysis rmd files.  


What is in each directory:  
  
* __Data__: contains data downloaded prior to analyses.  
* __Output__: contains data that the code makes  
* __Analysis__: contains your Rmarkdown files with the code.  
* __Docs__: contain the html output from the Rmd files in the analysis directory.  
* __Code__: contains .R files that are functions that the Rmd files in the analysis folder use. 
