---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

# **Duplex Sequencing Spike-Ins**  

<img src="assets/ds_coverages.png" width="400" height="300"  alt="">  

* ###Click [here](twinstrand_spikeins_data_generation.html) for the script that reads mutation annotation data and generates dataframes with annotated variants and the correct cell counts  

* ###Click [here](growthrate_plots.html) for initial growth rate plots  
<img src="assets/growthrates.png" width="400" height="300"  alt="">  

* ###Click [here](growthrate_plots_ENU.html) for growth rate plots for ENU mutants
<img src="assets/growthrates_ENU.png" width="400" height="300"  alt="">  

* ###Click [here](spikeins_depthofcoverages.html) for analysis looking at whether we achieved our desired depth of coverages
<img src="assets/coverages.png" width="400" height="300"  alt="">  

* ###Click [here](clinical_abundance_predictions.html) for analysis looking at how well our data predicts BCRABL clinical abundance compared to conventional IC50s
<img src="assets/clinical_abundance_predictions.png" width="600" height="300"  alt="">  

#### Click [here](https://github.com/haiderinam/duplex_sequencing_screen) to be redirected to the github page that contains all my data.  


What is in each directory:  
  
* __Data__: contains data downloaded prior to analyses.  
* __Output__: contains data that the code makes  
* __Analysis__: contains your Rmarkdown files with the code.  
* __Docs__: contain the html output from the Rmd files in the analysis directory.  
* __Code__: contains .R files that are functions that the Rmd files in the analysis folder use. 
