---
title: "SSCSvsTileseq"
author: "Haider Inam"
date: '2024-12-23'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath(".."))

library(stringr)
library(dplyr)
library(ggplot2)
library(plotly)
library(tictoc)
library(doParallel)
library(foreach)
library(RColorBrewer)
library("ggrepel")
library(reshape2)
library(ggpubr)
```

```{r cleanup}
source("code/resmuts_adder.R")
#Cleanup code for plotting
cleanup=theme_bw() +
  theme(plot.title = element_text(hjust=.5),
        panel.grid.major = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.background = element_blank(),
        # axis.line = element_line(color = "black"),
        axis.text = element_text(face="bold",color="black",size="11"),
        text=element_text(size=11,face="bold"),
        axis.title=element_text(face="bold",size="11"))
```

This is the analysis to allow benchmarking of ABL data to tileseq data

This google doc has notes on how to interpret the google doc
```{r}
# rm(list=ls())
a=data%>%filter(is_intended%in%0,seqtech%in%"sscs",dose%in%"300nM",protein_start%in%c(242:512))
data=read.csv("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/data/TileSeqandSSCS_full_kinase_alldoses_with_lfc_corrected_netgrowths_12192024.csv",header = T,stringsAsFactors = F)

ggplot(data,aes(x=factor(protein_start),y=ct_screen1_before,fill=factor(is_intended)))+geom_boxplot()+facet_wrap(~seqtech,nrow = 2)+cleanup+scale_y_continuous(trans="log10")+theme(axis.text.x=element_text(angle=90, hjust=1))



data_snr=data%>%filter(protein_start%in%c(242:512),!protein_start%in%c(354:361))%>%group_by(seqtech,dose,is_intended,protein_start)%>%summarize(nmuts=n(),coverage_mean=mean(ct_screen1_before),depth_mean=mean(depth_screen1_before))

data_snr_cast <- data_snr %>%
  pivot_wider(
    names_from = is_intended,
    values_from = c(nmuts,coverage_mean,depth_mean),
    names_glue = "{.value}_{ifelse(is_intended == 1, 'intended', 'unintended')}"
  )
data_snr_cast=data_snr_cast%>%mutate(snr=coverage_mean_intended/coverage_mean_unintended)

ggplot(data_snr_cast%>%filter(dose%in%"300nM"),aes(x=protein_start,y=snr))+geom_point()+facet_wrap(~seqtech)+scale_y_continuous(limits=c(0,60))

ggplot(data_snr_cast%>%filter(dose%in%"300nM"),aes(x=snr,fill=seqtech))+geom_density(alpha=0.6)+scale_x_continuous("Signal to noise",trans="log10")

ggplot(data_snr_cast%>%filter(dose%in%"300nM"),aes(x=protein_start,y=nmuts_intended))+geom_point()+facet_wrap(~seqtech)

ggplot(data_snr_cast%>%filter(dose%in%"300nM"),aes(x=protein_start,y=nmuts_unintended))+geom_point()+facet_wrap(~seqtech)

ggplot(data_snr_cast%>%filter(dose%in%"300nM"),aes(x=protein_start,y=coverage_mean_unintended))+geom_point()+facet_wrap(~seqtech)+scale_y_continuous(limits=c(0,200))

ggplot(data_snr_cast%>%filter(dose%in%"300nM"),aes(x=coverage_mean_unintended/depth_mean_unintended,fill=seqtech))+geom_density(alpha=.6)+scale_x_continuous("Allele fraction of unintended mutants",trans="log10")

ggplot(data_snr_cast%>%filter(dose%in%"300nM"),aes(x=coverage_mean_intended/depth_mean_intended,fill=seqtech))+geom_density(alpha=.6)+scale_x_continuous("Allele fraction of intended mutants",trans="log10")
# data_snr_cast=resmuts_adder(data_snr_cast)
# a=data%>%filter(seqtech%in%"sscs",dose%in%"300nM",is_intended%in%1,protein_start%in%c(245))


ggplot(data%>%filter(resmuts%in%T,dose%in%"300nM",is_intended%in%1),aes(x=species,y=ct_screen1_before,fill=seqtech))+geom_point(color="black",shape=21)+geom_line()+scale_y_continuous("Day 0 Coverage of Resistant Mutant",trans="log10")+cleanup+theme(axis.text.x=element_text(angle=90,hjust=.5,vjust=.5))

ggplot(data%>%filter(resmuts%in%T,dose%in%"300nM",is_intended%in%1),aes(x=species,y=ct_screen1_after,fill=seqtech))+geom_point(color="black",shape=21)+geom_line()+scale_y_continuous("Day 6 Coverage of Resistant Mutant",trans="log10")+cleanup+theme(axis.text.x=element_text(angle=90,hjust=.5,vjust=.5))

ggplot(data%>%filter(dose%in%"300nM"),aes(x=ct_screen1_before,))

ggplot(data%>%filter(is_intended%in%1,dose%in%"300nM"),aes(x=ct_screen1_before,color=seqtech))+stat_ecdf()+scale_x_continuous("Intended mutant coverage",trans="log10")+scale_y_continuous("CDF")

ggplot(data%>%filter(is_intended%in%1,dose%in%"300nM"),aes(x=ct_screen1_after,color=seqtech))+stat_ecdf()+scale_x_continuous("Intended mutant coverage",trans="log10")+scale_y_continuous("CDF")


ggplot(data%>%filter(dose%in%"300nM"),aes(x=ct_screen1_before,fill=seqtech))+geom_histogram()+facet_wrap(seqtech~is_intended)+scale_x_continuous(trans="log10")

ggplot(data%>%filter(is_intended%in%1,dose%in%"300nM",seqtech=="sscs"),aes(x=ct_screen1_before))+geom_histogram()+scale_x_continuous(trans="log10")

ggplot(data%>%filter(is_intended%in%1,dose%in%"300nM",seqtech=="sscs"),aes(x=ct_screen1_after))+geom_histogram()+scale_x_continuous(trans="log10")

ggplot(data%>%filter(dose%in%"300nM",is_intended%in%1),aes(x=ct_screen1_after,fill=seqtech))+geom_histogram()+facet_wrap(~seqtech)+scale_x_continuous(trans="log10")


# Next steps: try to fit the poisson model and the negative binomial model to the count distributions. See if the confidence intervals of the negative binomial distributions are bigger for the SSCS data.
ggplot(data%>%filter(is_intended%in%1,dose%in%"300nM",seqtech=="sscs"),aes(x=ct_screen1_before))+geom_histogram()+scale_x_continuous(limits=c(0,500))

a=data_snr_cast%>%filter(dose%in%"300nM")

```
SSCS data
Fitting negative binomial and poissons to data
```{r,eval=F}
sscsdata=data%>%filter(is_intended%in%1,dose%in%"300nM",seqtech=="sscs")
sscsdata=sscsdata%>%filter(ct_screen1_before<=5000)
sum(sscsdata$ct_screen1_before)
# nb_fit=glm.nb(sscsdata$ct_screen1_before~1)
# summary(nb_fit)

# Extract parameter estimates fir begatuve binomial
nb_mu <- mean(sscsdata$ct_screen1_before)
nb_size <- nb_fit$theta  # Dispersion parameter

# Fit a Poisson model (mean only)
poisson_mu <- mean(sscsdata$ct_screen1_before)


# Generate a range of counts for plotting
count_range <- 1:5000

# Calculate negative binomial probabilities for the range
nb_probs <- dnbinom(count_range, size = nb_size, mu = nb_mu)

# Calculate probabilities for Poisson
poisson_probs <- dpois(count_range, lambda = poisson_mu)

# plot(count_range, nb_probs, type = "b", col = "red", lwd = 2)

# Create data frames for the fits
fit_df <- data.frame(
  count = rep(count_range, 2),
  density = c(nb_probs, poisson_probs),
  distribution = rep(c("Negative Binomial", "Poisson"), each = length(count_range))
)


ggplot(sscsdata)+
  geom_histogram(color="black",fill="gray90",aes(x=ct_screen1_before,y=..density..))+
  # geom_line(color="red",data=fit_df,aes(x=count,y=density))+
  # Negative binomial fit
  geom_line(
    data = fit_df[fit_df$distribution == "Negative Binomial", ],
    aes(x = count, y = density, color = distribution), size = 1
  ) +
  # Poisson fit
  geom_line(
    data = fit_df[fit_df$distribution == "Poisson", ], 
    aes(x = count, y = density, color = distribution), size = 1, linetype = "dashed"
  ) +
  # scale_y_continuous(trans="log10")+
  scale_x_continuous("SSCS Mutant coverage",limits=c(1,1000))
  

```

Tileseq data
Fitting negative binomial and poissons to data
```{r,eval=F}
sscsdata=data%>%filter(is_intended%in%1,dose%in%"300nM",seqtech=="tileseq")
sscsdata=sscsdata%>%filter(ct_screen1_before<=10000)
sum(sscsdata$ct_screen1_before)
nb_fit=glm.nb(sscsdata$ct_screen1_before~1)
summary(nb_fit)

# Extract parameter estimates fir begatuve binomial
nb_mu <- mean(sscsdata$ct_screen1_before)
nb_size <- nb_fit$theta  # Dispersion parameter

# Fit a Poisson model (mean only)
poisson_mu <- mean(sscsdata$ct_screen1_before)


# Generate a range of counts for plotting
count_range <- 1:10000

# Calculate negative binomial probabilities for the range
nb_probs <- dnbinom(count_range, size = nb_size, mu = nb_mu)

# Calculate probabilities for Poisson
poisson_probs <- dpois(count_range, lambda = poisson_mu)

# plot(count_range, nb_probs, type = "b", col = "red", lwd = 2)

# Create data frames for the fits
fit_df <- data.frame(
  count = rep(count_range, 2),
  density = c(nb_probs, poisson_probs),
  distribution = rep(c("Negative Binomial", "Poisson"), each = length(count_range))
)


ggplot(sscsdata)+
  geom_histogram(color="black",fill="gray90",aes(x=ct_screen1_before,y=..density..))+
  # geom_line(color="red",data=fit_df,aes(x=count,y=density))+
  # Negative binomial fit
  geom_line(
    data = fit_df[fit_df$distribution == "Negative Binomial", ],
    aes(x = count, y = density, color = distribution), size = 1
  ) +
  # Poisson fit
  geom_line(
    data = fit_df[fit_df$distribution == "Poisson", ], 
    aes(x = count, y = density, color = distribution), size = 1, linetype = "dashed"
  ) +
  # scale_y_continuous(trans="log10")+
  scale_x_continuous("Tileseq Mutant coverage",limits=c(1,10000))+
  scale_y_continuous(limits=c(0,.001))
  

```
Doing an NGS, SSCS, DCS, and Tileseq comparison


Adding duplex data into the mix
The following data was taken from Marta 
This is lane 18b sample 8 vs 10 duplex and tileseq data
PARSING CODE

```{r}
#Cleanup code for plotting
source("code/plotting/cleanup.R")
source("code/cosmic_data_adder.R")
source("code/compare_screens.R")
source("code/compare_screens_archive.R")
source("code/compare_screens_twosamples.R")
source("code/merge_samples_twosamples.R")
source("code/plotting/heatmap_plotting_function.R")
```
### Data parsing  
```{r, eval=F}
# getwd()
# rm(list=ls())
cellcounts_matrix_dir="data/Consensus_Data/novogene_lane18b_rerun/TwistRegion1Screen_CellCounts_Matrix.csv"
comparisons=read.csv("data/Consensus_Data/novogene_lane18b_rerun/TwistRegion1Screen_Comparisons_Todo.csv")
# comparisons=comparisons%>%filter(Completed%in%"FALSE")

for(i in 1:nrow(comparisons)){
# for(i in 1:1){
  # i=4
  dirname=comparisons$dirname[i]
  pathname=paste("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/",dirname,sep = "")
  # Create directory if it doesn't already exist
  if (!file.exists(pathname)){
    dir.create(pathname)
} 
before_screen1_identifier=unlist(strsplit(comparisons$before_screen1_identifier[i],","))
after_screen1_identifier=unlist(strsplit(comparisons$after_screen1_identifier[i],","))
before_screen2_identifier=unlist(strsplit(comparisons$before_screen2_identifier[i],","))
after_screen2_identifier=unlist(strsplit(comparisons$after_screen2_identifier[i],","))
seqtech=unlist(strsplit(comparisons$seqtech[i],","))
  # length(after_screen1_identifier)
# screen_compare_means=compare_screens(comparisons$before_screen1_identifier[i],
#                                      comparisons$after_screen1_identifier[i],
#                                      comparisons$before_screen2_identifier[i],
#                                      comparisons$after_screen2_identifier[i])
screen_compare_means=compare_screens_archive(before_screen1_identifier,
                                     after_screen1_identifier,
                                     before_screen2_identifier,
                                     after_screen2_identifier,
                                     cellcounts_matrix_dir,
                                     seqtech)
# Adding a flag,"depleted", for when a mutant completely drops out of a screen in both replicates
screen_compare_means=screen_compare_means%>%mutate(depleted=case_when(ct_screen1_after%in%0.5&&ct_screen2_after%in%0.5~"TRUE",
                                                                      T~"FALSE"))



screen_compare_means_forexport=apply(screen_compare_means,2,as.character)
write.csv(screen_compare_means_forexport,file = paste(pathname,"/screen_comparison_",dirname,".csv",sep=""))
# pathname
start_site=240
end_site=322
# Plot 1. What does the heatmap look like from the average of the net growth rate?
heatmap_plotting_function(screen_compare_means%>%filter(is_intended%in%1),242,321,fill_variable = "netgr_obs_mean",fill_name = "Net growth rate")
# ggsave(paste(pathname,"/plot1_heatmap.pdf",sep=""),width=10,height=6,units="in",useDingbats=F)

# Plot 2a: What do the correlations look like for net growth rate (show mutants in text)?

ggplot(screen_compare_means%>%filter(is_intended%in%1),aes(x=netgr_obs_screen1,y=netgr_obs_screen2,color=resmuts,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(color="Known\nResistant\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")
# ggsave(paste(pathname,"/plot2a_Netgrowthrate_correlations_text.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)

# Plot 2b: What do the correlations look like for enrichment scores (show mutants in points)?
ggplot(screen_compare_means%>%filter(is_intended%in%1),aes(x=netgr_obs_screen1,y=netgr_obs_screen2,label=species))+geom_point(color="black",shape=21,size=2,aes(fill=resmuts))+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(fill="Known\nResistant\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")
# ggsave(paste(pathname,"/plot2b_Netgrowthrate_correlations_points.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)
# Plot 2c: What do the correlations look like for enrichment scores (show mutants in text)?
ggplot(screen_compare_means%>%filter(is_intended%in%1),aes(x=score_screen1,y=score_screen2,color=resmuts,label=species))+geom_text(size=2.5)+geom_abline()+cleanup+ stat_cor(method = "pearson")+labs(color="Known\nResistant\nMutant")+scale_x_continuous("Enrichment score screen 1")+scale_y_continuous("Enrichment score screen 2")
# ggsave(paste(pathname,"/plot2c_Enrichmentscores_correlations_text.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)


# Plot 2d: What do the correlations look like for net growth rates (show mutants in text + add depleted flag)?
ggplot(screen_compare_means%>%filter(is_intended%in%1,protein_start%in%c(start_site:end_site)),aes(x=netgr_obs_screen1,y=netgr_obs_screen2,label=species))+geom_point(color="black",shape=21,size=2,aes(fill=depleted),alpha=.7)+geom_abline()+cleanup+stat_cor(method = "pearson")+labs(fill="Depleting\nMutant")+scale_x_continuous("Net growth rate screen 1")+scale_y_continuous("Net growth rate screen 2")
# ggsave(paste(pathname,"/plot2d_Netgrowthrate_correlations_depletedmutants_points.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)

# Plot 3a: Plots: what are the overall net growth rate distributions?
ggplot(screen_compare_means%>%filter(is_intended%in%1),aes(x=netgr_obs_mean,fill=resmuts))+geom_density(alpha=0.7)+cleanup+labs(fill="Known\nResistant\nMutant")+scale_x_continuous("Mean net growth rate of screens")
# ggsave(paste(pathname,"/plot3a_Netgrowthrate_distributions_resmuts.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)
# Plot 3b: Plots: what are the net growth rate distributions?

library(reshape2)
screen_compare_melt=melt(screen_compare_means%>%dplyr::select(species,netgr_obs_screen1,netgr_obs_screen2),id.vars = "species",measure.vars =c("netgr_obs_screen1","netgr_obs_screen2"),variable.name = "Condition",value.name = "netgr_obs")


ggplot(screen_compare_melt,aes(x=netgr_obs,fill=Condition))+
  geom_density(alpha=0.7)+
  cleanup+
  scale_x_continuous("Net growth rate observed")+
  scale_fill_discrete(labels=c("Screen 1","Screen 2"))
# ggsave(paste(pathname,"/plot3b_Netgrowthrate_distributions.pdf",sep=""),width=6,height=4,units="in",useDingbats=F)

}

# dirname="K562_Medium_rep1vs2"

```

### Data plotting
```{r}
data=read.csv("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/baf3_Imat_high_rep1vsrep2_duplex/screen_comparison_baf3_Imat_high_rep1vsrep2_duplex.csv",header = T,stringsAsFactors = F)[-1]
data$seqtech="duplex"
data_duplex=data


data=read.csv("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/baf3_Imat_high_rep1vsrep2_ngs/screen_comparison_baf3_Imat_high_rep1vsrep2_ngs.csv",header = T,stringsAsFactors = F)[-1]
data$seqtech="ngs"
data_ngs=data

data=read.csv("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/baf3_Imat_high_rep1vsrep2_sscs/screen_comparison_baf3_Imat_high_rep1vsrep2_sscs.csv",header = T,stringsAsFactors = F)[-1]
data$seqtech="sscs"
data_sscs=data

data=read.csv("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/baf3_Imat_high_rep1vsrep2_tileseq/screen_comparison_baf3_Imat_high_rep1vsrep2_tileseq.csv",header = T,stringsAsFactors = F)[-1]
data$seqtech="tileseq"
data_tileseq=data

data=rbind(data_duplex,data_ngs,data_sscs,data_tileseq)

data_sum=data%>%
  filter(protein_start%in%c(242:322),is_intended%in%1)%>%
  group_by(seqtech)%>%
  summarize(mean_depth=mean(depth_screen1_before),
            num=n())

data=data%>%rowwise%>%mutate(n_nuc=nchar(alt))

```


Plotting unintended vs intended mutant distributions across sequencing technologies
```{r}
ggplot(data%>%filter(ct_screen1_before>=5,seqtech%in%"ngs",n_nuc%in%1),aes(x=ct_screen1_before,fill=factor(is_intended,levels=c(1,0))))+geom_density(alpha=0.7)+scale_x_continuous("Number of times a variant was seen",trans="log10")+cleanup+scale_fill_manual(values = c("blue","red"))+scale_y_continuous("Density")+theme(legend.position = "none")
ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot6a_abl_ngs_untreated.pdf",width=3.5,height=2.5,units="in",useDingbats=F)

ggplot(data%>%filter(ct_screen1_before>=5,seqtech%in%"tileseq",n_nuc%in%1),aes(x=ct_screen1_before,fill=factor(is_intended,levels=c(1,0))))+geom_density(alpha=0.7)+scale_x_continuous("Number of times a variant was seen",trans="log10")+cleanup+scale_fill_manual(values = c("blue","red"))+scale_y_continuous("Density")+theme(legend.position = "none")
ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot6b_abl_tileseq_untreated.pdf",width=3.5,height=2.5,units="in",useDingbats=F)

ggplot(data%>%filter(seqtech%in%"duplex",n_nuc%in%1),aes(x=ct_screen1_before,fill=factor(is_intended,levels=c(1,0))))+geom_density(alpha=0.7)+scale_x_continuous("Number of times a variant was seen",trans="log10")+cleanup+scale_fill_manual(values = c("blue","red"))+scale_y_continuous("Density")+theme(legend.position = "none")
ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot6c_abl_duplex_untreated.pdf",width=3.5,height=2.5,units="in",useDingbats=F)


#######################NNMD NGS#######################
nnmd_ngs_intended=data%>%filter(ct_screen1_before>=5,seqtech%in%"ngs",is_intended%in%1)
nnmd_ngs_intended=nnmd_ngs_intended[,"ct_screen1_before"]
nnmd_ngs_unintended=data%>%filter(ct_screen1_before>=5,seqtech%in%"ngs",is_intended%in%0)
nnmd_ngs_unintended=nnmd_ngs_unintended[,"ct_screen1_before"]

(mean(nnmd_ngs_unintended$ct_screen1_before)-mean(nnmd_ngs_intended$ct_screen1_before))/sd(nnmd_ngs_unintended$ct_screen1_before)
#######################NNMD Tileseq#######################

nnmd_ngs_intended=data%>%filter(ct_screen1_before>=5,seqtech%in%"tileseq",is_intended%in%1)
nnmd_ngs_intended=nnmd_ngs_intended[,"ct_screen1_before"]
nnmd_ngs_unintended=data%>%filter(ct_screen1_before>=5,seqtech%in%"tileseq",is_intended%in%0)
nnmd_ngs_unintended=nnmd_ngs_unintended[,"ct_screen1_before"]
(mean(nnmd_ngs_unintended$ct_screen1_before)-mean(nnmd_ngs_intended$ct_screen1_before))/sd(nnmd_ngs_unintended$ct_screen1_before)
#######################NNMD Duplex#######################

nnmd_ngs_intended=data%>%filter(seqtech%in%"duplex",is_intended%in%1)
nnmd_ngs_intended=nnmd_ngs_intended[,"ct_screen1_before"]
nnmd_ngs_unintended=data%>%filter(seqtech%in%"duplex",is_intended%in%0)
nnmd_ngs_unintended=nnmd_ngs_unintended[,"ct_screen1_before"]
(mean(nnmd_ngs_unintended$ct_screen1_before)-mean(nnmd_ngs_intended$ct_screen1_before))/sd(nnmd_ngs_unintended$ct_screen1_before)
##############################################
df <- data.frame(
  seqtech = c("NGS", "Tileseq", "Duplex"),
  nnmd = c(-0.6647498, -4.328837, -0.7182519)
)

ggplot(df,aes(x=seqtech,y=nnmd,fill=seqtech))+
  geom_col()+
  scale_fill_manual(values = c("#FB8072","#3287BD","#8ED3C7"))+
  # scale_y_continuous("Null-normalized mean difference")+
  cleanup+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.title.y = element_blank(),
        legend.position = "none")

ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot7.pdf",width=2,height=2,useDingbats=F)

###################Plotting ROC curves for being able to tell intended mutants (signal) from unintended mutants (noise)###############
library(pROC)
data_roc=data%>%filter(seqtech%in%"ngs",ct_screen1_before>5)
ctdata_ngs=data_roc%>%dplyr::select(ct_screen1_before,is_intended)
data_roc=data%>%filter(seqtech%in%"tileseq",ct_screen1_before>5)
ctdata_tileseq=data_roc%>%dplyr::select(ct_screen1_before,is_intended)
data_roc=data%>%filter(seqtech%in%"duplex")
ctdata_duplex=data_roc%>%dplyr::select(ct_screen1_before,is_intended)

ctdata_ngs$is_intended=as.numeric(ctdata_ngs$is_intended)
ctdata_tileseq$is_intended=as.numeric(ctdata_tileseq$is_intended)
ctdata_duplex$is_intended=as.numeric(ctdata_duplex$is_intended)

ggplot(ctdata_ngs,aes(x=ct_screen1_before,y=is_intended))+geom_point()
ggplot(ctdata_tileseq,aes(x=ct_screen1_before,y=is_intended))+geom_point()
ggplot(ctdata_duplex,aes(x=ct_screen1_before,y=is_intended))+geom_point()

glm.fit=glm(as.numeric(ctdata_ngs$is_intended)~ctdata_ngs$ct_screen1_before,family=binomial)
ctdata_ngs$glm_fits=glm.fit$fitted.values

glm.fit=glm(as.numeric(ctdata_tileseq$is_intended)~ctdata_tileseq$ct_screen1_before,family=binomial)
ctdata_tileseq$glm_fits=glm.fit$fitted.values

glm.fit=glm(as.numeric(ctdata_duplex$is_intended)~ctdata_duplex$ct_screen1_before,family=binomial)
ctdata_duplex$glm_fits=glm.fit$fitted.values

ggplot(ctdata_ngs,aes(x=ct_screen1_before,y=as.numeric(is_intended)))+geom_point()+geom_line(aes(x=ct_screen1_before,y=glm_fits))+scale_x_continuous(trans="log10")

ggplot(ctdata_tileseq,aes(x=ct_screen1_before,y=as.numeric(is_intended)))+geom_point()+geom_line(aes(x=ct_screen1_before,y=glm_fits))+scale_x_continuous(trans="log10")

ggplot(ctdata_duplex,aes(x=ct_screen1_before,y=as.numeric(is_intended)))+geom_point()+geom_line(aes(x=ct_screen1_before,y=glm_fits))+scale_x_continuous(trans="log10")

roc(as.numeric(ctdata_ngs$is_intended),ctdata_ngs$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#3287BD",lwd=4,print.auc.y=45)
roc(as.numeric(ctdata_tileseq$is_intended),ctdata_tileseq$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#8ED3C7",add=T,lwd=4,print.auc.y=55)
roc(as.numeric(ctdata_duplex$is_intended),ctdata_duplex$glm_fits,plot=T,legacy.axes=T,percent=T,xlab="False Positive Percentage",ylab="True Positive Percentage",print.auc=T,col="#FB8072",add=T,lwd=4,print.auc.y=65)

roc_1=roc(as.numeric(ctdata_ngs$is_intended),ctdata_ngs$glm_fits,plot=T)

roc_2=roc(as.numeric(ctdata_tileseq$is_intended),ctdata_tileseq$glm_fits)

roc_3=roc(as.numeric(ctdata_duplex$is_intended),ctdata_duplex$glm_fits)

pdf("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot8_roc_curve.pdf", width = 3, height = 3)
par(pty="s")
plot(roc_1,legacy.axes=T,percent=T,xlab="",ylab="",print.auc=T,col="#3287BD",lwd=3,print.auc.y=45)

plot(roc_2,plot=T,legacy.axes=T,percent=T,xlab="",ylab="",print.auc=T,col="#8ED3C7",add=T,lwd=3,print.auc.y=55)

plot(roc_3,plot=T,legacy.axes=T,percent=T,xlab="",ylab="",print.auc=T,col="#FB8072",add=T,lwd=3,print.auc.y=65)

dev.off()
```



Plots to make:
CDF
Tileseq depths are high vs dupseq
Tileseq error rates are high vs dupseq

```{r}
library(tidyr)
data_snr=data%>%filter(protein_start%in%c(242:322))%>%group_by(seqtech,is_intended,protein_start,n_nuc_min)%>%summarize(nmuts=n(),coverage_mean=mean(ct_screen1_before),coverage_sum=sum(ct_screen1_before),depth_mean=mean(depth_screen1_before))

data_snr_cast <- data_snr %>%
  pivot_wider(
    names_from = is_intended,
    values_from = c(nmuts,coverage_mean,coverage_sum,depth_mean),
    names_glue = "{.value}_{ifelse(is_intended == 1, 'intended', 'unintended')}"
  )
# data_snr_cast=data_snr_cast%>%mutate(snr=coverage_mean_intended/coverage_mean_unintended)
data_snr_cast=data_snr_cast%>%mutate(snr=coverage_sum_intended/coverage_sum_unintended)

ggplot(data_snr_cast,aes(x=protein_start,y=snr))+geom_point()+facet_wrap(~seqtech)+scale_y_continuous(limits=c(0,60))

ggplot(data_snr_cast,aes(x=snr,fill=seqtech))+geom_density(alpha=0.6)+scale_x_continuous("Signal to noise",trans="log10")

ggplot(data_snr_cast,aes(x=protein_start,y=nmuts_intended))+geom_point()+facet_wrap(~seqtech)

ggplot(data_snr_cast,aes(x=protein_start,y=nmuts_unintended))+geom_point()+facet_wrap(~seqtech)

ggplot(data_snr_cast,aes(x=protein_start,y=coverage_mean_unintended))+geom_point()+facet_wrap(~seqtech)+scale_y_continuous(limits=c(0,200))

ggplot(data_snr_cast,aes(x=protein_start,y=coverage_sum_unintended))+geom_point()+facet_wrap(~seqtech)+scale_y_continuous(trans = "log10")

ggplot(data_snr_cast,aes(x=coverage_mean_unintended/depth_mean_unintended,fill=seqtech))+geom_density(alpha=.6)+scale_x_continuous("Allele fraction of unintended mutants",trans="log10")

ggplot(data_snr_cast,aes(x=coverage_mean_intended/depth_mean_intended,fill=seqtech))+geom_density(alpha=.6)+scale_x_continuous("Allele fraction of intended mutants",trans="log10")
# data_snr_cast=resmuts_adder(data_snr_cast)
# a=data%>%filter(seqtech%in%"sscs",dose%in%"300nM",is_intended%in%1,protein_start%in%c(245))


ggplot(data%>%filter(resmuts%in%T,is_intended%in%1),aes(x=species,y=ct_screen1_before,fill=seqtech))+geom_point(color="black",shape=21)+geom_line()+scale_y_continuous("Day 0 Coverage of Resistant Mutant",trans="log10")+cleanup+theme(axis.text.x=element_text(angle=90,hjust=.5,vjust=.5))

ggplot(data%>%filter(resmuts%in%T,is_intended%in%1),aes(x=species,y=ct_screen1_after,fill=seqtech))+geom_point(color="black",shape=21)+geom_line()+scale_y_continuous("Day 6 Coverage of Resistant Mutant",trans="log10")+cleanup+theme(axis.text.x=element_text(angle=90,hjust=.5,vjust=.5))

ggplot(data%>%filter(is_intended%in%1),aes(x=ct_screen1_before,color=seqtech))+stat_ecdf()+scale_x_continuous("Intended mutant coverage",trans="log10")+scale_y_continuous("CDF")

ggplot(data%>%filter(is_intended%in%1),aes(x=ct_screen1_after,color=seqtech))+stat_ecdf()+scale_x_continuous("Intended mutant coverage",trans="log10")+scale_y_continuous("CDF")


ggplot(data,aes(x=ct_screen1_before,fill=seqtech))+geom_histogram()+facet_wrap(seqtech~is_intended)+scale_x_continuous(trans="log10")

ggplot(data%>%filter(is_intended%in%1,seqtech=="sscs"),aes(x=ct_screen1_before))+geom_histogram()+scale_x_continuous(trans="log10")

ggplot(data%>%filter(is_intended%in%1,seqtech=="sscs"),aes(x=ct_screen1_after))+geom_histogram()+scale_x_continuous(trans="log10")

ggplot(data%>%filter(is_intended%in%1),aes(x=ct_screen1_after,fill=seqtech))+geom_histogram()+facet_wrap(~seqtech)+scale_x_continuous(trans="log10")

```
Making final plots for figures
Plots to make:
CDF
Tileseq depths are high vs dupseq
Tileseq error rates are high vs dupseq

```{r}
# Demonstrating noise:
ggplot(data_snr_cast%>%filter(seqtech%in%c("duplex","tileseq")),aes(x=protein_start,y=coverage_sum_unintended))+
  geom_point(color="black",size=2,shape=21,aes(fill=seqtech))+
  # facet_wrap(~seqtech)+
  scale_fill_manual(values = c("#FB8072","#8ED3C7"))+
  scale_y_continuous("Coverage of unintended mutants",trans = "log10")+
  scale_x_continuous("Rediue on BCRABL")+
  cleanup+
  theme(legend.position = "none",
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color="black",size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))
# ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot1.pdf",width=2.5,height=2.5,units = "in",useDingbats=F)

ggplot(data_snr_cast%>%filter(seqtech%in%c("duplex","tileseq")),aes(x=protein_start,y=nmuts_unintended,fill=seqtech))+
  geom_point(color="black",size=2,shape=21,aes(fill=seqtech))+
  scale_fill_manual(values = c("#FB8072","#8ED3C7"))+
  geom_hline(yintercept = 40,linetype="dashed")+
  cleanup+
  theme(legend.position = "none",
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color="black",size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))
# ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot2.pdf",width=2.5,height=2.5,units = "in",useDingbats=F)

# Demonstrating coverage:
ggplot(data%>%filter(resmuts%in%T,is_intended%in%1,seqtech%in%c("duplex","tileseq")),aes(x=species,y=ct_screen1_before,fill=seqtech))+
  geom_point(color="black",shape=21)+
  geom_line()+
  scale_y_continuous("Day 0 Coverage of Resistant Mutant",trans="log10",limits=c(1,1.2e5))+
  cleanup+
  theme(axis.text.x=element_text(color="black",size=8,angle=90,hjust=.5,vjust=.5),
        legend.position = "none",
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))
# ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot3.pdf",width=2.5,height=2.5,units = "in",useDingbats=F)

ggplot(data%>%filter(resmuts%in%T,is_intended%in%1,seqtech%in%c("duplex","tileseq")),aes(x=species,y=ct_screen2_after,fill=seqtech))+
  geom_point(color="black",shape=21)+
  geom_line()+
  scale_y_continuous("Day 6 Coverage of Resistant Mutant",trans="log10",limits=c(1,1.2e5))+
  cleanup+
  theme(axis.text.x=element_text(color="black",size=8,angle=90,hjust=.5,vjust=.5),
        legend.position = "none",
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))
# ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot3b.pdf",width=2.5,height=2.5,units = "in",useDingbats=F)

ggplot(data%>%filter(is_intended%in%1,seqtech%in%c("duplex","tileseq")),aes(x=ct_screen1_before,color=seqtech))+
  stat_ecdf()+
  scale_x_continuous("Intended mutant coverage",trans="log10")+
  scale_y_continuous("CDF")+
  scale_color_manual(values=c("#FB8072","#8ED3C7"))+
  cleanup+
  theme(axis.text.x=element_text(color="black",size=8,angle=90,hjust=.5,vjust=.5),
        legend.position = "none",
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))
# ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot4.pdf",width=2.5,height=2.5,units = "in",useDingbats=F)


####Making a plot of CV as a function of counts
# Summarize the data table by ct_screen1_before, and calculate the CV between the screen 1 and screen 2 count (or net growth rate)
data_sum=data%>%filter(seqtech=="tileseq",is_intended==1)
data_sum=data_sum%>%
  mutate(ct_screen1_after=case_when(ct_screen1_after%in%NA~0,
                                    ct_screen1_after%in%0.5~0,
                                    T~ct_screen1_after),
         ct_screen2_after=case_when(ct_screen2_after%in%NA~0,
                                    ct_screen2_after%in%0.5~0,
                                    T~ct_screen2_after))

data_sum=data_sum%>%
  rowwise()%>%
  mutate(sd_ct=sd(c(ct_screen1_after,ct_screen2_after)),
         cv_ct=sd(c(ct_screen1_after,ct_screen2_after))/mean(c(ct_screen1_after,ct_screen2_after)),
         sd_netgr=sd(c(netgr_obs_screen1,netgr_obs_screen2)),
         cv_netgr=sd(c(netgr_obs_screen1,netgr_obs_screen2))/mean(c(netgr_obs_screen1,netgr_obs_screen2)))

data_sum$ct_screen1_after_group <- cut(data_sum$ct_screen1_after, breaks = seq(0, max(data_sum$ct_screen1_after) + 15, by = 15), right = FALSE)


ggplot(data_sum%>%filter(ct_screen1_after<=1000),aes(x=factor(ct_screen1_after_group),y=cv_netgr))+geom_boxplot()
ggplot(data_sum%>%filter(ct_screen1_after<=1000),aes(x=factor(ct_screen1_after_group),y=cv_ct))+geom_boxplot()
# sort(unique(data$seqtech))

```

```{r}
ggplot(data_snr_cast%>%filter(seqtech%in%c("duplex","tileseq","ngs")),aes(x=protein_start,y=coverage_sum_unintended))+
  geom_point(color="black",size=2,shape=21,aes(fill=seqtech))+
  # facet_wrap(~seqtech)+
  scale_fill_manual(values = c("#FB8072","#3287BD","#8ED3C7"))+
  scale_y_continuous("Coverage of unintended mutants",trans = "log10")+
  scale_x_continuous("Rediue on BCRABL")+
  cleanup+
  theme(legend.position = "none",
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color="black",size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))


ggplot(data_snr_cast%>%filter(n_nuc_min%in%1,seqtech%in%c("duplex","tileseq","ngs")),aes(x=protein_start,y=nmuts_unintended,fill=seqtech))+
  geom_point(color="black",size=2,shape=21,aes(fill=seqtech))+
  geom_line(aes(color=seqtech))+
  scale_fill_manual(values = c("#FB8072","#3287BD","#8ED3C7"))+
  scale_color_manual(values = c("#FB8072","#3287BD","#8ED3C7"))+
  # geom_hline(yintercept = 40,linetype="dashed")+
  cleanup+
  theme(legend.position = "none",
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color="black",size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))

```


#Standards with tileseq only
```{r}
library(binom)

standards=read.csv("output/ic50data_all_conc.csv",header = T,stringsAsFactors = F)
standards=standards%>%filter(conc%in%1.2)

# data=data%>%filter(is_intended%in%1)
standards=merge(data%>%filter(seqtech%in%"tileseq",is_intended%in%1),standards,by.x="species",by.y="mutant")

standards=standards%>%
  rowwise()%>%
  mutate(
    mean_fitness = mean(c(netgr_obs_screen1, netgr_obs_screen2)),
    sd_fitness = sd(c(netgr_obs_screen1, netgr_obs_screen2)),
    se_fitness = sd_fitness / sqrt(2),  # SE with two replicates
    lower_ci = mean_fitness - 1.96 * se_fitness,  # 95% CI lower bound
    upper_ci = mean_fitness + 1.96 * se_fitness   # 95% CI upper bound
  )

ggplot(standards%>%filter(!protein_start%in%c(354:361)),aes(x=.055-drug_effect,y=mean_fitness))+
  stat_cor(method="pearson",size=2,vjust=3)+
  geom_point(size=2,shape=21,fill="#8ED3C7")+
  geom_errorbar(aes(ymin=lower_ci,ymax=upper_ci,width=0.001))+
  # geom_point(size=2,color="red",aes(y=netgr_obs_screen2))+
  # stat_cor(method="pearson",size=2,aes(y=netgr_obs_screen1),color="red")+
  # scale_x_continuous("Predicted Net Growth Rate (Hrs-1) \nClonal IC50",limits=c(-0.01,.06))+
  # scale_y_continuous("Observed Net Growth Rate (Hrs-1) \nDMS Screen",limits=c(-0.01,.06))+
  scale_x_continuous("Predicted Net Growth Rate (Hrs-1) \nClonal IC50",limits=c(-0.026,.075))+
  scale_y_continuous("Observed Net Growth Rate (Hrs-1) \nDMS Screen",limits=c(-0.026,.075))+
  geom_abline(linetype="dashed")+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color="black",size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))
```

### Standards with duplex
```{r}
library(binom)

standards=read.csv("output/ic50data_all_conc.csv",header = T,stringsAsFactors = F)
standards=standards%>%filter(conc%in%1.2)

# data=data%>%filter(is_intended%in%1)
################### Standards with NGS data###############
standards_ngs=merge(data%>%filter(seqtech%in%"ngs",is_intended%in%1),standards,by.x="species",by.y="mutant")

standards_ngs=standards_ngs%>%
  rowwise()%>%
  mutate(
    mean_fitness = mean(c(netgr_obs_screen1, netgr_obs_screen2)),
    sd_fitness = sd(c(netgr_obs_screen1, netgr_obs_screen2)),
    se_fitness = sd_fitness / sqrt(2),  # SE with two replicates
    lower_ci = mean_fitness - 1.96 * se_fitness,  # 95% CI lower bound
    upper_ci = mean_fitness + 1.96 * se_fitness   # 95% CI upper bound
  )
################### Standards with tileseq data###############
standards_tileseq=merge(data%>%filter(seqtech%in%"tileseq",is_intended%in%1),standards,by.x="species",by.y="mutant")

standards_tileseq=standards_tileseq%>%
  rowwise()%>%
  mutate(
    mean_fitness = mean(c(netgr_obs_screen1, netgr_obs_screen2)),
    sd_fitness = sd(c(netgr_obs_screen1, netgr_obs_screen2)),
    se_fitness = sd_fitness / sqrt(2),  # SE with two replicates
    lower_ci = mean_fitness - 1.96 * se_fitness,  # 95% CI lower bound
    upper_ci = mean_fitness + 1.96 * se_fitness   # 95% CI upper bound
  )
################### Standards with dupseq data###############
standards_duplex=merge(data%>%filter(seqtech%in%"duplex",is_intended%in%1),standards,by.x="species",by.y="mutant")

standards_duplex=standards_duplex%>%
  rowwise()%>%
  mutate(
    mean_fitness = mean(c(netgr_obs_screen1, netgr_obs_screen2)),
    sd_fitness = sd(c(netgr_obs_screen1, netgr_obs_screen2)),
    se_fitness = sd_fitness / sqrt(2),  # SE with two replicates
    lower_ci = mean_fitness - 1.96 * se_fitness,  # 95% CI lower bound
    upper_ci = mean_fitness + 1.96 * se_fitness   # 95% CI upper bound
  )
standards_duplex$netgr_pred=0.055-standards_duplex$drug_effect

#########################################################


ggplot()+
  geom_point(size=2,shape=21,fill="#3287BD",aes(x=.055-standards_ngs$drug_effect,y=standards_ngs$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_ngs$drug_effect,ymin=standards_ngs$lower_ci,ymax=standards_ngs$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_ngs$drug_effect,y=standards_ngs$mean_fitness),method="pearson",color="#3287BD",size=2,vjust=3)+
  geom_point(size=2,shape=21,fill="#8ED3C7",aes(x=.055-standards_tileseq$drug_effect,y=standards_tileseq$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_tileseq$drug_effect,ymin=standards_tileseq$lower_ci,ymax=standards_tileseq$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_tileseq$drug_effect,y=standards_tileseq$mean_fitness),method="pearson",color="#8ED3C7",size=2,vjust=1.5)+
  geom_point(size=2,shape=21,fill="#FB8072",aes(x=.055-standards_duplex$drug_effect,y=standards_duplex$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_duplex$drug_effect,ymin=standards_duplex$lower_ci,ymax=standards_duplex$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_duplex$drug_effect,y=standards_duplex$mean_fitness),method="pearson",color="#FB8072",size=2,vjust=0)+
  
  
  scale_x_continuous("Predicted Net Growth Rate (Hrs-1) \nClonal IC50",limits=c(-0.016,.06))+
  scale_y_continuous("Observed Net Growth Rate (Hrs-1) \nDMS Screen",limits=c(-0.016,.06))+
  geom_abline(linetype="dashed")+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color="black",size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))

# ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot5.pdf",width=2.5,height=2.5,units = "in",useDingbats=F)

ggplot()+
  geom_point(size=2,shape=21,fill="#3287BD",aes(x=.055-standards_ngs$drug_effect,y=standards_ngs$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_ngs$drug_effect,ymin=standards_ngs$lower_ci,ymax=standards_ngs$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_ngs$drug_effect,y=standards_ngs$mean_fitness),method="pearson",color="#3287BD",size=2,vjust=3)+
scale_x_continuous("Predicted Net Growth Rate (Hrs-1) \nClonal IC50",limits=c(-0.016,.06))+
  scale_y_continuous("Observed Net Growth Rate (Hrs-1) \nDMS Screen",limits=c(-0.016,.06))+
  geom_abline(linetype="dashed")+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color="black",size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))
ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot5_ngs.pdf",width=2.5,height=2.5,units = "in",useDingbats=F)

ggplot()+
geom_point(size=2,shape=21,fill="#8ED3C7",aes(x=.055-standards_tileseq$drug_effect,y=standards_tileseq$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_tileseq$drug_effect,ymin=standards_tileseq$lower_ci,ymax=standards_tileseq$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_tileseq$drug_effect,y=standards_tileseq$mean_fitness),method="pearson",color="#8ED3C7",size=2,vjust=1.5)+
  scale_x_continuous("Predicted Net Growth Rate (Hrs-1) \nClonal IC50",limits=c(-0.016,.06))+
  scale_y_continuous("Observed Net Growth Rate (Hrs-1) \nDMS Screen",limits=c(-0.016,.06))+
  geom_abline(linetype="dashed")+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color="black",size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))
ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot5_tileseq.pdf",width=2.5,height=2.5,units = "in",useDingbats=F)

ggplot()+
geom_point(size=2,shape=21,fill="#FB8072",aes(x=.055-standards_duplex$drug_effect,y=standards_duplex$mean_fitness))+
  geom_errorbar(aes(x=.055-standards_duplex$drug_effect,ymin=standards_duplex$lower_ci,ymax=standards_duplex$upper_ci,width=0.001))+
  stat_cor(aes(x=.055-standards_duplex$drug_effect,y=standards_duplex$mean_fitness),method="pearson",color="#FB8072",size=2,vjust=0)+
scale_x_continuous("Predicted Net Growth Rate (Hrs-1) \nClonal IC50",limits=c(-0.016,.06))+
  scale_y_continuous("Observed Net Growth Rate (Hrs-1) \nDMS Screen",limits=c(-0.016,.06))+
  geom_abline(linetype="dashed")+
  cleanup+
  theme(legend.position = c(0.92,0.45),
        legend.title = element_text(size=8),
        legend.text = element_text(size=8),
        axis.text.x=element_text(color="black",size=8),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_text(color="black",size=8),
        plot.margin=grid::unit(c(1,1,1,1), "mm"))
ggsave("output/ABLEnrichmentScreens/ABL_Region1234_Tileseq/lane18b_rerun_comparisons/dcsvstileseqfigures/plot5_duplex.pdf",width=2.5,height=2.5,units = "in",useDingbats=F)

```
### Calculating the binomial confidence intervals of the 
```{r}
# binom.test(x=6,n=20000,conf.level = 0.95)
# standards_duplex2=standards_duplex%>%mutate(ct_screen1_before_ll=round(binom.test(ct_screen1_before,depth_screen1_before,conf.level = 0.95)$conf.int[1]*depth_screen1_before,0)+0.5,ct_screen1_before_ul=round(binom.test(ct_screen1_before,depth_screen1_before,conf.level = 0.95)$conf.int[2]*depth_screen1_before,0)+0.5,
#                                             fc_ul=log(ct_screen1_before_ul)/log(ct_screen1_before),
#                                             fc_ll=log(ct_screen1_before_ll)/log(ct_screen1_before))
# binom.test(6,20000,conf.level = 0.95)$conf.int[2]

data_duplex_sum=data_duplex%>%group_by(ct_screen1_before)%>%summarize(sum=sum(),n=n())
data_duplex_sum=data_duplex_sum%>%
  rowwise()%>%
  mutate(ct_screen1_before_ll=round(binom.test(ct_screen1_before,6900,conf.level = 0.95)$conf.int[1]*6900,0),
         ct_screen1_before_ul=round(binom.test(ct_screen1_before,6900,conf.level = 0.95)$conf.int[2]*6900,0))

ggplot(data_duplex_sum,aes(x=ct_screen1_before,y=n))+
  geom_ribbon(aes(xmin=ct_screen1_before_ll,xmax=ct_screen1_before_ul),alpha=.8,fill="#FB8072")+
  geom_line()+
  # geom_point(color="black",shape=21,fill="#FB8072")+
  cleanup+
  scale_x_continuous(trans="log10")




data_tileseq_sum=data_tileseq%>%group_by(ct_screen1_before)%>%summarize(sum=sum(),n=n())
data_tileseq_sum$ct_screen1_before=as.integer(data_tileseq_sum$ct_screen1_before)
data_tileseq_sum=data_tileseq_sum%>%
  filter(ct_screen1_before<=200)%>%
  rowwise()%>%
  mutate(ct_screen1_before_ll=round(binom.test(ct_screen1_before,4461942,conf.level = 0.95)$conf.int[1]*4461942,0),
         ct_screen1_before_ul=round(binom.test(ct_screen1_before,4461942,conf.level = 0.95)$conf.int[2]*4461942,0))

ggplot(data_tileseq_sum,aes(x=ct_screen1_before,y=n))+
  geom_ribbon(aes(xmin=ct_screen1_before_ll,xmax=ct_screen1_before_ul),alpha=.8,fill="#FB8072")+
  geom_line()+
  # geom_point(color="black",shape=21,fill="#FB8072")+
  cleanup+
  scale_x_continuous(trans="log10")+
  scale_y_continuous(trans="log10")
```

