---
title: "twinstrand_spikeins_data_generation"
author: "Haider Inam"
date: "4/2/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```

####Please change required directories this chunk if compiling in R rather than RmD
```{r}
#Inputs:
conc_for_predictions=0.8
net_gr_wodrug=0.05
#Reading required tables
ic50data=read.csv("data/heatmap_concat_data.csv",header = T,stringsAsFactors = F)
# ic50data=read.csv("../data/heatmap_concat_data.csv",header = T,stringsAsFactors = F)

twinstrand_maf=read.table("data/Twinstrand/prj00053-2019-12-02.deliverables/all.mut",sep="\t",header = T,stringsAsFactors = F)
# twinstrand_maf=read.table("../data/Twinstrand/prj00053-2019-12-02.deliverables/all.mut",sep="\t",header = T,stringsAsFactors = F)

names=read.table("data/Twinstrand/prj00053-2019-12-02.deliverables/manifest.tsv",sep="\t",header = T,stringsAsFactors = F)
# names=read.table("../data/Twinstrand/prj00053-2019-12-02.deliverables/manifest.tsv",sep="\t",header = T,stringsAsFactors = F)

```

```{r,include=F}
# rm(list=ls())
library(knitr)
library(tictoc)
library(workflowr)
library(VennDiagram)
library(dplyr)
library(foreach)
library(doParallel)
library(ggplot2)
library(reshape2)
library(RColorBrewer)
library(devtools)
library(ggsignif)
library(plotly)
library(BiocManager)
library(drc)
# library(ensembldb) #Loading this with Dplyr commands seems to throw an error in Rmd
# library(EnsDb.Hsapiens.v86) #Loading this with Dplyr commands seems to throw an error in Rmd

######################Cleanup for GGPlot2#########################################
cleanup=theme_bw() +
  theme(plot.title = element_text(hjust=.5),
        panel.grid.major = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(color = "black"),
        axis.text = element_text(face="bold",color="black",size="11"),
        text=element_text(size=11,face="bold"),
        axis.title=element_text(face="bold",size="11"))
```


##Data Parsing--  Dose Response Data  
###Importing model with BCR-ABL mutant dose responses (Chuan's data)
###Also applying the 2-parameter logistic. Upper and lower limits are fixed. May be good or bad.
```{r}
#Deciding not to use nls() because it's a pain in the ...
#https://www.youtube.com/watch?v=aXpJE7IGiPY this has a nice overview of curve fitting
# library(dplyr)
# rm(list=ls())
####Getting effect of drug on growth rate####

ic50data=ic50data[c(1:10),]
ic50data_long=melt(ic50data,id.vars = "conc",variable.name = "species",value.name = "y")
#Removing useless mutants (for example keeping only maxipreps and removing low growth rate mutants)
ic50data_long=ic50data_long%>%filter(species%in%c("Wt","V299L_H","E355A","D276G_maxi","H396R","F317L","F359I","E459K","G250E","F359C","F359V","M351T","L248V","E355G_maxi","Q252H_maxi","Y253F","F486S_maxi","H396P_maxi","E255K","Y253H","T315I","E255V"))



#Making standardized names
ic50data_long$mutant=ic50data_long$species
ic50data_long=ic50data_long%>%
  # filter(conc=="0.625")%>%
  # filter(conc=="1.25")%>%
  mutate(mutant=case_when(species=="F486S_maxi"~"F486S",
                          species=="H396P_maxi"~"H396P",
                          species=="Q252H_maxi"~"Q252H",
                          species=="E355G_maxi"~"E355G",
                          species=="D276G_maxi"~"D276G",
                          species=="V299L_H" ~ "V299L",
                          species==mutant ~as.character(mutant)))

# ic50data_long_625$species[order((ic50data_long_625$y),decreasing = T)]

#In the next step, I'm ordering mutants by decreasing resposne to the 625nM dose. Then I use this to change the levels of the species factor from more to less resistant. This helps with ggplot because now I can color the mutants with decreasing resistance
ic50data_long_625=ic50data_long%>%filter(conc==.625)
ic50data_long$species=factor(ic50data_long$species,levels = as.character(ic50data_long_625$species[order((ic50data_long_625$y),decreasing = T)]))

#Plotting the normalized dose response curves
getPalette = colorRampPalette(brewer.pal(9, "Spectral"))

plotly=ggplot(ic50data_long,aes(x=log(conc),y=y,color=factor(species)))+
  facet_wrap(~factor(species))+
  geom_line()+
  geom_point()+
  cleanup+
  scale_color_manual(values = getPalette(length(unique(ic50data_long$species))))+
  theme(axis.text = element_blank(),
        axis.ticks = element_blank())
ggplotly(plotly)
```
###Dose response curve fitting with 4-parameter logistic
####First iteration: Have a y_model for only the drug concentrations Chuan used
Essentially, all this is doing is adding a column for y-model to IC50data_long. Default was just y (proportion alive).
```{r,eval=F}
########Four parameter logistic########
#Reference: https://journals.plos.org/plosone/article/file?type=supplementary&id=info:doi/10.1371/journal.pone.0146021.s001
#In short: For each dose in each species, get the response
# rm(list=ls())
ic50data_long_model=data.frame()
for (species_curr in sort(unique(ic50data_long$species))){
  ic50data_species_specific=ic50data_long%>%filter(species==species_curr)
  x=ic50data_species_specific$conc
  y=ic50data_species_specific$y
  #Next: Appproximating Response from dose (inverse of the prediction)
  ic50.ll4=drm(y~conc,data=ic50data_long%>%filter(species==species_curr),fct=LL.3(fixed=c(NA,1,NA)))
    b=coef(ic50.ll4)[1]
    c=0
    d=1
    e=coef(ic50.ll4)[2]
  ###Getting predictions
  ic50data_species_specific=ic50data_species_specific%>%group_by(conc)%>%mutate(y_model=c+((d-c)/(1+exp(b*(log(conc)-log(e))))))
  ic50data_species_specific=data.frame(ic50data_species_specific) #idk why I have to end up doing this
  ic50data_long_model=rbind(ic50data_long_model,ic50data_species_specific)
}
ic50data_long=ic50data_long_model

#In the next step, I'm ordering mutants by decreasing resposne to the 625nM dose. Then I use this to change the levels of the species factor from more to less resistant. This helps with ggplot because now I can color the mutants with decreasing resistance
ic50data_long_625=ic50data_long%>%filter(conc==.625)
ic50data_long$species=factor(ic50data_long$species,levels = as.character(ic50data_long_625$species[order((ic50data_long_625$y_model),decreasing = T)]))

#Adding drug effect
##########Changed this on 2/20. Using y from 4 parameter logistic rather than raw values
ic50data_long=ic50data_long%>%
  filter(!species=="Wt")%>%
  mutate(drug_effect=-log(y_model)/72)

#Adding Net growth rate
ic50data_long$netgr_pred=.05-ic50data_long$drug_effect


```
####Plotting modeled dose responses  
```{r,eval=F}
getPalette = colorRampPalette(brewer.pal(9, "Spectral"))
plotly=ggplot(ic50data_long,aes(x=log(conc),color=factor(species)))+
  facet_wrap(~factor(species))+
  geom_line(aes(y=y_model))+
  geom_point(aes(y=y))+
  cleanup+
  scale_color_manual(values = getPalette(length(unique(ic50data_long$species))))+
  theme(axis.text = element_blank(),
        axis.ticks = element_blank())
ggplotly(plotly)
```
####Plotting species changes in dose responses over different concentrations  
```{r,eval=F}
plotly=ggplot(ic50data_long,aes(x=species,y=y_model))+
  facet_wrap(~factor(conc))+
  geom_col(aes(fill=factor(species)))+
  cleanup+
  scale_fill_manual(values = getPalette(length(unique(ic50data_long$species))))+
  theme(axis.text = element_blank(),
        axis.ticks = element_blank())
ggplotly(plotly)
```


###Dose response curve fitting with 4-parameter logistic
####Second iteration: Get y_model for predefined concentration ranges of interest
```{r}
conc.list=seq(.5,1.5,by=.1)
ic50.model.pred=data.frame(matrix(NA,nrow=length(conc.list)*length(unique(ic50data_long$species)),ncol=0))
for(species_curr in sort(unique(ic50data_long$mutant))){
  ic50data_species_specific=ic50data_long%>%filter(mutant==species_curr)
  #Next: Appproximating Response from dose (inverse of the prediction)
  ic50.ll4=drm(y~conc,data=ic50data_species_specific,fct=LL.3(fixed=c(NA,1,NA)))
  #Extracting coefficients
  b=coef(ic50.ll4)[1]
  c=0
  d=1
  e=coef(ic50.ll4)[2]
  rm(ic50.model.pred.species.specific)
  ic50.model.pred.species.specific=data.frame(matrix(NA,nrow=length(conc.list),ncol=0))
  i=1
  ic50.model.pred.species.specific$mutant=species_curr
  #For loop for the unique concentrations
  for(conc.curr in conc.list){
    ic50.model.pred.species.specific$conc[i]=conc.curr
    ic50.model.pred.species.specific$y_model[i]=c+((d-c)/(1+exp(b*(log(conc.curr)-log(e)))))
    i=i+1
  }
  ic50.model.pred=rbind(ic50.model.pred,ic50.model.pred.species.specific)
}
#Adding drug effect
ic50.model.pred=ic50.model.pred%>%
  filter(!mutant=="Wt")%>%
  mutate(drug_effect=-log(y_model)/72)

    #Adding Net growth rate
    # ic50.model.pred$netgr_pred=.05-ic50.model.pred$drug_effect
ic50data_long=ic50.model.pred
ic50data_all_conc=ic50data_long
```



####Changing the format of the IC50s dataframe so that it matches twinstrand data labeling etc  
####Also converting dose response to expected change in growth rate
####This requires estimating a growth rate without drug. Note that I am using k=0.05 or 14 hours right now.
```{r}
#Variables when making predictions:
#Your assumed fitness without drug
ic50data_long$netgr_pred=net_gr_wodrug-ic50data_long$drug_effect
#Your assumed concentration
ic50data_long=ic50data_long%>%filter(conc==conc_for_predictions) ###Can remove this filter if you wanna look at how well predictions would match up if there was a systematic difference in the concentrations Chuan used and you used in your IC50s
    ##########Changed this on 2/20. Using y from 4 parameter logistic rather than raw values
# ic50data_formerge=ic50data_long%>%filter(!species=="Wt")%>%mutate(drug_effect=-log(y)/72)
# ic50data_formerge=ic50data_long%>%filter(!species=="Wt")%>%mutate(drug_effect=-log(y_model)/72)
```


##Data Parsing-- Duplex Sequencing Data
###Importing Twinstrand Mutation calls dataframe
####The twinstrand dataframe has sampleIDs. I'm merging this dataframe with a 'names' df that has details on what those sample IDs mean
#####Here I also converted genomic coordinates and nucleotide changes to residue changes. I did all of our 20 spike-in mutants and others that I could find.   
#####Other mutants included unique mutants found in the ENU data. i.e. A397P, F311L, F359C, H214R, H396P, K285N, L324R.   
Ideally, in the future I will use Biomart or a similar package that can do this automatically.
Ideally, I'll convert the fasta/bamh files into maf files myself
Got residues and positions from here: #https://www.rcsb.org/pdb/chromosome.do?v=hg38&chromosome=chr9&pos=130862947
One thing that was tripping me up is that I was searching the database based on start position and not end-position
This NCBI tool is also a good resource: https://www.ncbi.nlm.nih.gov/genome/gdv/browser/genome/?id=GCF_000001405.39
However, *this* is probably the best tool to go straight from genomic coordinate to protein change: https://www.ebi.ac.uk/thornton-srv/databases/cgi-bin/DisaStr/GetPage.pl?varmap=TRUE
```{r}


twinstrand_maf_merge=merge(twinstrand_maf,names,by.x = "Sample",by.y = "TwinstrandId")

#Of the 20 mutants, I don't see F486, F359C
twinstrand_maf_merge$mutant=0
twinstrand_maf_merge=twinstrand_maf_merge%>%
  mutate(mutant=case_when(End==130872896 & ALT=="T" ~ "T315I",
                          End==130862970 & ALT=="C" ~ "Y253H",
                          End==130862977 & ALT=="T" ~ "E255V",
                          End==130873004 & ALT=="C" ~ "M351T",
                          End==130862962 & ALT=="A" ~ "G250E",
                          End==130874969 & ALT=="C" ~ "H396P",
                          End==130862955 & ALT=="G" ~ "L248V",
                          End==130874969 & ALT=="G" ~ "H396R",
                          End==130862971 & ALT=="T" ~ "Y253F",
                          End==130862969 & ALT=="T" ~ "Q252H",
                          End==130862976 & ALT=="A" ~ "E255K",
                          End==130872901 & ALT=="C" ~ "F317L",
                          End==130873027 & ALT=="C" ~ "F359L",
                          End==130873027 & ALT=="G" ~ "F359V",
                          End==130873027 & ALT=="A" ~ "F359I",
                          End==130873016 & ALT=="G" ~ "E355G",
                          End==130873016 & ALT=="C" ~ "E355A",
                          End==130878519 & ALT=="A" ~ "E459K",
                          End==130872911 & ALT=="G" ~ "Y320C",
                          End==130872133 & ALT=="G" ~ "D276G",
                          End==130862969 & ALT=="C" ~ "Q252Hsyn", ###The mutants below were found only in the ENU mutagenized pools
                          End==130872885 & ALT=="G" ~ "F311L",
                          End==130873028 & ALT=="G" ~ "F359C",
                          End==130874971 & ALT=="C" ~ "A397P",
                          End==130862854 & ALT=="G" ~ "H214R",
                          End==130872146 & ALT=="C" ~ "V280syn",
                          End==130872161 & ALT=="T" ~ "K285N",
                          End==130872923 & ALT=="G" ~ "L324R",
                          End==130872983 & ALT=="T" ~ "A344D")) #Not observed on D6. Dropped out! Note that D276G looked like it was contaminant DNA because it was barely at detection threshold at D0

#Ordering mutants by level of drug resistance. Note that since we don't know the level of DR for the unique ENU mutants, I have left them out here.
twinstrand_maf_merge$mutant=factor(twinstrand_maf_merge$mutant,levels = c("T315I","Y253H","E255V","M351T","G250E","H396P","L248V","H396R","Y253F","Q252H","E255K","F317L","F359L","F359V","F359I","E355G","E355A","E459K","Y320C","D276G","F311L","F359C","A397P","H214R","K285N","L324R","A344D"))

# twinstrand_maf_merge=twinstrand_maf_merge%>%
#   mutate(mutant=case_when(End==130872896 & ALT=="T" ~ "T315I",
#                           End==130862970 & ALT=="C" ~ "Y253H",
#                           End==130862977 & ALT=="T" ~ "E255V",
#                           End==130873004 & ALT=="C" ~ "M351T",
#                           End==130862962 & ALT=="A" ~ "G250E",
#                           End==130874969 & ALT=="C" ~ "H396P",
#                           End==130862955 & ALT=="G" ~ "L248V",
#                           End==130874969 & ALT=="G" ~ "H396R",
#                           End==130862971 & ALT=="T" ~ "Y253F",
#                           End==130862969 & ALT=="T" ~ "Q252H",
#                           End==130862976 & ALT=="A" ~ "E255K",
#                           End==130872901 & ALT=="C" ~ "F317L",
#                           End==130873027 & ALT=="C" ~ "F359L",
#                           End==130873027 & ALT=="G" ~ "F359V",
#                           End==130873027 & ALT=="A" ~ "F359I",
#                           End==130873016 & ALT=="G" ~ "E355G",
#                           End==130873016 & ALT=="C" ~ "E355A",
#                           End==130878519 & ALT=="A" ~ "E459K",
#                           End==130872911 & ALT=="G" ~ "Y320C",
#                           End==130872133 & ALT=="G" ~ "D276G")) 


#Adding columns for experiment names, experiment frequencies, and time
##############Experiment Name#################
twinstrand_maf_merge$experiment[twinstrand_maf_merge$CustomerName%in%c("M3D0","M3D3","M3D6")]="M3"
twinstrand_maf_merge$experiment[twinstrand_maf_merge$CustomerName%in%c("M4D0","M4D3","M4D6")]="M4"
twinstrand_maf_merge$experiment[twinstrand_maf_merge$CustomerName%in%c("M5D0","M5D3","M5D6")]="M5"
twinstrand_maf_merge$experiment[twinstrand_maf_merge$CustomerName%in%c("M6D0","M6D3","M6D6")]="M6"
twinstrand_maf_merge$experiment[twinstrand_maf_merge$CustomerName%in%c("M7D0","M7D3","M7D6")]="M7"
twinstrand_maf_merge$experiment[twinstrand_maf_merge$CustomerName%in%c("Enu3_D3","Enu3_D6")]="Enu_3"
twinstrand_maf_merge$experiment[twinstrand_maf_merge$CustomerName%in%c("Enu4_D0","Enu4_D3","Enu4_D6")]="Enu_4"
##############Spike in frequency#################
twinstrand_maf_merge$Spike_in_freq[twinstrand_maf_merge$CustomerName%in%c("M3D0","M3D3","M3D6")]=1000
twinstrand_maf_merge$Spike_in_freq[twinstrand_maf_merge$CustomerName%in%c("M4D0","M4D3","M4D6")]=5000
twinstrand_maf_merge$Spike_in_freq[twinstrand_maf_merge$CustomerName%in%c("M5D0","M5D3","M5D6")]=1000
twinstrand_maf_merge$Spike_in_freq[twinstrand_maf_merge$CustomerName%in%c("M6D0","M6D3","M6D6")]=5000
twinstrand_maf_merge$Spike_in_freq[twinstrand_maf_merge$CustomerName%in%c("M7D0","M7D3","M7D6")]=1000
twinstrand_maf_merge$Spike_in_freq[twinstrand_maf_merge$CustomerName%in%c("Enu3_D3","Enu3_D6")]=1000
twinstrand_maf_merge$Spike_in_freq[twinstrand_maf_merge$CustomerName%in%c("Enu4_D0","Enu4_D3","Enu4_D6")]=1000
##############Time point#################
twinstrand_maf_merge$time_point[twinstrand_maf_merge$CustomerName%in%c("M3D0","M6D0","Enu4_D0")]="D0"
twinstrand_maf_merge$time_point[twinstrand_maf_merge$CustomerName%in%c("M3D3","M4D3","M5D3","M6D3","M7D3","Enu3_D3","Enu4_D3")]="D3"
twinstrand_maf_merge$time_point[twinstrand_maf_merge$CustomerName%in%c("M3D6","M4D6","M5D6","M6D6","M7D6","Enu3_D6","Enu4_D6")]="D6"
```
####Converting MAFs of all mutants to counts by using the flow cytometry count data for each experiment.  
```{r}
#To start off converting MAFs into 'Total number of mutant cell' numbers, we will use only mixing experiment 3 as an example.
##########M3##########
twinstrand_maf_merge$totalcells=0
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M3"&twinstrand_maf_merge$time_point=="D0"]=493000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M3"&twinstrand_maf_merge$time_point=="D3"]=1295000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M3"&twinstrand_maf_merge$time_point=="D6"]=13600000
##########M5##########
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M5"&twinstrand_maf_merge$time_point=="D0"]=588000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M5"&twinstrand_maf_merge$time_point=="D3"]=1299000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M5"&twinstrand_maf_merge$time_point=="D6"]=11294000
##########M7##########
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M7"&twinstrand_maf_merge$time_point=="D0"]=611000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M7"&twinstrand_maf_merge$time_point=="D3"]=857000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M7"&twinstrand_maf_merge$time_point=="D6"]=14568000
##########M4##########
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M4"&twinstrand_maf_merge$time_point=="D0"]=405000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M4"&twinstrand_maf_merge$time_point=="D3"]=980000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M4"&twinstrand_maf_merge$time_point=="D6"]=1783000
##########M6##########
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M6"&twinstrand_maf_merge$time_point=="D0"]=510000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M6"&twinstrand_maf_merge$time_point=="D3"]=798000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="M6"&twinstrand_maf_merge$time_point=="D6"]=842000
##########ENU3##########
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="Enu_3"&twinstrand_maf_merge$time_point=="D0"]=166000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="Enu_3"&twinstrand_maf_merge$time_point=="D3"]=1282000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="Enu_3"&twinstrand_maf_merge$time_point=="D6"]=97200000
##########ENU4##########
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="Enu_4"&twinstrand_maf_merge$time_point=="D0"]=316000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="Enu_4"&twinstrand_maf_merge$time_point=="D3"]=1264000
twinstrand_maf_merge$totalcells[twinstrand_maf_merge$experiment=="Enu_4"&twinstrand_maf_merge$time_point=="D6"]=40000000

########Converting MAF to Total Count##########
twinstrand_maf_merge=twinstrand_maf_merge%>%mutate(totalmutant=AltDepth/Depth*totalcells)
```


####Deriving growthrates from twinstrand_maf_merge
```{r}
detach("package:dplyr", character.only = TRUE)
library("dplyr", character.only = TRUE)
twinstrand_simple=twinstrand_maf_merge%>%filter(tki_resistant_mutation=="True",!is.na(mutant),!is.na(experiment))
twinstrand_simple=twinstrand_simple%>%dplyr::select("mutant","experiment","Spike_in_freq","time_point","totalmutant")
twinstrand_simple_cast=dcast(twinstrand_simple,mutant+experiment+Spike_in_freq~time_point,value.var="totalmutant")

twinstrand_simple_cast$d0d3=log(twinstrand_simple_cast$D3/twinstrand_simple_cast$D0)/72
twinstrand_simple_cast$d3d6=log(twinstrand_simple_cast$D6/twinstrand_simple_cast$D3)/72
twinstrand_simple_cast$d0d6=log(twinstrand_simple_cast$D6/twinstrand_simple_cast$D0)/144
#Check if ln(final/initial)/time is the correct formula. Also notice how I'm using days not hours
twinstrand_simple_melt=melt(twinstrand_simple_cast[,-c(4:6)],id.vars=c("mutant","experiment","Spike_in_freq"),variable.name = "duration",value.name = "netgr_obs") #!!!!!!!!!!!value name should be drug effect. And drug effect should be drug_effect_obs i think. NO. I think this should be drug_effect_obs. Fixed 4/2/20
twinstrand_simple_melt$drug_effect_obs=net_gr_wodrug-twinstrand_simple_melt$netgr_obs

# twinstrand_simple_melt_merge=merge(twinstrand_simple_melt,ic50data_formerge,"mutant")
twinstrand_simple_melt_merge=merge(twinstrand_simple_melt,ic50data_long,"mutant")
```
####Saving Dataframes
```{r}
head(twinstrand_maf_merge)
head(twinstrand_simple_melt_merge)
head(ic50data_all_conc)
# write.csv(twinstrand_maf_merge,"twinstrand_maf_merge.csv")
# write.csv(twinstrand_simple_melt_merge,"twinstrand_simple_melt_merge.csv")
# write.csv(ic50data_all_conc,"ic50data_all_conc.csv")
```

